<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Thomas Keaney, Theresa Jones and Luke Holman" />


<title>Sexual selection and the population genetics of a selfish gene</title>

<script src="site_libs/jquery-3.5.1/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/simplex.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.18/datatables.js"></script>
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.20/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.20/js/jquery.dataTables.min.js"></script>
<link href="site_libs/dt-ext-scroller-1.10.20/css/scroller.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-scroller-1.10.20/js/dataTables.scroller.min.js"></script>
<script src="site_libs/pdfmake-1.10.20/pdfmake.js"></script>
<script src="site_libs/pdfmake-1.10.20/vfs_fonts.js"></script>
<link href="site_libs/dt-ext-buttons-1.10.20/css/buttons.dataTables.min.css" rel="stylesheet" />
<script src="site_libs/dt-ext-buttons-1.10.20/js/dataTables.buttons.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.flash.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.html5.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.colVis.min.js"></script>
<script src="site_libs/dt-ext-buttons-1.10.20/js/buttons.print.min.js"></script>
<link href="site_libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sexual selection & the population genetics of a selfish gene</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Supp_methods.html">Supplementary methods</a>
</li>
<li>
  <a href="Empirical_analysis.html">Empirical analysis</a>
</li>
<li>
  <a href="Population_genetic_model.html">Population genetic model</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/SD_sexual_selection">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Sexual selection and the population genetics of a selfish gene</h1>
<h3 class="subtitle">Empirical analysis</h3>
<h4 class="author">Thomas Keaney, Theresa Jones and Luke Holman</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-06-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SD_sexual_selection/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200925code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200925)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200925code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200925)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneySDsexualselectiontree204ad9233d25660bc05fa5f52499c50c06a823c1targetblank204ad92a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/SD_sexual_selection/tree/204ad9233d25660bc05fa5f52499c50c06a823c1" target="_blank">204ad92</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneySDsexualselectiontree204ad9233d25660bc05fa5f52499c50c06a823c1targetblank204ad92a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/SD_sexual_selection/tree/204ad9233d25660bc05fa5f52499c50c06a823c1" target="_blank">204ad92</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rapp.history

Unstaged changes:
    Modified:   SD_crossing_scheme.png
    Modified:   Supplementary_material.Rmd
    Modified:   Supplementary_material.pdf
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Empirical_analysis.Rmd</code>) and HTML (<code>docs/Empirical_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/204ad9233d25660bc05fa5f52499c50c06a823c1/analysis/Empirical_analysis.Rmd" target="_blank">204ad92</a>
</td>
<td>
tkeaney
</td>
<td>
2021-06-02
</td>
<td>
final touches before submission
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/bfbb766ecab13ca6424dc808319c92afb02df293/docs/Empirical_analysis.html" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/0289923cc9148e43c6dac70e6635342b5870183c/analysis/Empirical_analysis.Rmd" target="_blank">0289923</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
<td>
Edits to make the manuscript smaller
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/b74026dc3dac91c652af5efd08c4f5c4d2cc7527/docs/Empirical_analysis.html" target="_blank">b74026d</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/93c7c3a71de35c31657355ebeb80558f67dcd775/analysis/Empirical_analysis.Rmd" target="_blank">93c7c3a</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-26
</td>
<td>
small fixes
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/f30258aac3f1b78957f599e4001cf340ea4a4304/docs/Empirical_analysis.html" target="_blank">f30258a</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/877c09b4c442811e7ffc734deea113689bd61bd1/analysis/Empirical_analysis.Rmd" target="_blank">877c09b</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
<td>
final updates before submission
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/e28d4c6ee418b8caa4ad46c3e2b7ef2f67dc9932/docs/Empirical_analysis.html" target="_blank">e28d4c6</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/Empirical_analysis.html" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/3565002e7456108b162fe417c020d80f199d63d3/analysis/Empirical_analysis.Rmd" target="_blank">3565002</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-the-data" class="section level1">
<h1>Load packages and the data</h1>
<pre class="r"><code>library(tidyverse) # for data wrangling and plotting
library(reshape2) # for the melt function
library(brms) # for Bayesian models
library(tidybayes) # pretty Bayesian stuff
library(pander) # for slick tables
library(kableExtra) # for scrolling tables
library(lubridate) # working with times and durations
library(rcartocolor) # for nice colour palettes
library(patchwork) # for creating figures with multiple panels
library(workflowr) # for website building
library(DT) # for search- and saveable tables</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>Load in the data from each of the experiments and prepare for analysis</p>
<pre class="r"><code># load the drive test data and create the total_female column

drive_test_data &lt;- 
  read.csv(&quot;data/Drive_test.csv&quot;) %&gt;% 
  mutate(total_female = SD.FEMALE + bw.FEMALE) %&gt;% 
  as_tibble()

# reorder the factor levels for the SD variable, in ascending order of drive.

drive_test_data$SD &lt;- 
  factor(drive_test_data$SD, levels = c(&quot;W1118&quot;, &quot;MAD&quot;, &quot;72&quot;, &quot;5&quot;))

drive_test_data &lt;- 
  drive_test_data %&gt;% 
  mutate(SEX = as.factor(SEX),
         ID = as.factor(ID))

# Load in the data from Experiment 2. Remove unrequired columns, convert durations to a format that R can read, expressed in seconds, and log transform copulation latency.

precop_data &lt;- 
  read.csv(&quot;data/two_choice_test_data.csv&quot;) %&gt;% 
  select(-c(Mating_start_time, Mating_end_time)) %&gt;% 
  mutate(Enter_time = as.numeric(hms(Enter_time)),
         Copulation_latency = as.numeric(hms(Copulation_latency)),
         Copulation_duration = as.numeric(hms(Copulation_duration)),
         Rearing_vial = as.factor(Rearing_vial)) %&gt;% 
  as_tibble()

# reorder the factor levels for the SD variable, in ascending order of drive 

precop_data$SD &lt;- 
  factor(precop_data$SD, levels = c(&quot;W1118&quot;, &quot;MAD&quot;, &quot;72&quot;))

precop_duration_data &lt;-
  precop_data %&gt;% 
  filter(Successful_male == &quot;SD&quot;)

# read in the data from Experiment 3 and remove unrequired columns

postcop_data &lt;- 
  read.csv(&quot;data/SD_mating_data.csv&quot;) %&gt;% 
  select(- c(Start_1, Mating.start_1, Mating.end_1, Start_2, Mating.start_2, Mating.end_2)) %&gt;% 
  as_tibble()

# reorder the factor levels for the SD variable, in ascending order of drive 

postcop_data$SD &lt;- 
  factor(postcop_data$SD, levels = c(&quot;W1118&quot;, &quot;MAD&quot;, &quot;72&quot;, &quot;5&quot;))

# First convert time data into readiable seconds. Then change the mating order variable so that SD is replaced with &#39;first&#39; and LHM is replaced with &#39;second&#39;. Add a censoring column, where 0 = the female remated and 1 = the female did not remate, which can be used in survival models. Finaly, create the total_offspring column and a mating duration difference column.

postcop_data &lt;-
  postcop_data %&gt;% 
    
  # convert into readable seconds
  mutate(Latency_1 = as.numeric(hms(Latency_1)),
         Duration_1 = as.numeric(hms(Duration_1)),
         Latency_2 = as.numeric(hms(Latency_2)),
         Duration_2 = as.numeric(hms(Duration_2))) %&gt;% 
  
  mutate(Mating.Order = if_else(Mating.Order == &quot;SD&quot;, &quot;first&quot;, &quot;second&quot;),
         Censored = ifelse(Remating == &quot;Y&quot;, 0, 1),
         Block = as.factor(Block),
         ID = as.factor(ID),
         Rearing_vial = as.factor(Vial),
         total_offspring = GFP.progeny + Treatment.progeny,
         Mating_duration_diff = if_else(Mating.Order == &quot;first&quot;, -1 * (Duration_2 - Duration_1), Duration_2 - Duration_1))


# create sperm comp data

# remove rows where we could not measure GFP progeny/treatment progeny or total offspring. This occurred when the female failed to mate with the 2nd male (hence there was no sperm competition), when a female escaped or died between mating opportunities, and rarely due to a data recording error. When the total number of offspring produced by a female = 0, we were unable to measure sperm competitive ability.

sperm_comp_data &lt;- 
  postcop_data %&gt;%
  select(ID, Block, Mating.Order, Rearing_vial, SD, Treatment.progeny, GFP.progeny, total_offspring, Duration_1, Duration_2, Mating_duration_diff) %&gt;% 
  filter(!is.na(GFP.progeny)) %&gt;% 
  filter(!(total_offspring == 0))
  

# create remating data

# remove NAs corresponding to females that didn&#39;t mate in the first mating trial or that escaped or died and only include females that mated with SD males first 

remating_data &lt;-
  postcop_data %&gt;% 
  filter(!is.na(Remating),
         Mating.Order == &quot;first&quot;) %&gt;%
  select(ID, Block, Mating.Order, Rearing_vial, SD, Latency_1, Duration_1, Remating, Latency_2, Duration_2, Censored)</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="pilot-experiment" class="section level1">
<h1>Pilot experiment</h1>
<p><strong>Confirming that <em>SD</em> exhibits segregation distortion</strong></p>
<p><span class="math inline">\(~\)</span></p>
<div id="inspect-the-raw-data" class="section level3">
<h3>Inspect the raw data</h3>
<pre class="r"><code># Create a function to build HTML searchable tables

my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons =
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;Keaney_2021_Exp1_data&#39;)),
      pageLength = 119
    )
  )
}


my_data_table(drive_test_data)</code></pre>
<div id="htmlwidget-a926a1e1393f95e2e55b" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-a926a1e1393f95e2e55b">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119"],[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2,2],["F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","M","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F","F"],["5","5","5","5","5","5","5","5","5","5","72","72","72","72","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","5","5","5","5","5","5","5","5","5","5","72","72","72","72","72","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","5","5","5","5","5","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","5","5","5","5","5","72","72","72","72","72","MAD","MAD","W1118","W1118","W1118","W1118","W1118","MAD","MAD","MAD"],[35,27,25,34,25,30,26,30,24,22,38,30,26,30,25,41,38,29,29,33,21,27,35,26,37,33,30,22,25,35,36,32,39,31,41,30,40,31,39,39,30,37,51,48,51,62,67,53,38,59,35,49,25,37,37,34,58,37,50,33,27,15,34,12,42,62,30,53,40,14,36,23,28,34,39,20,30,18,9,50,44,45,48,65,48,46,82,36,43,46,66,36,38,61,43,22,26,39,36,33,33,27,36,31,28,30,29,33,45,34,35,33,33,54,52,42,24,43,28],[12,19,21,18,16,15,19,9,12,16,20,11,12,12,23,17,16,13,17,13,8,8,11,11,14,16,12,13,18,19,14,26,16,15,13,17,20,14,14,33,30,32,61,83,58,53,64,60,41,78,32,52,17,28,28,33,63,44,71,31,35,13,46,11,47,64,34,68,46,17,21,39,23,20,27,22,24,18,11,58,40,64,49,52,57,45,76,39,46,37,59,29,32,68,27,22,20,33,43,17,20,17,21,18,12,24,14,25,11,13,14,9,26,27,24,21,13,22,17],[27,48,41,32,32,38,34,36,32,27,29,31,35,36,36,38,26,29,36,40,29,37,41,28,42,34,31,47,28,37,38,41,40,34,46,33,37,39,33,10,7,10,1,0,0,0,1,1,9,0,31,0,22,0,24,0,2,0,0,26,0,1,14,6,8,2,24,3,5,7,28,34,25,36,24,23,19,17,14,0,20,6,2,0,19,2,0,13,14,0,6,26,32,2,21,21,31,32,41,27,29,32,29,31,36,46,36,37,53,34,46,41,27,34,37,37,29,30,39],[16,17,14,17,13,14,8,15,9,6,18,11,19,11,22,21,8,17,22,13,26,11,14,13,18,13,17,15,23,16,20,22,19,13,17,17,17,16,15,3,4,11,0,0,0,1,0,0,15,0,37,0,22,0,25,0,0,0,0,34,0,1,8,7,1,2,41,2,6,10,24,20,23,47,30,19,27,24,12,0,10,9,0,0,9,0,0,10,9,0,1,13,29,1,22,22,32,40,36,16,16,21,29,16,20,23,21,21,16,23,19,9,25,17,22,27,16,16,18],[21,31,31,28,36,25,23,27,31,27,30,36,35,33,24,25,25,29,32,26,34,27,29,36,28,31,43,32,21,22,31,37,30,25,37,48,33,32,43,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],[62,75,66,66,57,68,60,66,56,49,67,61,61,66,61,79,64,58,65,73,50,64,76,54,79,67,61,69,53,72,74,73,79,65,87,63,77,70,72,49,37,47,52,48,51,62,68,54,47,59,66,49,47,37,61,34,60,37,50,59,27,16,48,18,50,64,54,56,45,21,64,57,53,70,63,43,49,35,23,50,64,51,50,65,67,48,82,49,57,46,72,62,70,63,64,43,57,71,77,60,62,59,65,62,64,76,65,70,98,68,81,74,60,88,89,79,53,73,67]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ID<\/th>\n      <th>DAY<\/th>\n      <th>SEX<\/th>\n      <th>SD<\/th>\n      <th>SD.FEMALE<\/th>\n      <th>SD.MALE<\/th>\n      <th>bw.FEMALE<\/th>\n      <th>bw.MALE<\/th>\n      <th>white.MALE<\/th>\n      <th>total_female<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Keaney_2021_Exp1_data"}],"pageLength":119,"columnDefs":[{"className":"dt-right","targets":[1,4,5,6,7,8,9]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,25,50,100,119]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Column explanations</strong></p>
<p>ID: individual flies.</p>
<p>Day: offspring counts were conducted over two days.</p>
<p>Sex: the sex of the experimental individual.</p>
<p>SD: the variant of the SD allele (or control allele) the experimental individual carried, homologous to the second chromosome carrying the <em>bw</em> mutation.</p>
<p>SD.FEMALE: the number of adult female offspring with red eyes - indicating inheritance of the <em>SD</em> allele</p>
<p>SD.MALE: the number of adult male offspring with red eyes - indicating inheritance of the <em>SD</em> allele</p>
<p>bw.FEMALE: the number of adult female offspring with brown eyes - indicating inheritance of the <em>bw</em> allele</p>
<p>bw.MALE: the number of adult male offspring with brown eyes - indicating inheritance of the <em>bw</em> allele</p>
<p>white.MALE: the number of adult male offspring with white eyes (inheritance of the <em>SD</em> allele is unknown for these offspring when the experimental individual was female)</p>
<p>total_female: the total number of female offspring sired by the experimental individual.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="run-the-model-and-derive-predictions" class="section level3">
<h3>Run the model and derive predictions</h3>
<p>Parameter estimates for the fixed effects in the model are displayed on the log-odds scale.</p>
<pre class="r"><code>drive_model &lt;- 
  brm(SD.FEMALE | trials(total_female) ~ SEX * SD + (1|ID),
      data = drive_test_data, 
      family = binomial,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      iter = 8000,
      warmup = 2000,
      chains = 4,
      cores = 4,
      seed = 2,
      file = &quot;fits/drive_model&quot;)

fixef(drive_model) %&gt;%
  pander(emphasize.strong.rows = c(6:8))</code></pre>
<table style="width:94%;">
<colgroup>
<col width="23%" />
<col width="16%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">-0.01973</td>
<td align="center">0.2122</td>
<td align="center">-0.4401</td>
<td align="center">0.39</td>
</tr>
<tr class="even">
<td align="center"><strong>SEXM</strong></td>
<td align="center">0.1677</td>
<td align="center">0.3015</td>
<td align="center">-0.4168</td>
<td align="center">0.7622</td>
</tr>
<tr class="odd">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">-0.1516</td>
<td align="center">0.3058</td>
<td align="center">-0.7506</td>
<td align="center">0.46</td>
</tr>
<tr class="even">
<td align="center"><strong>SD72</strong></td>
<td align="center">-0.07362</td>
<td align="center">0.3059</td>
<td align="center">-0.668</td>
<td align="center">0.5179</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD5</strong></td>
<td align="center">-0.08937</td>
<td align="center">0.3001</td>
<td align="center">-0.6763</td>
<td align="center">0.5048</td>
</tr>
<tr class="even">
<td align="center"><strong>SEXM:SDMAD</strong></td>
<td align="center"><strong>1.734</strong></td>
<td align="center"><strong>0.4387</strong></td>
<td align="center"><strong>0.8901</strong></td>
<td align="center"><strong>2.624</strong></td>
</tr>
<tr class="odd">
<td align="center"><strong>SEXM:SD72</strong></td>
<td align="center"><strong>2.235</strong></td>
<td align="center"><strong>0.4541</strong></td>
<td align="center"><strong>1.352</strong></td>
<td align="center"><strong>3.135</strong></td>
</tr>
<tr class="even">
<td align="center"><strong>SEXM:SD5</strong></td>
<td align="center"><strong>2.747</strong></td>
<td align="center"><strong>0.4451</strong></td>
<td align="center"><strong>1.888</strong></td>
<td align="center"><strong>3.628</strong></td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(drive_model, type = &quot;hist&quot;, nsamples = 11, binwidth = 2) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20drive%20test%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-drive-test-plot-1">
Past versions of “diagnostic drive test plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-drive-test-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic drive test plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic drive test plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get model predictions</p>
<pre class="r"><code># Define new data for prediction
  
new &lt;-
  expand.grid(total_female = 100, SD = unique(drive_test_data$SD), SEX = unique(drive_test_data$SEX)) %&gt;% 
  mutate(Var1 = 1:8)

# Define new data for prediction with posteriors

new_posterior &lt;-
  expand.grid(total_female = 100, SD = unique(drive_test_data$SD), SEX = unique(drive_test_data$SEX)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:8, sep = &quot;&quot;)) %&gt;% 
  as_tibble()

# Get the summarised predicted means

predictions &lt;- 
  left_join(
    melt(fitted(drive_model, newdata = new, re_formula = NA)) %&gt;% 
      spread(Var2, value), new, by = &quot;Var1&quot;) %&gt;% 
  rename(Sex = SEX, `k - the estimated inheritance (%)` = Estimate) %&gt;% 
  select(-c(Var1, total_female)) %&gt;% 
  select(SD, Sex, `k - the estimated inheritance (%)`, Est.Error, Q2.5, Q97.5)

# Now get the predicted posterior

posterior_prediction &lt;- 
  as_tibble(fitted(drive_model, newdata = new_posterior, re_formula = NA, summary = FALSE)) %&gt;% # 8 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  tidyr::gather(key = id, value = percent_focal_offspring, -posterior_draw) %&gt;%
  left_join(new_posterior %&gt;% select(-total_female), by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The percentage of female offspring that inherited an <em>SD</em> allele from a heterozygous parent <em>k</em>, split by the sex of the parent. <em>SD</em> only drives when present in males. n = 14-15 for all groups.</p>
<pre class="r"><code>predictions %&gt;% 
  mutate(`Variant carried` = recode(predictions$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;),
         Sex = recode(predictions$Sex, &quot;F&quot; = &quot;Female&quot;, &quot;M&quot; = &quot;Male&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;% 
pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table>
<colgroup>
<col width="19%" />
<col width="9%" />
<col width="39%" />
<col width="13%" />
<col width="8%" />
<col width="8%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Sex</th>
<th align="center">k - the estimated inheritance (%)</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">Female</td>
<td align="center">47.31</td>
<td align="center">5.39</td>
<td align="center">36.99</td>
<td align="center">58.11</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">Female</td>
<td align="center">47.7</td>
<td align="center">5.47</td>
<td align="center">36.96</td>
<td align="center">58.38</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">Female</td>
<td align="center">45.78</td>
<td align="center">5.46</td>
<td align="center">35.26</td>
<td align="center">56.7</td>
</tr>
<tr class="even">
<td align="center">W1118</td>
<td align="center">Female</td>
<td align="center">49.51</td>
<td align="center">5.25</td>
<td align="center">39.17</td>
<td align="center">59.63</td>
</tr>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">Male</td>
<td align="center">94.13</td>
<td align="center">1.47</td>
<td align="center">90.88</td>
<td align="center">96.61</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">Male</td>
<td align="center">90.72</td>
<td align="center">2.24</td>
<td align="center">85.7</td>
<td align="center">94.54</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">Male</td>
<td align="center">84.7</td>
<td align="center">3.1</td>
<td align="center">77.96</td>
<td align="center">90.04</td>
</tr>
<tr class="even">
<td align="center">W1118</td>
<td align="center">Male</td>
<td align="center">53.65</td>
<td align="center">5.5</td>
<td align="center">42.72</td>
<td align="center">64.28</td>
</tr>
</tbody>
</table>
<pre class="r"><code>posterior_prediction %&gt;%
    mutate(SD = recode(posterior_prediction$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;),
         SEX = recode(posterior_prediction$SEX, &quot;F&quot; = &quot;Females&quot;, &quot;M&quot; = &quot;Males&quot;)) %&gt;% 
  
  ggplot(aes(SD, percent_focal_offspring)) + 
  stat_eye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  geom_hline(yintercept = 50, linetype = 2) +
  facet_wrap( ~ SEX) + 
  scale_y_continuous(&quot;Estimated inheritance (%)&quot;,
                     breaks = c(40, 50, 60, 70, 80, 90)) +
  xlab(&quot;Variant carried (homologous to bw chromosome)&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-6-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S2</strong>: The estimated percentage of female offspring that inherited an <em>SD</em> allele from a heterozygous parent, split by the sex of the parent. Black points indicate the estimated mean with associated 66 and 95% uncertainty intervals, while coloured area shows the posterior distribution. The dotted line indicates 50% inheritance; the expectation in the absence of segregation distortion.</p>
<p><span class="math inline">\(~\)</span></p>
<p>From these predictions we can calculate viability corrected estimates of <em>k</em> using the formula presented in <a href="https://www.ncbi.nlm.nih.gov/pubmed/1906417">Temin</a> (1991, Genetics).</p>
<p><em>k<sub>c</sub></em> = number of SD progeny / (number of SD progeny + (number of bw progeny / <em>W</em> ))</p>
<p>Where <em>W</em> represents the transmission of the <em>SD</em> allele through females and is defined by:</p>
<p><em>W</em> = number of <em>bw</em> progeny / number of <em>SD</em> progeny</p>
<p><em>k<sub>c</sub></em> values are presented in the table below</p>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Table S2</strong>: the viability corrected inheritance of each <em>SD</em> variant and the control allele from a <em>SD/+</em> (or <em>+/+</em>) male.</p>
<pre class="r"><code>post &lt;- posterior_samples(drive_model) %&gt;% 
  as_tibble()

k_c_SD5 &lt;-
  post %&gt;% 
  select(b_Intercept, b_SD5, `b_SEXM:SD5`) %&gt;% 
  mutate(Female_SD_progeny = inv_logit_scaled(b_Intercept + b_SD5) * 100,
         Female_bw_progeny = 100 - Female_SD_progeny,
         W_SD5 = Female_bw_progeny / Female_SD_progeny,
         Male_SD_progeny = inv_logit_scaled(b_Intercept + `b_SEXM:SD5`) * 100,
         Male_bw_progeny = 100 - Male_SD_progeny) %&gt;% 
  select(Male_SD_progeny, Male_bw_progeny, W_SD5) %&gt;% 
  mutate(k_c = Male_SD_progeny / (Male_SD_progeny + (Male_bw_progeny / W_SD5))) %&gt;% 
  summarise(`k corrected for viability costs` = median(k_c),
            `Q2.5%` = quantile(k_c, probs = 0.025),
            `Q97.5%` = quantile(k_c, probs = 0.975)) %&gt;% 
  mutate(`Variant carried` = &quot;SD-5&quot;)

  

k_c_SD72 &lt;-
  post %&gt;% 
  select(b_Intercept, b_SD72, `b_SEXM:SD72`) %&gt;% 
  mutate(Female_SD_progeny = inv_logit_scaled(b_Intercept + b_SD72) * 100,
         Female_bw_progeny = 100 - Female_SD_progeny,
         W_SD72 = Female_bw_progeny / Female_SD_progeny,
         Male_SD_progeny = inv_logit_scaled(b_Intercept + `b_SEXM:SD72`) * 100,
         Male_bw_progeny = 100 - Male_SD_progeny) %&gt;% 
  select(Male_SD_progeny, Male_bw_progeny, W_SD72) %&gt;% 
  mutate(k_c = Male_SD_progeny / (Male_SD_progeny + (Male_bw_progeny / W_SD72))) %&gt;% 
  summarise(`k corrected for viability costs` = median(k_c),
            `Q2.5%` = quantile(k_c, probs = 0.025),
            `Q97.5%` = quantile(k_c, probs = 0.975)) %&gt;% 
  mutate(`Variant carried` = &quot;SD-72&quot;)


k_c_SDMAD &lt;-
  post %&gt;% 
  select(b_Intercept, b_SDMAD, `b_SEXM:SDMAD`) %&gt;% 
  mutate(Female_SD_progeny = inv_logit_scaled(b_Intercept + b_SDMAD) * 100,
         Female_bw_progeny = 100 - Female_SD_progeny,
         W_SDMAD = Female_bw_progeny / Female_SD_progeny,
         Male_SD_progeny = inv_logit_scaled(b_Intercept + `b_SEXM:SDMAD`) * 100,
         Male_bw_progeny = 100 - Male_SD_progeny) %&gt;% 
  select(Male_SD_progeny, Male_bw_progeny, W_SDMAD) %&gt;% 
  mutate(k_c = Male_SD_progeny / (Male_SD_progeny + (Male_bw_progeny / W_SDMAD))) %&gt;% 
  summarise(`k corrected for viability costs` = median(k_c),
            `Q2.5%` = quantile(k_c, probs = 0.025),
            `Q97.5%` = quantile(k_c, probs = 0.975)) %&gt;% 
  mutate(`Variant carried` = &quot;SD-Mad&quot;)


k_c_W1118 &lt;-
  post %&gt;% 
  select(b_Intercept, b_SEXM) %&gt;% 
  mutate(Female_W1118_progeny = inv_logit_scaled(b_Intercept) * 100,
         Female_bw_progeny = 100 - Female_W1118_progeny,
         W_W1118 = Female_bw_progeny / Female_W1118_progeny,
         Male_W1118_progeny = inv_logit_scaled(b_Intercept + b_SEXM) * 100,
         Male_bw_progeny = 100 - Male_W1118_progeny) %&gt;% 
  select(Male_W1118_progeny, Male_bw_progeny, W_W1118) %&gt;% 
  mutate(k_c = Male_W1118_progeny / (Male_W1118_progeny + (Male_bw_progeny / W_W1118))) %&gt;% 
  summarise(`k corrected for viability costs` = median(k_c),
            `Q2.5%` = quantile(k_c, probs = 0.025),
            `Q97.5%` = quantile(k_c, probs = 0.975)) %&gt;% 
  mutate(`Variant carried` = &quot;W1118&quot;)


bind_rows(k_c_SD5, k_c_SD72, k_c_SDMAD, k_c_W1118) %&gt;% 
  select(`Variant carried`, everything()) %&gt;% 
  pander(round = 3)</code></pre>
<table style="width:88%;">
<colgroup>
<col width="25%" />
<col width="38%" />
<col width="11%" />
<col width="12%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">k corrected for viability costs</th>
<th align="center">Q2.5%</th>
<th align="center">Q97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">0.944</td>
<td align="center">0.818</td>
<td align="center">0.985</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">0.909</td>
<td align="center">0.724</td>
<td align="center">0.975</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">0.868</td>
<td align="center">0.634</td>
<td align="center">0.963</td>
</tr>
<tr class="even">
<td align="center">W1118</td>
<td align="center">0.542</td>
<td align="center">0.397</td>
<td align="center">0.682</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="experiment-1" class="section level1">
<h1>Experiment 1</h1>
<p><strong>Testing the mating success of <em>SD/+</em> males in the presence of a competitor</strong></p>
<p><span class="math inline">\(~\)</span></p>
<div id="inspect-the-raw-data-1" class="section level2">
<h2>Inspect the raw data</h2>
<pre class="r"><code># Create a function to build HTML searchable tables

my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons =
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;Keaney_2021_Exp2_data&#39;)),
      pageLength = 130
    )
  )
}


my_data_table(precop_data)</code></pre>
<div id="htmlwidget-892e731f0d3bf7eee3ce" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-892e731f0d3bf7eee3ce">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43,44,45,46,47,48,49,50,51,52,53,54,55,56,57,58,59,60,61,62,63,64,65,66,67,68,69,70,71,72,73,74,75,76,77,78,79,80,81,82,83,84,85,86,87,88,89,90,91,92,93,94,95,96,97,98,100,101,102,103,104,105,106,107,108,109,110,111,112,113,114,115,116,117,118,119,120,121,122,123,124,125,126,127,128,129,130,131],["MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","72","72","72","72","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","72","72","72","72","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118"],["1","1","1","1","1","1","1","1","1","2","2","2","2","2","2","2","2","2","3","3","3","3","3","3","3","3","3","4","4","4","4","4","4","4","4","4","5","5","5","5","5","5","5","5","5","6","6","6","6","6","7","7","7","7","7","7","7","7","7","8","8","8","8","8","8","8","8","8","9","9","9","9","9","9","9","9","9","10","10","10","10","10","10","10","10","10","11","11","11","11","11","11","11","11","11","12","12","12","12","12","12","12","12","13","13","13","13","13","13","13","13","13","14","14","14","14","14","14","14","14","14","15","15","15","15","15","15","15","15","15"],["9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00","9:42:00"],[110,110,110,110,110,110,110,110,110,294,294,294,294,294,294,294,294,294,551,551,551,551,551,551,551,551,551,838,838,838,838,1448,838,838,838,1448,1448,1448,1448,1448,1448,1448,1448,1448,1448,1448,1448,1448,1943,1943,1943,1943,1943,1943,1943,1943,1943,1943,1943,2864,2864,2864,2864,2864,2864,2864,2864,2864,3218,3218,3218,3218,3218,3218,3218,3218,3218,4095,4095,4095,4095,4095,4095,4095,4095,4095,5058,5058,5058,5058,5058,5058,5058,5058,5058,5058,5058,5058,5058,5553,5553,5553,5553,5553,5553,5553,5553,5553,5553,5553,5553,6439,6439,6439,6439,6439,6439,6439,6439,6439,6439,7170,7170,7170,7170,7170,7170,7170,7170,7170],[1392,960,942,764,1406,230,480,981,497,811,2133,740,1273,1982,609,640,658,1655,2389,422,7673,616,590,null,2676,450,1069,1607,701,715,798,1827,830,1273,813,1552,809,1464,1210,583,869,7036,617,635,10695,3695,2406,599,356,1298,1753,1732,415,10200,466,445,1262,395,1033,377,5310,95,4418,1796,430,445,1407,301,973,1497,1462,994,1012,1517,4092,1030,1071,605,3618,657,691,1758,1992,1675,2640,657,606,662,1758,606,null,1094,682,985,1788,6784,3305,null,2794,1666,240,1317,556,262,3277,1115,null,516,1706,4413,1145,null,3100,2113,907,475,1594,2347,799,501,3129,2446,null,523,3854,1406,null,5573,1784,2422],[1860,1056,1478,1615,1544,1358,1599,1078,1540,1490,1308,1181,1493,1468,1627,1596,1537,1111,1410,1516,1613,1428,1454,null,666,1468,1742,1299,1823,1507,959,1548,1392,949,1409,1350,273,1438,1692,1713,1427,1453,995,1661,1604,1398,2079,1315,1594,1361,1147,927,1386,1734,1941,1356,1397,1555,1374,1226,1484,1643,1317,1273,1529,1860,1662,1185,1742,1218,1253,2012,1703,1198,1289,1976,1644,3066,1945,1472,1147,1118,2106,852,1372,417,976,821,950,935,null,389,801,1389,1033,1048,1474,null,1806,1380,1578,1237,1262,1556,822,764,null,1363,1599,1313,1068,null,1435,1285,2491,965,1619,1216,869,1659,1020,901,null,1037,1693,1454,null,1004,1563,925],["LBw","LBw","SD","LBw","LBw","LBw","LBw","SD","LBw","SD","SD","SD","LBw","LBw","LBw","LBw","LBw","SD","LBw","SD","SD","LBw","SD",null,"LBw","LBw","LBw","LBw","LBw","SD","SD","SD","SD","LBw","LBw","LBw","LBw","SD","LBw","LBw","SD","SD","SD","SD","SD","SD","LBw","SD","LBw","SD","SD","SD","LBw","LBw","LBw","LBw","LBw","LBw","SD","LBw","SD","LBw","SD","SD","LBw","LBw","LBw","LBw","LBw","SD","LBw","LBw","LBw","SD","SD","LBw","SD","LBw","SD","SD","LBw","SD","LBw","SD","LBw","SD","SD","SD","LBw","LBw",null,"LBw","LBw","SD","LBw","LBw","LBw",null,"LBw","SD","SD","LBw","SD","LBw","LBw","SD",null,"SD","SD","SD","LBw",null,"LBw","LBw","LBw","LBw","LBw","SD","SD","LBw","LBw","LBw",null,"LBw","SD","LBw",null,"LBw","LBw","LBw"],["Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","N","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","N","Y","Y","Y","Y","Y","Y","N","Y","Y","Y","Y","Y","Y","Y","Y","N","Y","Y","Y","Y","N","Y","Y","Y","Y","Y","Y","Y","Y","Y","Y","N","Y","Y","Y","N","Y","Y","Y"],["4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:51","4:03:41","4:03:41","4:03:41","4:03:41","4:03:51","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:41","4:03:51","4:03:41","4:03:41","4:03:41","4:03:51","4:03:41","4:03:41","4:03:41"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ID<\/th>\n      <th>SD<\/th>\n      <th>Rearing_vial<\/th>\n      <th>Experiment_start<\/th>\n      <th>Enter_time<\/th>\n      <th>Copulation_latency<\/th>\n      <th>Copulation_duration<\/th>\n      <th>Successful_male<\/th>\n      <th>Mating_occurred<\/th>\n      <th>Finish_time<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Keaney_2021_Exp2_data"}],"pageLength":130,"columnDefs":[{"className":"dt-right","targets":[0,4,5,6]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,25,50,100,130]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Column explanations</strong></p>
<p>ID: individual flies.</p>
<p>SD: the variant of the SD allele (or control allele) the experimental individual carried, homologous to the second chromosome carrying the <em>bw</em> mutation.</p>
<p>Rearing_vial: the vial the individual male developed in. This variable is included to capture variation explained by the rearing environment e.g. small differences in food moisture content or quantity.</p>
<p>Experiment_start: I started data collection for Experiment 2 on the 10/12/2020 at 09:42 AEDT - lights were turned on at 07:00 AEDT.</p>
<p>Enter_time: the time in seconds after the start of the experiment that a female was introduced into a vial containing a <em>SD/+</em> or control (<em>w<sup>1118</sup></em>) male and a <em>LBw</em> male e.g. a value of 270 indicates that the female was added to the vial 270 seconds after the start of the experiment.</p>
<p>Copulation_latency: the time in seconds from a females entrance to a vial to the onset of copulation (Mating_start_time - Enter_time).</p>
<p>Copulation_duration: the time in seconds that the female mated with one of the two males (Mating_end_time - Mating_start_time).</p>
<p>Successful_male: the male that successfully mated with the female</p>
<p>Mating_occurred: did the female mate with one of the two males before the experiment ended?</p>
<p>Finish_time: the time in hms at which the Experiment ended.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="mating-success" class="section level2">
<h2>Mating success</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we model mating success as a YES/No response variable.</p>
<p><span class="math inline">\(~\)</span></p>
<div id="run-the-model-and-derive-predictions-1" class="section level3">
<h3>Run the model and derive predictions</h3>
<p>Parameter estimates for the fixed effects in the model are displayed on the log-odds scale.</p>
<pre class="r"><code>precop_success_model &lt;- 
  brm(Successful_male ~ SD + (1 | Rearing_vial),
      data = precop_data, 
      family = bernoulli,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      iter = 8000,
      warmup = 2000,
      chains = 4,
      cores = 4,
      seed = 2,
      file = &quot;fits/precop_success_model&quot;)

fixef(precop_success_model) %&gt;%
  pander()</code></pre>
<table style="width:82%;">
<colgroup>
<col width="22%" />
<col width="15%" />
<col width="16%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">-0.5512</td>
<td align="center">0.3672</td>
<td align="center">-1.282</td>
<td align="center">0.1607</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">0.2022</td>
<td align="center">0.5175</td>
<td align="center">-0.8132</td>
<td align="center">1.201</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">0.2804</td>
<td align="center">0.5311</td>
<td align="center">-0.7346</td>
<td align="center">1.325</td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(precop_success_model, type = &quot;hist&quot;, nsamples = 11, binwidth = 0.5) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20mating%20success%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-mating-success-plot-1">
Past versions of “diagnostic mating success plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-mating-success-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic mating success plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic mating success plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_precop_success &lt;-
  expand.grid(SD = unique(precop_data$SD)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:3, sep = &quot;&quot;)) %&gt;% 
  as_tibble()

# Get predicted means

predictions_precop_success &lt;- 
  as.data.frame(fitted(precop_success_model, newdata = new_posterior_precop_success, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = prop_SD_success, -posterior_draw) %&gt;%
  left_join(new_posterior_precop_success, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  mutate(percent_SD_success = prop_SD_success * 100) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD) %&gt;% 
  summarise(`Estimated % of males mating` = median(percent_SD_success),
            `2.5%` = quantile(percent_SD_success, probs = 0.025),
            `97.5%` = quantile(percent_SD_success, probs = 0.975))


# Now get the predicted posterior

posterior_prediction_precop_success &lt;- 
  as.data.frame(fitted(precop_success_model, newdata = new_posterior_precop_success, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = prop_SD_success, -posterior_draw) %&gt;%
  left_join(new_posterior_precop_success, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  mutate(percent_SD_success = prop_SD_success * 100) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The estimated percentage of <em>SD/+</em> or control males that successfully mated with a female when competing with a single <em>LBw</em> male.</p>
<pre class="r"><code>predictions_precop_success %&gt;% 
  mutate(`Variant carried` = recode(predictions_precop_success$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;%
  pander(split.cell = 40, split.table = Inf)</code></pre>
<table style="width:89%;">
<colgroup>
<col width="25%" />
<col width="41%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Estimated % of males mating</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">36.6</td>
<td align="center">21.73</td>
<td align="center">54.01</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">41.5</td>
<td align="center">25.18</td>
<td align="center">58.79</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">43.28</td>
<td align="center">26.04</td>
<td align="center">62.52</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_ms &lt;- posterior_samples(precop_success_model)

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the inv_logit_scaled() function converts the posterior draws onto the response scale 

post_diff_ms &lt;- post_ms %&gt;% 
  mutate(p_w1118 =  inv_logit_scaled(b_Intercept),
         p_SD_MAD = inv_logit_scaled(b_SDMAD + b_Intercept),
         p_SD_72 = inv_logit_scaled(b_SD72 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = `% difference`) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;))</code></pre>
<p>The mating success of <em>SD/+</em> males relative to the <em>w<sup>1118</sup></em> control males. A value of 2 means that <em>SD/+</em> males were twice as successful as <em>w<sup>1118</sup></em> control males.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_ms %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mating success relative to w1118`  = mean(`% difference`),
            `2.5%` = quantile(`% difference`, probs = 0.025),
            `97.5%` = quantile(`% difference`, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table style="width:88%;">
<colgroup>
<col width="18%" />
<col width="48%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mating success relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">1.24</td>
<td align="center">0.64</td>
<td align="center">2.25</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">1.19</td>
<td align="center">0.61</td>
<td align="center">2.08</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure 1a and b</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel a</p>
<pre class="r"><code>mean_mating_success_plot &lt;-
  posterior_prediction_precop_success %&gt;% 
  mutate(SD = recode(posterior_prediction_precop_success$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;)) %&gt;%
  
  ggplot(aes(SD, percent_SD_success)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;)) +
  coord_flip() +
  ylab(&quot;Estimated mating success (%) when competing\n against a single standard competitor&quot;) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel b</p>
<pre class="r"><code>Mating_success_logodds &lt;-
  post_ms %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72) %&gt;% 
  gather(key = parameter, value = logodds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;))) %&gt;%
  
  ggplot(aes(parameter, logodds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(breaks = c(-1, 0, 1, 2, 3)) +
  xlab(NULL) +
  ylab(&quot;Log-odds difference from w1118&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank()) </code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="mating-latency" class="section level2">
<h2>Mating latency</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we take a more in depth look at mating success by looking at mating latency - the duration between a female’s introduction to the competing males’ vial and the onset of copulation.</p>
<p>Fit the model using mating latency as the response variable and specifying a <em>weibull distribution</em> (a time-to-event model).</p>
<p>Fixed effects are shown on the odds scale <strong>(the weibull model uses a log-link function rather than a logit-link function)</strong></p>
<pre class="r"><code># Fit the model

mating_latency_survival_model&lt;- 
  brm(Copulation_latency ~ SD + (1|Rearing_vial),
      data = precop_duration_data,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      family = weibull, inits = 0,
      cores = 4, chains = 4, iter = 8000, warmup = 2000,
      control = list(adapt_delta = 0.99, max_treedepth = 10),
      seed = 1, file = &quot;fits/mating_latency_survival_model&quot;)

# display model results

fixef(mating_latency_survival_model) %&gt;%
  pander()</code></pre>
<table style="width:81%;">
<colgroup>
<col width="22%" />
<col width="15%" />
<col width="16%" />
<col width="12%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">7.946</td>
<td align="center">0.2658</td>
<td align="center">7.426</td>
<td align="center">8.484</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">-0.6498</td>
<td align="center">0.3659</td>
<td align="center">-1.364</td>
<td align="center">0.08986</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">-0.4926</td>
<td align="center">0.3682</td>
<td align="center">-1.218</td>
<td align="center">0.2387</td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(mating_latency_survival_model, type = &quot;hist&quot;, nsamples = 11, binwidth = 1500) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20mating%20latency%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-mating-latency-plot-1">
Past versions of “diagnostic mating latency plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-mating-latency-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic mating latency plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic mating latency plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_mating_latency &lt;-
  expand.grid(SD = unique(precop_duration_data$SD)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:3, sep = &quot;&quot;)) %&gt;% 
  as_tibble()
  

# Get predicted means

predictions_mating_latency &lt;- 
  as.data.frame(fitted(mating_latency_survival_model, newdata = new_posterior_mating_latency, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_mating_latency, -posterior_draw) %&gt;%
  mutate(mean_mating_latency = mean_mating_latency / 60) %&gt;%
  left_join(new_posterior_mating_latency, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD) %&gt;% 
  summarise(`Estimated time to mating (mins)` = median(mean_mating_latency),
            `2.5%` = quantile(mean_mating_latency, probs = 0.025),
            `97.5%` = quantile(mean_mating_latency, probs = 0.975))


# Now get the predicted posterior

posterior_prediction_mating_latency &lt;- 
  as.data.frame(fitted(mating_latency_survival_model, newdata = new_posterior_mating_latency, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_mating_latency, -posterior_draw) %&gt;%
  mutate(mean_mating_latency = mean_mating_latency / 60) %&gt;% # if we want to express in secs then delete this
  left_join(new_posterior_mating_latency, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The mean time taken for <em>SD/+</em> or control males to mate with a <em>LH<sub>m</sub></em> female.</p>
<pre class="r"><code>predictions_mating_latency %&gt;%
  mutate(`Variant carried` = recode(predictions_mating_latency$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;%
  pander(split.cell = 40, split.table = Inf, round = 0)</code></pre>
<table style="width:93%;">
<colgroup>
<col width="25%" />
<col width="47%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Estimated time to mating (mins)</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">47</td>
<td align="center">28</td>
<td align="center">81</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">24</td>
<td align="center">15</td>
<td align="center">41</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">29</td>
<td align="center">17</td>
<td align="center">49</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_ml &lt;- posterior_samples(mating_latency_survival_model) %&gt;% 
  as_tibble()

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the exp() function converts the posterior draws onto the response scale 

post_diff_ml &lt;- post_ml %&gt;% 
  mutate(p_w1118 =  exp(b_Intercept),
         p_SD_MAD = exp(b_SDMAD + b_Intercept),
         p_SD_72 = exp(b_SD72 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = proportion) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;))</code></pre>
<p>The mean mating latency of males that successfully mated with <em>LH<sub>m</sub></em> females when competing against <em>Lbw</em> males, relative to the <em>w<sup>1118</sup></em> control males. A value of 0.5 means that males mated in half the time of <em>w<sup>1118</sup></em> males.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_ml %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean mating latency relative to w1118`  = mean(proportion),
            `2.5%` = quantile(proportion, probs = 0.025),
            `97.5%` = quantile(proportion, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table style="width:94%;">
<colgroup>
<col width="18%" />
<col width="55%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean mating latency relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">0.66</td>
<td align="center">0.3</td>
<td align="center">1.27</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">0.56</td>
<td align="center">0.26</td>
<td align="center">1.09</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure S3</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel a</p>
<pre class="r"><code>mean_mating_latency_plot &lt;-
  posterior_prediction_mating_latency %&gt;%
  mutate(SD = recode(posterior_prediction_mating_latency$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;%
  ggplot(aes(SD, mean_mating_latency)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip(ylim = c(0, 100)) +
  ylab(&quot;Estimated time to mating\n (minutes)&quot;) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel b</p>
<pre class="r"><code>mating_latency_odds &lt;- 
  post_ml %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72) %&gt;% 
  gather(key = parameter, value = logodds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;))) %&gt;%
  
  
  ggplot(aes(parameter, logodds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(&quot;Odds difference from w1118&quot;,
                     breaks = c(-2, -1, 0, 1)) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())   </code></pre>
<p>Combine panels into a single plot</p>
<pre class="r"><code>mean_mating_latency_plot /
mating_latency_odds + plot_annotation(tag_levels = &quot;a&quot;)</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-17-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S3:</strong> The duration required for an <em>SD/+</em> male to mate with a <em>LH<sub>m</sub></em> female and how this compares to <em>w<sup>1118</sup></em> control males. Panel <strong>a</strong> shows the estimated mating latency for <em>SD/+</em> and control males. Panel <strong>b</strong> shows effect sizes on the odds scale for the <em>SD</em> variants. Negative values indicate that <em>SD/+</em> males mated faster than <em>w<sup>1118</sup></em> control males. Black points indicate the estimated mean with associated 66 and 95% uncertainty intervals, while coloured area shows the posterior distribution.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="copulation-duration" class="section level2">
<h2>Copulation duration</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we take a more in depth look at mating success by looking at copulation duration - an indicator of male investment in mating.</p>
<p>Fit the model using copulation duration as the response variable and specifying a <em>weibull distribution</em> (a time-to-event model).</p>
<p>Fixed effects are shown on the odds scale <strong>(the weibull model uses a log-link function rather than a logit-link function)</strong></p>
<pre class="r"><code>copulation_duration_survival_model&lt;- 
  brm(Copulation_duration ~ SD + (1|Rearing_vial),
      data = precop_duration_data,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      family = weibull, inits = 0,
      cores = 4, chains = 4, iter = 8000, warmup = 2000,
      control = list(adapt_delta = 0.99, max_treedepth = 10),
      seed = 1, file = &quot;fits/copulation_duration_survival_model&quot;)

# display model results

fixef(copulation_duration_survival_model) %&gt;%
  pander()</code></pre>
<table style="width:83%;">
<colgroup>
<col width="22%" />
<col width="16%" />
<col width="16%" />
<col width="13%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">7.176</td>
<td align="center">0.05795</td>
<td align="center">7.062</td>
<td align="center">7.289</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">-0.008013</td>
<td align="center">0.07812</td>
<td align="center">-0.1619</td>
<td align="center">0.1437</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">-0.009322</td>
<td align="center">0.07935</td>
<td align="center">-0.1649</td>
<td align="center">0.1495</td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(copulation_duration_survival_model, type = &quot;hist&quot;, nsamples = 11, binwidth = 200) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20copulation%20duration%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-copulation-duration-plot-1">
Past versions of “diagnostic copulation duration plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-copulation-duration-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic copulation duration plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic copulation duration plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_copulation_duration &lt;-
  expand.grid(SD = unique(precop_duration_data$SD)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:3, sep = &quot;&quot;)) %&gt;% 
  as_tibble()
  

# Get predicted means

predictions_copulation_duration &lt;- 
  as.data.frame(fitted(copulation_duration_survival_model, newdata = new_posterior_copulation_duration, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_copulation_duration, -posterior_draw) %&gt;%
  mutate(mean_copulation_duration = mean_copulation_duration / 60) %&gt;%
  left_join(new_posterior_copulation_duration, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD) %&gt;% 
  summarise(`Estimated copulation duration (mins)` = median(mean_copulation_duration),
            `2.5%` = quantile(mean_copulation_duration, probs = 0.025),
            `97.5%` = quantile(mean_copulation_duration, probs = 0.975))


# Now get the predicted posterior

posterior_prediction_copulation_duration &lt;- 
  as.data.frame(fitted(copulation_duration_survival_model, newdata = new_posterior_copulation_duration, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_copulation_duration, -posterior_draw) %&gt;%
  mutate(mean_copulation_duration = mean_copulation_duration / 60) %&gt;% # if we want to express in secs then delete this
  left_join(new_posterior_copulation_duration, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The mean copulation duration when a <em>SD/+</em> or control male mated with a <em>LH<sub>m</sub></em> female.</p>
<pre class="r"><code>predictions_copulation_duration %&gt;%
  mutate(`Variant carried` = recode(predictions_copulation_duration$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;%
  pander(split.cell = 40, split.table = Inf, round = 1)</code></pre>
<table>
<colgroup>
<col width="25%" />
<col width="54%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Estimated copulation duration (mins)</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">21.8</td>
<td align="center">19.5</td>
<td align="center">24.4</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">21.6</td>
<td align="center">19.3</td>
<td align="center">24.1</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">21.6</td>
<td align="center">19.4</td>
<td align="center">24.2</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_cd &lt;- posterior_samples(copulation_duration_survival_model) %&gt;% 
  as_tibble()

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the inv_logit_scaled() function converts the posterior draws onto the response scale 

post_diff_cd &lt;- post_cd %&gt;% 
  mutate(p_w1118 =  exp(b_Intercept),
         p_SD_MAD = exp(b_SDMAD + b_Intercept),
         p_SD_72 = exp(b_SD72 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = proportion) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;))</code></pre>
<p>The mean copulation duration for <em>SD/+</em> males that successfully mated with <em>LH<sub>m</sub></em> females when competing against <em>Lbw</em> males, relative to the <em>w<sup>1118</sup></em> control males. A value of 0.5 means that males mated in half the time of <em>w<sup>1118</sup></em> males.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_cd %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean copulation duration relative to w1118`  = mean(proportion),
            `2.5%` = quantile(proportion, probs = 0.025),
            `97.5%` = quantile(proportion, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 3)</code></pre>
<table style="width:94%;">
<colgroup>
<col width="18%" />
<col width="54%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean copulation duration relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">0.993</td>
<td align="center">0.849</td>
<td align="center">1.162</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">0.995</td>
<td align="center">0.851</td>
<td align="center">1.157</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure S4</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel a</p>
<pre class="r"><code>mean_copulation_duration_plot &lt;-
  posterior_prediction_copulation_duration %&gt;%
  mutate(SD = recode(posterior_prediction_copulation_duration$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;%
  ggplot(aes(SD, mean_copulation_duration)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip(ylim = c(15, 30)) +
  ylab(&quot;Estimated copulation duration\n (minutes)&quot;) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel b</p>
<pre class="r"><code>copulation_duration_odds &lt;- 
  post_cd %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72) %&gt;% 
  gather(key = parameter, value = odds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;))) %&gt;%
  
  
  ggplot(aes(parameter, odds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(&quot;Odds difference from w1118&quot;,
                     breaks = c(-0.25, 0, 0.25)) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())   </code></pre>
<p>Combine panels into a single plot</p>
<pre class="r"><code>mean_copulation_duration_plot /
(copulation_duration_odds) + plot_annotation(tag_levels = &quot;a&quot;)</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-23-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S4:</strong> No difference in the duration of mating between a <em>SD/+</em> male and a <em>LH<sub>m</sub></em> female, compared to <em>w<sup>1118</sup></em> control males. Panel <strong>a</strong> shows the estimated copulation duration for <em>SD/+</em> and control males. Panel <strong>b</strong> shows effect sizes on the odds scale for the <em>SD</em> variants. Positive values indicate that <em>SD/+</em> males mated for longer than <em>w<sup>1118</sup></em> control males. Black points indicate the estimated mean with associated 66 and 95% uncertainty intervals, while coloured area shows the posterior distribution.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="experiment-2" class="section level1">
<h1>Experiment 2</h1>
<p><strong>Testing sperm competitive success and female remating frequency</strong></p>
<p><span class="math inline">\(~\)</span></p>
<div id="inspect-the-raw-data-2" class="section level2">
<h2>Inspect the raw data</h2>
<pre class="r"><code># Create a function to build HTML searchable tables

my_data_table &lt;- function(df){
  datatable(
    df, rownames=FALSE,
    autoHideNavigation = TRUE,
    extensions = c(&quot;Scroller&quot;,  &quot;Buttons&quot;),
    options = list(
      dom = &#39;Bfrtip&#39;,
      deferRender=TRUE,
      scrollX=TRUE, scrollY=400,
      scrollCollapse=TRUE,
      buttons =
        list(&#39;pageLength&#39;, &#39;colvis&#39;, &#39;csv&#39;, list(
          extend = &#39;pdf&#39;,
          pageSize = &#39;A4&#39;,
          orientation = &#39;landscape&#39;,
          filename = &#39;Keaney_2021_Exp3_data&#39;)),
      pageLength = 442
    )
  )
}


my_data_table(postcop_data %&gt;% 
                filter(Remating != &quot;NA&quot;) %&gt;% 
                select(ID, Block, Mating.Order, Rearing_vial, SD, Latency_1, Duration_1, Remating, Latency_2, Duration_2, Mating_duration_diff, Treatment.progeny, GFP.progeny, total_offspring, Censored)) </code></pre>
<div id="htmlwidget-8fcfc6a7afb73e276327" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-8fcfc6a7afb73e276327">{"x":{"filter":"none","extensions":["Scroller","Buttons"],"autoHideNavigation":true,"data":[["115","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","144","145","146","147","148","149","150","151","152","153","154","155","156","157","159","160","161","162","163","164","165","166","167","168","169","170","171","172","174","175","176","177","178","179","180","181","182","183","184","185","186","187","190","192","193","194","195","196","197","198","199","200","201","202","203","204","206","208","209","210","211","212","213","214","215","216","217","218","220","221","222","223","224","225","226","227","228","229","230","231","232","233","234","235","236","237","238","241","242","243","244","245","246","247","248","249","250","251","252","253","254","255","256","257","258","259","260","261","262","263","264","265","266","267","268","269","270","271","272","273","274","275","277","278","279","280","281","282","283","284","285","286","287","288","290","291","292","293","294","295","296","297","298","299","300","302","304","305","306","307","308","309","310","311","312","313","314","316","317","318","319","320","321","322","323","324","325","326","328","329","330","331","332","333","334","335","336","337","338","339","340","341","342","343","344","345","346","347","348","349","350","351","352","353","354","355","356","357","358","359","360","361","362","363","364","365","366","367","368","369","370","371","372","373","374","375","376","377","378","379","380","381","382","383","384","385","386","387","388","389","390","391","392","393","394","395","396","397","398","399","400","401","402","403","404","405","406","407","408","409","410","411","412","413","414","415","416","417","418","419","420","421","422","423","424","425","426","427","428","429","430","431","432","433","434","435","436","437","438","439","440","441","442","443","444","445","446","447","448","449","450","451","452","453","454","455","456","457","458","459","460","461","462","463","464","465","466","467","468","469","470","471","472","473","474","475","476","477","478","479","480","481","482","483","484","485","486","487","488","489","490","491","492","493","494","495","496","497","498","499","500","501","502","503","504","505","506","507","508","509","510","511","512","513","514","515","516","517","518","519","520","521","522","523","524","525","526","527","528","529","530","531","532","533","534","535","536","537","538","539","540","541","542","543","544","545","546","547","548","549","550","551","552","553","554","555","556","557","558","559","560","561","562","563","564","565","566","567","568","569","570","571","572","573","574"],["1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","1","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","2","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3","3"],["first","first","first","first","first","second","second","second","second","second","second","second","second","second","first","first","first","first","second","second","second","second","first","first","second","first","first","first","first","second","second","second","second","second","second","second","second","second","first","first","first","second","second","second","second","second","second","second","second","second","first","first","first","first","first","second","second","second","first","first","first","first","first","second","second","second","second","second","second","first","first","second","second","second","second","second","second","second","second","second","second","first","first","first","second","second","second","second","second","second","second","second","second","first","first","first","first","second","second","second","second","second","second","first","first","first","first","first","second","second","second","second","second","first","first","second","second","second","second","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","first","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second","second"],["1","1","1","1","1","1","1","1","1","1","1","1","1","1","2","2","2","2","2","2","2","2","3","3","3","4","4","4","4","4","4","4","4","4","4","4","4","4","5","5","5","5","5","5","5","5","5","5","5","5","6","6","6","6","6","6","6","6","7","7","7","7","7","7","7","7","7","7","7","8","8","8","8","8","8","8","8","8","8","8","8","9","9","9","9","9","9","9","9","9","9","9","9","10","10","10","10","10","10","10","10","10","10","11","11","11","11","11","11","11","11","11","11","12","12","12","12","12","12","16","15","20","17","22","21","18","13","24","14","19","23","16","15","20","17","22","21","18","13","24","14","19","23","16","15","20","22","21","18","24","14","19","23","16","15","22","21","18","13","14","19","15","20","22","21","18","13","24","14","19","23","16","20","21","18","13","24","14","15","20","22","21","13","24","23","20","22","21","24","23","22","21","24","16","15","17","22","21","18","13","24","14","19","23","16","15","20","17","22","21","18","13","24","14","19","23","16","15","20","17","22","21","18","13","24","14","19","23","16","15","20","17","22","21","18","13","24","14","19","23","16","15","20","17","22","21","18","13","24","14","19","23","16","20","17","22","21","18","13","24","19","23","16","20","22","21","18","24","23","20","22","21","18","24","23","25","34","37","26","29","30","32","35","28","36","33","31","27","25","34","37","26","29","32","28","36","33","31","27","25","34","37","26","29","30","32","35","28","36","33","31","27","25","34","37","29","30","32","35","36","33","31","27","25","34","37","26","29","30","35","33","25","30","32","35","31","34","37","30","32","35","36","33","31","37","32","35","36","31","37","35","36","31","34","37","26","29","30","32","35","28","36","34","37","26","29","30","32","35","28","36","34","37","26","29","30","32","35","28","36","34","37","26","29","30","32","35","36","37","26","35","36","37","26","29","30","32","35","36","37","26","29","30","32","35","36","37","32","35","36","37","32","30","32","35","37","37","32","30","32","35","33","31","27","31","27","33","31","27","33","31","27","33","31","33","31","33","31","31","29","30","32"],["5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","5","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","72","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","MAD","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","W1118","5","5","MAD","72","W1118","MAD","72","5","W1118","5","72","W1118","5","5","MAD","72","W1118","MAD","72","5","W1118","5","72","W1118","5","5","MAD","W1118","MAD","72","W1118","5","72","W1118","5","5","W1118","MAD","72","5","5","72","5","MAD","W1118","MAD","72","5","W1118","5","72","W1118","5","MAD","MAD","72","5","W1118","5","5","MAD","W1118","MAD","5","W1118","W1118","MAD","W1118","MAD","W1118","W1118","W1118","MAD","W1118","5","5","72","W1118","MAD","72","5","W1118","5","72","W1118","5","5","MAD","72","W1118","MAD","72","5","W1118","5","72","W1118","5","5","MAD","72","W1118","MAD","72","5","W1118","5","72","W1118","5","5","MAD","72","W1118","MAD","72","5","W1118","5","72","W1118","5","5","MAD","72","W1118","MAD","72","5","W1118","5","72","W1118","5","MAD","72","W1118","MAD","72","5","W1118","72","W1118","5","MAD","W1118","MAD","72","W1118","W1118","MAD","W1118","MAD","72","W1118","W1118","5","MAD","W1118","5","72","72","MAD","W1118","5","W1118","MAD","72","5","5","MAD","W1118","5","72","MAD","5","W1118","MAD","72","5","5","MAD","W1118","5","72","72","MAD","W1118","5","W1118","MAD","72","5","5","MAD","W1118","72","72","MAD","W1118","W1118","MAD","72","5","5","MAD","W1118","5","72","72","W1118","MAD","5","72","MAD","W1118","72","MAD","W1118","72","MAD","W1118","W1118","MAD","72","W1118","MAD","W1118","W1118","72","W1118","W1118","W1118","72","MAD","W1118","5","72","72","MAD","W1118","5","W1118","MAD","W1118","5","72","72","MAD","W1118","5","W1118","MAD","W1118","5","72","72","MAD","W1118","5","W1118","MAD","W1118","5","72","72","MAD","W1118","W1118","W1118","5","W1118","W1118","W1118","5","72","72","MAD","W1118","W1118","W1118","5","72","72","MAD","W1118","W1118","W1118","MAD","W1118","W1118","W1118","MAD","72","MAD","W1118","W1118","W1118","MAD","72","MAD","W1118","MAD","72","5","72","5","MAD","72","5","MAD","72","5","MAD","72","MAD","72","MAD","72","72","72","72","MAD"],[2045,4666,5901,6144,6442,371,1403,3667,1577,5459,5730,996,1800,372,836,2930,2067,3194,323,819,3292,920,4514,7964,418,1687,987,3127,null,732,800,1139,1709,1364,2277,1142,1093,1420,1555,4913,1427,181,699,1473,247,1004,6075,910,527,964,2560,821,3056,1194,2801,273,516,2065,3624,5537,1037,1377,1029,418,355,516,1667,854,3963,2722,4721,10512,819,677,1631,7137,3974,1066,275,1202,1037,4826,3818,2217,732,800,1222,426,2951,1767,1872,1708,399,3918,1638,998,7197,1754,271,2050,920,1618,3399,1683,1074,5163,4632,2846,732,800,1857,3050,3521,1092,3307,346,819,1255,920,4718,1192,1731,740,6572,6820,4248,6130,2245,1096,5197,2206,9987,640,2291,4379,2800,1545,424,2320,3929,4288,3275,3360,5290,2090,1655,3102,2505,3068,2262,10132,5900,5367,1678,1060,6106,6882,3005,10064,3264,1270,3125,2638,8615,1243,8530,5207,1255,8465,2560,6350,6200,2592,1730,3860,893,5940,3098,3676,3686,1908,5700,1415,7735,3937,1273,7586,5601,5714,4302,3486,7495,2277,1254,1195,1496,560,2878,1483,4208,7169,623,6188,399,4355,1910,1671,987,3380,6513,760,2853,782,3965,459,7764,1845,760,746,738,8634,1705,5319,8546,2523,970,1082,515,2070,1095,4629,1841,440,712,1120,1487,3222,2235,3864,465,1233,2185,1094,449,169,4163,2445,4719,2887,878,2122,315,374,8865,575,1552,324,466,1456,1631,1418,724,811,322,1165,549,696,7700,2851,714,1911,640,356,378,5526,4150,629,4644,5590,2307,2875,4699,7276,11019,1725,1010,7602,2004,2494,3755,4237,10581,3717,445,3584,1640,1251,2607,3838,2650,2591,2239,726,3343,2516,3299,11096,1958,6686,6941,870,1595,6837,1799,602,7652,2760,1510,2708,3765,2660,8105,3222,5658,4068,2601,9690,4193,2440,1845,5995,4935,560,2835,1940,2075,2399,3672,1395,1865,4996,5084,7465,6456,5488,820,7435,798,1300,1253,6388,395,755,2487,1310,4215,925,2750,885,875,366,3547,3960,340,320,3492,795,1657,5491,3691,1514,1000,390,1740,1490,830,2260,5420,960,2290,1870,590,1106,656,1111,3083,4444,1870,1602,1832,2178,2837,1712,598,2554,5145,1698,2106,1645,1645,1645,3236,null,820,3155,1060,1945,992,992,667,1614,1163,501,1121,865,865,1475,705,705,1309,1328,2361,685,685,1705,1347,5444,560,560,3525,765,130,693,4475,1245,2360,5411,420,420,null,null,null],[942,1077,1734,1377,1519,1403,1956,1104,1567,1639,1536,1346,429,1318,1298,1404,3589,1019,2024,1282,959,749,869,1237,1433,1109,1392,1019,null,null,1273,462,1594,1936,1523,1494,1350,1040,1262,834,1189,2762,1402,1402,1482,1229,1412,1410,1739,1097,838,1097,1404,2014,846,1855,null,1137,1189,1070,1117,1193,1180,1212,1746,1441,1433,1428,1541,791,1322,855,1272,1014,1096,1495,1929,1343,1915,1664,1524,858,1608,1385,1370,914,1982,1368,1872,1459,997,1348,1392,933,1466,1527,1312,1408,1996,851,894,1446,1370,1527,1117,1180,1045,1647,1369,1157,1430,2658,1663,1207,1702,1205,1282,1021,894,1035,1140,1074,1222,1398,1276,1141,1140,1365,1190,1048,1350,1394,1130,1484,1533,893,1125,1260,1369,1216,1849,714,1162,1190,1170,810,1413,881,1532,1096,1294,1423,1013,1189,1110,896,1703,1705,1333,1462,1395,1122,1651,865,1191,1493,1340,940,1160,990,760,1255,970,950,1325,1022,1261,1575,1119,831,1002,1240,1090,865,1025,970,1493,1664,1535,1250,1163,1160,463,1026,1065,537,967,1122,1010,240,1601,1143,1493,1061,1902,1280,1404,973,1401,1357,1322,1458,748,2358,1538,1605,1910,1581,1575,1270,1785,1190,1493,1080,1377,1222,1097,1436,1470,818,1197,1181,1390,1079,801,826,1353,1170,1889,1000,1486,1370,1276,999,null,1717,1750,1291,1833,1419,1061,1423,1349,1217,1080,1022,1306,1419,990,1444,1197,1030,1229,914,1391,1081,1263,1186,1233,882,1282,1585,1298,1615,1554,770,1005,2835,1300,1665,1795,1145,1356,1911,1365,1460,1099,1135,1350,1613,1110,1606,1040,1047,1600,1305,1240,1060,1475,1231,1509,1665,1430,1430,959,1605,636,1445,1190,1254,1320,1300,1209,927,1295,1361,875,1780,757,1680,1635,1310,720,2472,1280,1475,1265,1340,1560,1150,1199,1032,955,1600,775,1240,631,1223,763,890,1068,1163,1396,1290,1199,2140,1123,1022,1160,690,282,1235,1670,1094,1330,1619,2818,1177,1005,1005,1505,1455,823,2035,1895,2015,990,1230,1741,921,595,1420,2125,1211,1333,1380,1227,1987,1500,1122,1116,2133,1380,875,1375,783,461,1366,1132,1016,1965,1140,1396,1595,1314,1098,1159,1124,1366,1366,1366,1194,null,960,1410,1232,1130,1366,1010,1077,962,1978,1591,1704,1366,1016,1229,1589,1720,2005,1712,1100,1366,1159,1470,2275,1177,1016,1069,1275,1681,1252,1815,1588,1515,975,592,1016,1366,1301,1016,1162],["Y","Y","Y","Y","N","N","N","Y","N","N","Y","N","Y","N","Y","Y","N","Y","Y","Y","N","Y","Y","Y","Y","N","N","N","Y","Y","Y","Y","N","Y","Y","Y","Y","Y","N","Y","N","Y","Y","Y","Y","Y","N","N","Y","N","N","Y","N","Y","N","Y","Y","N","Y","N","Y","N","N","Y","Y","N","Y","Y","Y","Y","Y","Y","N","Y","Y","Y","Y","N","N","Y","Y","N","N","Y","N","Y","Y","N","Y","N","Y","Y","N","N","N","N","N","Y","Y","Y","N","Y","N","Y","N","N","N","Y","N","Y","Y","Y","N","Y","N","Y","Y","Y","Y","Y","Y","Y","Y","N","N","N","Y","Y","N","N","N","Y","Y","N","Y","N","N","N","N","N","Y","Y","N","Y","N","N","N","N","Y","N","N","Y","Y","Y","N","N","N","N","N","N","Y","Y","N","N","N","N","N","N","N","N","N","N","Y","N","N","Y","N","N","Y","N","Y","N","Y","N","N","N","N","N","N","Y","N","N","N","N","N","N","N","N","Y","N","N","N","Y","Y","Y","N","Y","N","N","Y","N","Y","Y","Y","Y","Y","N","N","N","Y","Y","Y","Y","Y","N","N","Y","N","N","N","N","N","Y","Y","N","N","Y","N","N","Y","Y","Y","Y","N","Y","Y","N","Y","Y","N","N","Y","N","N","N","N","N","N","N","Y","N","N","N","N","N","Y","N","N","N","N","Y","N","N","N","N","Y","Y","N","Y","Y","Y","Y","Y","Y","N","Y","N","Y","N","Y","N","Y","N","Y","Y","Y","Y","N","Y","Y","Y","Y","Y","N","Y","Y","N","Y","N","Y","Y","N","Y","Y","Y","N","N","Y","Y","N","N","Y","Y","Y","N","N","Y","Y","N","Y","N","Y","Y","N","Y","Y","Y","Y","Y","N","N","Y","N","Y","N","Y","N","Y","N","N","N","Y","N","N","N","Y","N","N","Y","N","Y","Y","N","N","Y","Y","Y","Y","Y","Y","Y","N","Y","Y","N","Y","N","N","Y","Y","N","Y","Y","Y","N","Y","N","Y","N","N","Y","N","N","Y","Y","N","N","N","N","N","Y","N","Y","N","N","Y","Y","N","N","N","N","N","N","Y","N","N","N","N","Y","N","N","N","N","N","N","Y","N","N","N","N","N","Y","Y","N","N","N","Y","N","N","Y","Y","N"],[370,1688,8061,4129,null,null,null,735,null,null,9362,null,5308,null,801,10709,null,4470,8888,1941,null,1264,6121,2945,2258,null,null,null,4129,698,1822,654,null,9945,693,469,10925,1612,null,1164,null,9391,7880,8117,11309,1169,null,null,9528,null,null,5886,null,7966,null,1275,789,null,3656,null,4462,null,null,536,988,null,5690,408,4429,2370,2888,10787,null,760,3994,3889,788,null,null,1231,1060,null,null,7352,null,10392,636,null,3771,null,5219,899,null,null,null,null,null,344,11863,1499,null,7733,null,6754,null,null,null,4659,null,1060,4425,4296,null,7948,null,322,2174,368,5550,7801,8511,5383,10765,14011,null,13233,10599,6740,null,null,null,1772,6185,null,3239,null,13960,null,null,null,1908,5160,null,1147,null,null,12135,null,10657,null,null,7041,4479,5463,null,null,null,null,null,null,10202,8461,null,null,null,11073,null,null,11966,null,null,null,7932,null,11426,1190,null,null,6679,null,766,null,10019,null,null,null,null,null,null,6203,null,null,null,11758,11008,null,null,15353,756,null,null,null,872,8078,2598,null,9633,12587,null,1929,null,9472,9472,1097,12158,3054,null,null,11760,4906,3680,9192,2663,7972,null,null,10456,14247,13050,null,null,null,10622,588,null,null,485,11178,14730,697,1821,5557,9461,12510,813,9399,null,6467,1113,null,null,1420,null,12863,null,12875,null,null,null,322,13400,12727,null,null,null,5238,12541,null,null,null,6469,null,null,null,null,2039,917,null,567,7383,1215,1256,9568,459,12261,10762,null,1420,null,905,null,3254,null,6677,2232,5579,1235,12391,1030,7642,8442,1468,443,null,8132,2596,null,1182,null,6659,2106,null,10412,2684,1139,null,null,5088,648,null,null,7564,1012,8030,null,null,1637,4578,null,576,null,1487,782,null,9846,6732,1647,1621,3554,null,null,10431,12613,6593,null,10352,null,7314,null,12713,null,7481,null,null,null,498,null,null,729,null,913,4567,11315,null,855,2999,5565,1131,1107,1087,1559,null,3252,1008,null,4688,null,null,8007,579,null,597,3386,5348,null,845,null,7057,null,null,1586,null,null,8577,4110,null,null,null,null,null,7575,null,4128,null,null,2875,2858,null,null,null,null,null,null,1547,null,null,null,null,3445,null,null,null,null,null,null,832,null,null,null,null,null,717,2685,null,null,null,642,null,null,1396,3979,11249],[1710,1176,946,1378,null,null,null,1203,null,null,842,null,324,null,1044,1979,null,1658,1411,1131,null,1649,1710,227,1041,null,null,null,842,1599,1131,1203,null,1378,981,892,946,1229,null,909,null,1479,1573,1274,1781,7091,null,null,1429,null,null,767,null,1271,null,1697,1245,null,1411,null,1207,null,null,1001,1094,null,1174,1298,1174,1411,1668,1386,null,1203,1685,1229,1234,null,null,1685,890,null,null,916,null,1122,1203,null,1229,null,1047,1406,null,null,null,null,null,1180,842,803,null,1226,null,1738,null,null,null,1870,null,1203,1161,1229,null,1081,null,1694,441,1245,1174,1528,1695,732,2569,1238,null,152,1403,1735,null,null,null,1445,158,null,1183,null,536,null,null,null,1664,1081,null,1539,null,null,737,null,1285,null,null,1695,1998,1735,null,null,null,null,null,null,1490,1573,null,null,null,592,null,null,1666,null,null,null,1403,null,1572,1127,null,null,1734,null,1617,null,1332,null,null,null,null,null,null,1483,null,null,null,1791,2740,null,null,1483,1386,null,null,null,1193,1425,1701,null,411,1528,null,1531,null,915,870,1099,1528,839,null,null,1528,1297,334,2616,829,564,null,null,1014,1403,975,null,null,null,1528,1405,null,null,1493,1695,1485,1324,848,1388,1735,1734,917,1014,null,1152,1350,null,null,505,null,1011,null,1403,null,null,null,1080,1070,670,null,null,null,1279,1403,null,null,null,1222,null,null,null,null,2229,1775,null,1257,1692,1423,1911,501,1343,587,832,null,1439,null,1368,null,985,null,1404,1418,1112,1845,1412,1394,1121,1403,2014,1495,null,339,1554,null,1302,null,1251,1620,null,1229,1330,1448,null,null,1609,1844,null,null,1384,1928,1797,null,null,1182,1609,null,1898,null,1522,1652,null,1497,1636,1330,1320,1050,null,null,978,998,1473,null,2047,null,1829,null,1482,null,1254,null,null,null,1613,null,null,1343,null,880,1586,1086,null,1543,1668,1829,1491,1016,1557,522,null,867,1491,null,1797,null,null,1459,1002,null,1559,980,824,null,1113,null,596,null,null,942,null,null,1524,1254,null,null,null,null,null,473,null,1555,null,null,1397,1397,null,null,null,null,null,null,501,null,null,null,null,644,null,null,null,null,null,null,1266,null,null,null,null,null,1473,1371,null,null,null,1266,null,null,687,1397,1429],[-768,-99,788,-1,null,null,null,99,null,null,-694,null,-105,null,254,-575,null,-639,-613,-151,null,900,-841,1010,-392,null,null,null,null,null,-142,741,null,-558,-542,-602,-404,189,null,-75,null,-1283,171,-128,299,5862,null,null,-310,null,null,330,null,743,null,-158,null,null,-222,null,-90,null,null,-211,-652,null,-259,-130,-367,-620,-346,531,null,189,589,-266,-695,null,null,21,-634,null,null,469,null,208,-779,null,-643,null,50,58,null,null,null,null,null,-228,-1154,-48,null,-220,null,-211,null,null,null,-223,null,46,-269,-1429,null,126,null,489,-841,224,280,-493,-555,342,-1347,160,null,989,-263,-370,null,null,null,-51,972,null,350,null,589,null,null,null,185,-367,null,-349,null,null,676,null,247,null,null,-272,-985,-546,null,null,null,null,null,null,-95,-451,null,null,null,901,null,null,-506,null,null,null,-433,null,-247,-105,null,null,-615,null,-615,null,-242,null,null,null,null,null,null,-233,null,null,null,765,1675,null,null,361,376,null,null,null,-300,364,-201,null,-993,555,null,174,null,-543,122,-1259,-10,-766,null,null,-47,27,-1451,1426,-664,-516,null,null,-83,-33,-495,null,null,null,138,326,null,null,140,525,-404,324,-638,18,459,735,null,-703,null,-139,-483,null,null,-918,null,-206,null,381,null,null,null,-364,-127,-360,null,null,null,198,140,null,null,null,-60,null,null,null,null,-1459,-770,null,43,-27,372,-766,855,568,778,628,null,-304,null,245,null,621,null,-357,182,193,-605,-352,81,110,106,-349,-65,null,620,51,null,143,null,3,-300,null,-20,-403,-153,null,null,171,-1087,null,null,-74,-1208,675,null,null,83,-269,null,-748,null,-490,-697,null,-722,-396,-699,-97,-287,null,null,185,398,-183,null,93,null,-807,null,-792,null,-19,null,null,null,-6,null,null,338,null,-625,131,263,null,-352,-347,839,261,-725,636,-73,null,-1258,280,null,417,null,null,-41,-120,null,-574,-400,-51,null,330,null,-770,null,null,-1023,null,null,-71,-60,null,null,null,null,null,-893,null,null,null,null,165,267,null,null,null,null,null,null,-1203,null,null,null,null,-1076,null,null,null,null,null,null,-1009,null,null,null,null,null,221,-444,null,null,null,674,null,null,-614,381,267],[0,0,10,0,null,null,null,43,null,null,21,null,65,null,1,0,null,2,18,26,null,0,0,2,6,null,null,null,1,33,37,39,null,30,16,40,12,44,null,null,null,34,31,46,35,42,null,null,37,null,null,0,null,2,null,24,33,null,2,null,0,null,null,40,32,null,null,28,43,0,1,57,null,35,38,74,28,null,null,1,31,null,null,36,null,0,29,null,24,null,43,24,null,null,null,null,null,30,31,50,null,32,null,15,null,null,null,9,null,32,41,18,null,3,null,37,0,60,34,0,0,2,2,1,null,0,0,0,null,null,null,0,0,null,1,null,0,null,null,null,0,11,null,1,null,null,10,null,14,null,null,2,6,0,null,null,null,null,null,null,13,6,null,null,null,0,null,null,0,null,null,null,9,null,0,4,null,null,0,null,0,null,1,null,null,null,null,null,null,7,null,null,null,23,null,null,null,31,31,null,null,null,15,50,20,null,15,30,null,16,null,26,36,39,44,42,null,null,25,9,22,28,56,19,null,null,23,30,44,null,null,null,20,19,null,null,55,0,5,1,31,17,40,30,65,20,null,61,23,null,null,17,null,64,null,47,null,null,null,29,18,39,null,null,null,42,30,null,null,null,43,null,null,null,null,7,0,null,0,0,0,0,3,0,0,0,null,0,null,0,null,7,null,12,0,3,0,0,0,0,null,0,0,null,1,1,null,0,null,0,1,null,0,1,0,null,null,0,1,null,null,0,0,0,null,null,0,28,null,23,null,2,1,null,1,0,0,0,0,null,null,8,0,0,null,0,null,0,null,6,null,14,null,null,null,29,null,null,29,null,30,32,24,null,0,50,40,42,23,31,26,null,33,28,null,56,null,null,20,36,null,22,4,35,null,24,null,68,null,null,27,null,null,40,17,null,null,null,null,null,25,null,23,null,null,20,1,null,null,null,null,null,null,16,null,null,null,null,19,null,null,null,null,null,null,19,null,null,null,null,null,18,27,null,null,null,31,null,null,4,34,14],[18,0,14,62,null,null,null,5,null,null,14,null,0,null,42,26,null,17,15,1,null,19,42,39,0,null,null,null,41,0,0,0,null,1,2,0,4,2,null,null,null,0,1,2,15,1,null,null,1,null,null,36,null,17,null,1,0,null,19,null,15,null,null,0,0,null,null,0,0,33,20,1,null,0,0,2,0,null,null,0,0,null,null,0,null,0,0,null,1,null,1,0,null,null,null,null,null,0,4,1,null,2,null,12,null,null,null,20,null,0,1,1,null,33,null,0,0,1,3,68,88,23,44,7,null,44,3,0,null,null,null,0,0,null,35,null,1,null,null,null,26,59,null,11,null,null,29,null,67,null,null,36,6,29,null,null,null,null,null,null,55,35,null,null,null,2,null,null,0,null,null,null,35,null,11,21,null,null,45,null,61,null,24,null,null,null,null,null,null,46,null,null,null,19,null,null,null,0,2,null,null,null,1,19,18,null,2,3,null,0,null,5,5,0,0,3,null,null,1,3,0,3,0,7,null,null,0,0,1,null,null,null,10,0,null,null,1,0,7,0,0,3,2,2,1,0,null,0,4,null,null,0,null,0,null,12,null,null,null,0,2,7,null,null,null,20,9,null,null,null,1,null,null,null,null,11,32,null,26,30,24,30,40,25,38,26,null,2,null,27,null,4,null,12,12,31,25,34,7,11,null,0,17,null,7,9,null,4,null,29,26,null,31,27,10,null,null,3,37,null,null,32,34,21,null,null,17,0,null,0,null,10,20,null,27,33,36,4,20,null,null,28,33,44,null,28,null,25,null,27,null,42,null,null,null,0,null,null,0,null,0,0,0,null,0,0,0,0,0,3,1,null,0,8,null,1,null,null,0,0,null,0,0,0,null,0,null,4,null,null,4,null,null,5,1,null,null,null,null,null,1,null,2,null,null,0,0,null,null,null,null,null,null,0,null,null,null,null,0,null,null,null,null,null,null,2,null,null,null,null,null,0,0,null,null,null,0,null,null,1,0,0],[18,0,24,62,null,null,null,48,null,null,35,null,65,null,43,26,null,19,33,27,null,19,42,41,6,null,null,null,42,33,37,39,null,31,18,40,16,46,null,null,null,34,32,48,50,43,null,null,38,null,null,36,null,19,null,25,33,null,21,null,15,null,null,40,32,null,null,28,43,33,21,58,null,35,38,76,28,null,null,1,31,null,null,36,null,0,29,null,25,null,44,24,null,null,null,null,null,30,35,51,null,34,null,27,null,null,null,29,null,32,42,19,null,36,null,37,0,61,37,68,88,25,46,8,null,44,3,0,null,null,null,0,0,null,36,null,1,null,null,null,26,70,null,12,null,null,39,null,81,null,null,38,12,29,null,null,null,null,null,null,68,41,null,null,null,2,null,null,0,null,null,null,44,null,11,25,null,null,45,null,61,null,25,null,null,null,null,null,null,53,null,null,null,42,null,null,null,31,33,null,null,null,16,69,38,null,17,33,null,16,null,31,41,39,44,45,null,null,26,12,22,31,56,26,null,null,23,30,45,null,null,null,30,19,null,null,56,0,12,1,31,20,42,32,66,20,null,61,27,null,null,17,null,64,null,59,null,null,null,29,20,46,null,null,null,62,39,null,null,null,44,null,null,null,null,18,32,null,26,30,24,30,43,25,38,26,null,2,null,27,null,11,null,24,12,34,25,34,7,11,null,0,17,null,8,10,null,4,null,29,27,null,31,28,10,null,null,3,38,null,null,32,34,21,null,null,17,28,null,23,null,12,21,null,28,33,36,4,20,null,null,36,33,44,null,28,null,25,null,33,null,56,null,null,null,29,null,null,29,null,30,32,24,null,0,50,40,42,23,34,27,null,33,36,null,57,null,null,20,36,null,22,4,35,null,24,null,72,null,null,31,null,null,45,18,null,null,null,null,null,26,null,25,null,null,20,1,null,null,null,null,null,null,16,null,null,null,null,19,null,null,null,null,null,null,21,null,null,null,null,null,18,27,null,null,null,31,null,null,5,34,14],[0,0,0,0,1,1,1,0,1,1,0,1,0,1,0,0,1,0,0,0,1,0,0,0,0,1,1,1,0,0,0,0,1,0,0,0,0,0,1,0,1,0,0,0,0,0,1,1,0,1,1,0,1,0,1,0,0,1,0,1,0,1,1,0,0,1,0,0,0,0,0,0,1,0,0,0,0,1,1,0,0,1,1,0,1,0,0,1,0,1,0,0,1,1,1,1,1,0,0,0,1,0,1,0,1,1,1,0,1,0,0,0,1,0,1,0,0,0,0,0,0,0,0,1,1,1,0,0,1,1,1,0,0,1,0,1,1,1,1,1,0,0,1,0,1,1,1,1,0,1,1,0,0,0,1,1,1,1,1,1,0,0,1,1,1,1,1,1,1,1,1,1,0,1,1,0,1,1,0,1,0,1,0,1,1,1,1,1,1,0,1,1,1,1,1,1,1,1,0,1,1,1,0,0,0,1,0,1,1,0,1,0,0,0,0,0,1,1,1,0,0,0,0,0,1,1,0,1,1,1,1,1,0,0,1,1,0,1,1,0,0,0,0,1,0,0,1,0,0,1,1,0,1,1,1,1,1,1,1,0,1,1,1,1,1,0,1,1,1,1,0,1,1,1,1,0,0,1,0,0,0,0,0,0,1,0,1,0,1,0,1,0,1,0,0,0,0,1,0,0,0,0,0,1,0,0,1,0,1,0,0,1,0,0,0,1,1,0,0,1,1,0,0,0,1,1,0,0,1,0,1,0,0,1,0,0,0,0,0,1,1,0,1,0,1,0,1,0,1,1,1,0,1,1,1,0,1,1,0,1,0,0,1,1,0,0,0,0,0,0,0,1,0,0,1,0,1,1,0,0,1,0,0,0,1,0,1,0,1,1,0,1,1,0,0,1,1,1,1,1,0,1,0,1,1,0,0,1,1,1,1,1,1,0,1,1,1,1,0,1,1,1,1,1,1,0,1,1,1,1,1,0,0,1,1,1,0,1,1,0,0,1]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ID<\/th>\n      <th>Block<\/th>\n      <th>Mating.Order<\/th>\n      <th>Rearing_vial<\/th>\n      <th>SD<\/th>\n      <th>Latency_1<\/th>\n      <th>Duration_1<\/th>\n      <th>Remating<\/th>\n      <th>Latency_2<\/th>\n      <th>Duration_2<\/th>\n      <th>Mating_duration_diff<\/th>\n      <th>Treatment.progeny<\/th>\n      <th>GFP.progeny<\/th>\n      <th>total_offspring<\/th>\n      <th>Censored<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"dom":"Bfrtip","deferRender":true,"scrollX":true,"scrollY":400,"scrollCollapse":true,"buttons":["pageLength","colvis","csv",{"extend":"pdf","pageSize":"A4","orientation":"landscape","filename":"Keaney_2021_Exp3_data"}],"pageLength":442,"columnDefs":[{"className":"dt-right","targets":[5,6,8,9,10,11,12,13,14]}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[10,25,50,100,442]}},"evals":[],"jsHooks":[]}</script>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Column explanations</strong></p>
<p>ID: identifier for individual flies.</p>
<p>Block: the experiment was replicated on three consecutive generations of flies, over a 6 week period.</p>
<p>Mating.Order: did the <em>SD/+</em> male mate with the female first or second?</p>
<p>Rearing_vial: the vial within which the individual developed in.</p>
<p>SD: the variant of the <em>SD</em> allele (or control allele) the experimental individual carried.</p>
<p>Latency_1: the time in seconds between the male entering the mating vial and the onset of the first mating.</p>
<p>Duration_1: the time in seconds between the onset and end of the first mating.</p>
<p>Remating: did the female remate within the three hour period, four days after her first mating.</p>
<p>Latency_2: the time in minutes between the male entering the mating vial and the onset of the second mating.</p>
<p>Duration_2: the time in minutes between the onset and end of the second mating.</p>
<p>Treatment.progeny: the number of larvae sired by the <em>SD/+</em> or control male. These larvae did not inherit the <em>UBI</em> GFP construct and therefore did not express green fluorescence.</p>
<p>GFP.progeny: the number of larvae sired by the <em>LH<sub>m</sub><sup>UBU</sup></em> competitor male. These larvae inherited the <em>UBI</em> GFP construct and therefore expressed green fluorescence.</p>
<p>total_offspring: the total number of offspring counted in each vial.</p>
<p>Censored: Identical to remating column except in a 1/0 format for survival model right censoring. 1 = the female remated within three hours, 0 = the female did not remate.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="a-rough-look-at-sperm-precedence" class="section level2">
<h2>A rough look at sperm precedence</h2>
<p>Take a glance at how mating order affected fertilisation success</p>
<pre class="r"><code>  sperm_comp_data %&gt;%
  group_by(Mating.Order) %&gt;% 
  summarise(total_SD = sum(Treatment.progeny),
            total_GFP = sum(GFP.progeny),
            total = sum(total_offspring),
            `Number of mating trios` = n()) %&gt;% 
  mutate(Mating.Order = if_else(Mating.Order == &quot;first&quot;, &quot;SD/+ male mated first&quot;, &quot;LHm-UBI male mated first&quot;)) %&gt;%
  mutate(`% offspring sired by first mate` = if_else(Mating.Order == &quot;SD/+ male mated first&quot;, total_SD / total * 100, total_GFP / total * 100)) %&gt;% 
  rename(`Mating Order` = Mating.Order,
         `Offspring sired by SD/+ males` = total_SD,
         `Offspring sired by LHm males` = total_GFP,
         `Total offspring sired` = total) %&gt;% 
  select(`Mating Order`, `Number of mating trios`, everything()) %&gt;% 
  pander(split.cell = 40, split.table = Inf)</code></pre>
<table>
<colgroup>
<col width="15%" />
<col width="14%" />
<col width="18%" />
<col width="17%" />
<col width="13%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Mating Order</th>
<th align="center">Number of mating trios</th>
<th align="center">Offspring sired by SD/+ males</th>
<th align="center">Offspring sired by LHm males</th>
<th align="center">Total offspring sired</th>
<th align="center">% offspring sired by first mate</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD/+ male mated first</td>
<td align="center">97</td>
<td align="center">294</td>
<td align="center">2545</td>
<td align="center">2839</td>
<td align="center">10.36</td>
</tr>
<tr class="even">
<td align="center">LHm-UBI male mated first</td>
<td align="center">130</td>
<td align="center">4011</td>
<td align="center">308</td>
<td align="center">4319</td>
<td align="center">7.131</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="measuring-p1" class="section level2">
<h2>Measuring P1</h2>
<p>Fit the model.</p>
<p>Parameter estimates for the fixed effects in the model are displayed on the log-odds scale.</p>
<pre class="r"><code># filter so that only trials where the SD male mated with the female first are included

sperm_comp_data_P1 &lt;- 
  sperm_comp_data %&gt;%
  filter(Mating.Order == &quot;first&quot;) %&gt;% 
  mutate(Mating_duration_diff = as.numeric(Mating_duration_diff))
  
# fit the model 

sperm_comp_model_P1 &lt;- brm(Treatment.progeny | trials(total_offspring) ~ SD + Block + (1|Rearing_vial) + (1|ID),
                           data = sperm_comp_data_P1, 
                           family = binomial,
                           prior = c(prior(normal(0, 5), class = Intercept), 
                                     prior(normal(0, 3), class = b)),
                           iter = 8000,
                           warmup = 2000,
                           chains = 4,
                           cores = 4,
                           seed = 2,
                           control = list(adapt_delta = 0.99, max_treedepth = 15),
                           file = &quot;fits/sperm_comp_model_P1&quot;)


# display model results

fixef(sperm_comp_model_P1) %&gt;%
  pander(emphasize.strong.rows = 4)</code></pre>
<table style="width:96%;">
<colgroup>
<col width="22%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">-1.802</td>
<td align="center">1.006</td>
<td align="center">-3.806</td>
<td align="center">0.1443</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">-1.572</td>
<td align="center">1.071</td>
<td align="center">-3.671</td>
<td align="center">0.5493</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">-1.421</td>
<td align="center">1.025</td>
<td align="center">-3.453</td>
<td align="center">0.5931</td>
</tr>
<tr class="even">
<td align="center"><strong>SD5</strong></td>
<td align="center"><strong>-2.465</strong></td>
<td align="center"><strong>0.9809</strong></td>
<td align="center"><strong>-4.462</strong></td>
<td align="center"><strong>-0.5663</strong></td>
</tr>
<tr class="odd">
<td align="center"><strong>Block2</strong></td>
<td align="center">-0.3604</td>
<td align="center">1.024</td>
<td align="center">-2.349</td>
<td align="center">1.659</td>
</tr>
<tr class="even">
<td align="center"><strong>Block3</strong></td>
<td align="center">-1.482</td>
<td align="center">0.952</td>
<td align="center">-3.396</td>
<td align="center">0.3335</td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(sperm_comp_model_P1, type = &quot;hist&quot;, nsamples = 11, binwidth = 2) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20P1%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-P1-plot-1">
Past versions of “diagnostic P1 plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-P1-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic P1 plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic P1 plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the P1 model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_P1 &lt;-
  expand.grid(total_offspring = 100, SD = c(&quot;W1118&quot;, &quot;MAD&quot;, &quot;72&quot;, &quot;5&quot;), Block = unique(sperm_comp_data_P1$Block)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:12, sep = &quot;&quot;)) %&gt;% 
  as_tibble()

predictions_P1 &lt;- 
  as.data.frame(fitted(sperm_comp_model_P1, newdata = new_posterior_P1, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = percent_focal_offspring, -posterior_draw) %&gt;%
  left_join(new_posterior_P1, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD) %&gt;% 
  summarise(`Estimated % of offspring sired` = median(percent_focal_offspring),
            `2.5%` = quantile(percent_focal_offspring, probs = 0.025),
            `97.5%` = quantile(percent_focal_offspring, probs = 0.975))


# Now get the predicted posterior but don&#39;t summarise

posterior_prediction_P1 &lt;- 
  as.data.frame(fitted(sperm_comp_model_P1, newdata = new_posterior_P1, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = percent_focal_offspring, -posterior_draw) %&gt;%
  left_join(new_posterior_P1, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The estimated percentage of offspring sired by <em>SD/+</em> or control males, when they mated first.</p>
<pre class="r"><code>predictions_P1 %&gt;% 
  mutate(`Variant carried` = recode(predictions_P1$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;% 
  pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table style="width:92%;">
<colgroup>
<col width="25%" />
<col width="45%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Estimated % of offspring sired</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">8.17</td>
<td align="center">0.99</td>
<td align="center">44.4</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">1.83</td>
<td align="center">0.17</td>
<td align="center">16.28</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">2.15</td>
<td align="center">0.21</td>
<td align="center">16.97</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">0.78</td>
<td align="center">0.08</td>
<td align="center">5.79</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_P1 &lt;- 
  posterior_samples(sperm_comp_model_P1) %&gt;% 
  as_tibble()

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the inv_logit_scaled() function converts the posterior draws onto the response scale 

post_diff_P1 &lt;- 
  post_P1 %&gt;% 
  mutate(p_w1118 =  inv_logit_scaled(b_Intercept),
         p_SD_MAD = inv_logit_scaled(b_SDMAD + b_Intercept),
         p_SD_72 = inv_logit_scaled(b_SD72 + b_Intercept),
         p_SD_5 = inv_logit_scaled(b_SD5 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118,
         `SD-5` = p_SD_5 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = `% difference`) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))</code></pre>
<p>The mean proportion of offspring sired by each <em>SD</em> variant relative to the <em>w<sup>1118</sup></em> control, when these males mated with <em>LH<sub>m</sub></em> females first. A value of 0.5 means that the proportion of offspring sired was half that of <em>w<sup>1118</sup></em> males.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_P1 %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean proportion of offspring sired relative to w1118`  = mean(`% difference`),
            `2.5%` = quantile(`% difference`, probs = 0.025),
            `97.5%` = quantile(`% difference`, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table style="width:90%;">
<colgroup>
<col width="18%" />
<col width="51%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean proportion of offspring sired relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">0.15</td>
<td align="center">0.02</td>
<td align="center">0.6</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">0.43</td>
<td align="center">0.05</td>
<td align="center">1.65</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">0.39</td>
<td align="center">0.04</td>
<td align="center">1.7</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure 1c and d</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel c</p>
<pre class="r"><code>mean_P1_plot &lt;-
  posterior_prediction_P1 %&gt;%
  mutate(SD = recode(posterior_prediction_P1$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;%
  ggplot(aes(SD, percent_focal_offspring)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-5&quot; = &quot;#4c9b82&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;W1118&quot; = &quot;#d3f2a3&quot;)) +
  coord_flip(ylim = c(0, 50)) +
  ylab(&quot;Estimated % of offspring sired when mating first (P1)&quot;) +
  xlab(&quot;Variant carried&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel d</p>
<pre class="r"><code>P1_logodds &lt;-
  post_P1 %&gt;% 
  mutate(`SD-5` = b_SD5,
         `SD-72` = b_SD72,
         `SD-Mad` = b_SDMAD) %&gt;% 
  gather(key = parameter, value = logodds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))) %&gt;%
  
  ggplot(aes(parameter, logodds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) + 
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(breaks = c(-4, -2, 0, 2)) +
  xlab(NULL) +
  ylab(&quot;Log-odds difference from w1118&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank()) </code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="measuring-p2" class="section level2">
<h2>Measuring P2</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Fit the model.</p>
<p>Parameter estimates for the fixed effects in the model are displayed on the log-odds scale.</p>
<pre class="r"><code># Filter so that only trials where the SD male mated with the female second are included

sperm_comp_data_P2 &lt;- 
  sperm_comp_data %&gt;%
  filter(Mating.Order == &quot;second&quot;) 

# fit the model

sperm_comp_model_P2 &lt;- brm(Treatment.progeny | trials(total_offspring) ~ SD + Block + (1|Rearing_vial) + (1|ID),
                           data = sperm_comp_data_P2, 
                           family = binomial,
                           prior = c(prior(normal(0, 5), class = Intercept), 
                                     prior(normal(0, 3), class = b)),
                           iter = 8000,
                           warmup = 2000,
                           chains = 4,
                           cores = 4,
                           seed = 2,
                           control = list(adapt_delta = 0.999, max_treedepth = 15),
                           file = &quot;fits/sperm_comp_model_P2&quot;)

# display model results

fixef(sperm_comp_model_P2) %&gt;%
  pander(emphasize.strong.rows = c(2, 4))</code></pre>
<table style="width:96%;">
<colgroup>
<col width="22%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">3.828</td>
<td align="center">0.4994</td>
<td align="center">2.875</td>
<td align="center">4.84</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center"><strong>1.498</strong></td>
<td align="center"><strong>0.6331</strong></td>
<td align="center"><strong>0.289</strong></td>
<td align="center"><strong>2.764</strong></td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">-0.1273</td>
<td align="center">0.5419</td>
<td align="center">-1.201</td>
<td align="center">0.924</td>
</tr>
<tr class="even">
<td align="center"><strong>SD5</strong></td>
<td align="center"><strong>-1.247</strong></td>
<td align="center"><strong>0.5699</strong></td>
<td align="center"><strong>-2.378</strong></td>
<td align="center"><strong>-0.1209</strong></td>
</tr>
<tr class="odd">
<td align="center"><strong>Block2</strong></td>
<td align="center">-0.7958</td>
<td align="center">0.4789</td>
<td align="center">-1.727</td>
<td align="center">0.1544</td>
</tr>
<tr class="even">
<td align="center"><strong>Block3</strong></td>
<td align="center">1.101</td>
<td align="center">0.5725</td>
<td align="center">-0.01236</td>
<td align="center">2.23</td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(sperm_comp_model_P2, type = &quot;hist&quot;, nsamples = 11, binwidth = 2) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20P2%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-P2-plot-1">
Past versions of “diagnostic P2 plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-P2-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic P2 plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic P2 plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the P2 model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_P2 &lt;-
  expand.grid(total_offspring = 100, SD = c(&quot;W1118&quot;, &quot;MAD&quot;, &quot;72&quot;, &quot;5&quot;), Block = unique(sperm_comp_data_P2$Block)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:12, sep = &quot;&quot;)) %&gt;% 
  as_tibble()

# Get predicted means

predictions_P2 &lt;- 
  as.data.frame(fitted(sperm_comp_model_P2, newdata = new_posterior_P2, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = percent_focal_offspring, -posterior_draw) %&gt;%
  left_join(new_posterior_P2, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD) %&gt;% 
  summarise(`Estimated % of offspring sired` = median(percent_focal_offspring),
            `2.5%` = quantile(percent_focal_offspring, probs = 0.025),
            `97.5%` = quantile(percent_focal_offspring, probs = 0.975))


# Now get the predicted posterior

posterior_prediction_P2 &lt;- 
  as.data.frame(fitted(sperm_comp_model_P2, newdata = new_posterior_P2, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = percent_focal_offspring, -posterior_draw) %&gt;%
  left_join(new_posterior_P2, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The estimated percentage of offspring sired by <em>SD/+</em> or control males, when they mated second.</p>
<pre class="r"><code>predictions_P2 %&gt;% 
  mutate(`Variant carried` = recode(predictions_P2$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;% 
  pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table style="width:93%;">
<colgroup>
<col width="25%" />
<col width="45%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Estimated % of offspring sired</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">97.9</td>
<td align="center">91.61</td>
<td align="center">99.7</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">99.52</td>
<td align="center">97.61</td>
<td align="center">99.94</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">97.63</td>
<td align="center">90.34</td>
<td align="center">99.65</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">93.16</td>
<td align="center">74.51</td>
<td align="center">98.91</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code># get posterior samples

post_P2 &lt;- posterior_samples(sperm_comp_model_P2)

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the inv_logit_scaled() function converts the posterior draws onto the response scale 

post_diff_P2 &lt;- post_P2 %&gt;% 
  mutate(p_w1118 =  inv_logit_scaled(b_Intercept),
         p_SD_MAD = inv_logit_scaled(b_SDMAD + b_Intercept),
         p_SD_72 = inv_logit_scaled(b_SD72 + b_Intercept),
         p_SD_5 = inv_logit_scaled(b_SD5 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118,
         `SD-5` = p_SD_5 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = `% difference`) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;)) </code></pre>
<p>The mean proportion of offspring sired by each <em>SD</em> variant relative to the <em>w<sup>1118</sup></em> control, when these males mated with <em>LH<sub>m</sub></em> females second. A value of 0.5 means that the proportion of offspring sired was half that of <em>w<sup>1118</sup></em> males.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_P2 %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean proportion of offspring sired relative to w1118`  = mean(`% difference`),
            `2.5%` = quantile(`% difference`, probs = 0.025),
            `97.5%` = quantile(`% difference`, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 3)</code></pre>
<table style="width:92%;">
<colgroup>
<col width="18%" />
<col width="51%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean proportion of offspring sired relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">0.945</td>
<td align="center">0.855</td>
<td align="center">0.997</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">0.997</td>
<td align="center">0.969</td>
<td align="center">1.027</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">1.019</td>
<td align="center">1.002</td>
<td align="center">1.049</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure 1e and f</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel e</p>
<pre class="r"><code>mean_P2_plot &lt;-
  posterior_prediction_P2 %&gt;%
  mutate(SD = recode(posterior_prediction_P2$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;% 
  ggplot(aes(SD, percent_focal_offspring)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip(ylim = c(70, 100)) +
  ylab(&quot;Estimated % of offspring sired when mating second (P2)&quot;) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel f</p>
<pre class="r"><code>P2_logodds &lt;-
  post_P2 %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72,
         `SD-5` = b_SD5) %&gt;% 
  gather(key = parameter, value = logodds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))) %&gt;%
  
  ggplot(aes(parameter, logodds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) + 
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(breaks = c(-3, -2, -1, 0, 1, 2, 3)) +
  xlab(NULL) +
  ylab(&quot;Log-odds difference from w1118&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank()) </code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="are-females-that-mate-first-with-a-male-harbouring-sd-more-likely-to-remate" class="section level2">
<h2>Are females that mate first with a male harbouring <em>SD</em> more likely to remate?</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Fit the model - here we model remating as a binary YES/NO response.</p>
<pre class="r"><code>remating_model &lt;- 
  brm(Remating ~ SD + Block + (1|Rearing_vial),
      data = remating_data, 
      family = bernoulli,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      iter = 8000,
      warmup = 2000,
      chains = 4,
      cores = 4,
      seed = 2,
      control = list(adapt_delta = 0.99),
      file = &quot;fits/remating_model&quot;)

# display model results

fixef(remating_model) %&gt;%
  pander(emphasize.strong.rows = 4)</code></pre>
<table style="width:90%;">
<colgroup>
<col width="22%" />
<col width="16%" />
<col width="18%" />
<col width="16%" />
<col width="16%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">-0.8325</td>
<td align="center">0.4563</td>
<td align="center">-1.737</td>
<td align="center">0.04211</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">0.5413</td>
<td align="center">0.4877</td>
<td align="center">-0.419</td>
<td align="center">1.511</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">0.5333</td>
<td align="center">0.479</td>
<td align="center">-0.3908</td>
<td align="center">1.493</td>
</tr>
<tr class="even">
<td align="center"><strong>SD5</strong></td>
<td align="center"><strong>1.968</strong></td>
<td align="center"><strong>0.4963</strong></td>
<td align="center"><strong>1.027</strong></td>
<td align="center"><strong>2.978</strong></td>
</tr>
<tr class="odd">
<td align="center"><strong>Block2</strong></td>
<td align="center">-0.8197</td>
<td align="center">0.4633</td>
<td align="center">-1.733</td>
<td align="center">0.08466</td>
</tr>
<tr class="even">
<td align="center"><strong>Block3</strong></td>
<td align="center">0.7758</td>
<td align="center">0.4474</td>
<td align="center">-0.09273</td>
<td align="center">1.672</td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The outcome variable in this model is binary, hence why only 0 and 1 values appear. The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(remating_model, type = &quot;hist&quot;, nsamples = 11, binwidth = 0.5) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20remating%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-remating-plot-1">
Past versions of “diagnostic remating plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-remating-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic remating plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic remating plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_remating &lt;-
  expand.grid(SD = unique(remating_data$SD), Block = unique(remating_data$Block)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:12, sep = &quot;&quot;)) %&gt;% 
  as_tibble()

# Get predicted means

predictions_remating &lt;- 
  as.data.frame(fitted(remating_model, newdata = new_posterior_remating, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = prop_females_remating, -posterior_draw) %&gt;%
  left_join(new_posterior_remating, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  mutate(percent_females_remating = prop_females_remating * 100) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD, Block) %&gt;% 
  summarise(`Estimated % of mates remating` = median(percent_females_remating),
            `2.5%` = quantile(percent_females_remating, probs = 0.025),
            `97.5%` = quantile(percent_females_remating, probs = 0.975))


# Now get the predicted posterior

posterior_prediction_remating &lt;- 
  as.data.frame(fitted(remating_model, newdata = new_posterior_remating, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = prop_females_remating, -posterior_draw) %&gt;%
  left_join(new_posterior_remating, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  mutate(percent_females_remating = prop_females_remating * 100) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The estimated percentage of females that remated that were first mated to a <em>SD/+</em> or control male.</p>
<pre class="r"><code>predictions_remating %&gt;% 
  mutate(`Variant carried` = SD) %&gt;% 
  mutate(`Variant carried` = recode(SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;% 
  ungroup() %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;%
  pander(split.cell = 40, split.table = Inf)</code></pre>
<table>
<colgroup>
<col width="24%" />
<col width="10%" />
<col width="43%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Block</th>
<th align="center">Estimated % of mates remating</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">1</td>
<td align="center">30.36</td>
<td align="center">14.97</td>
<td align="center">51.05</td>
</tr>
<tr class="even">
<td align="center">W1118</td>
<td align="center">2</td>
<td align="center">16.22</td>
<td align="center">7.473</td>
<td align="center">29.98</td>
</tr>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">3</td>
<td align="center">48.62</td>
<td align="center">30.85</td>
<td align="center">66.63</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">1</td>
<td align="center">42.93</td>
<td align="center">22.99</td>
<td align="center">65.09</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">2</td>
<td align="center">24.9</td>
<td align="center">11.75</td>
<td align="center">43.87</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">3</td>
<td align="center">61.8</td>
<td align="center">41.97</td>
<td align="center">78.74</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">1</td>
<td align="center">42.54</td>
<td align="center">23.09</td>
<td align="center">64.14</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">2</td>
<td align="center">24.72</td>
<td align="center">11.72</td>
<td align="center">43.59</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">3</td>
<td align="center">61.53</td>
<td align="center">41.95</td>
<td align="center">78.84</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">1</td>
<td align="center">75.46</td>
<td align="center">55.54</td>
<td align="center">89.24</td>
</tr>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">2</td>
<td align="center">57.68</td>
<td align="center">39.09</td>
<td align="center">75.63</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">3</td>
<td align="center">86.97</td>
<td align="center">74.02</td>
<td align="center">94.58</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_r &lt;- posterior_samples(remating_model)

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the inv_logit_scaled() function converts the posterior draws onto the response scale 

post_diff_r &lt;- post_r %&gt;% 
  mutate(p_w1118 =  inv_logit_scaled(b_Intercept),
         p_SD_MAD = inv_logit_scaled(b_SDMAD + b_Intercept),
         p_SD_72 = inv_logit_scaled(b_SD72 + b_Intercept),
         p_SD_5 = inv_logit_scaled(b_SD5 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118,
         `SD-5` = p_SD_5 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = `% difference`) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))</code></pre>
<p>The mean proportion of mates remating with <em>LH<sub>m</sub><sup>UBI</sup></em> males for <em>SD/+</em> males relative to the <em>w<sup>1118</sup></em> control males. A value of 2 means that remating was twice as likely when females initially mated with a <em>SD/+</em> male compared with a <em>w<sup>1118</sup></em> male.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_r %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean proportion of mates remating relative to w1118`  = mean(`% difference`),
            `2.5%` = quantile(`% difference`, probs = 0.025),
            `97.5%` = quantile(`% difference`, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table style="width:89%;">
<colgroup>
<col width="18%" />
<col width="50%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean proportion of mates remating relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">2.61</td>
<td align="center">1.52</td>
<td align="center">4.66</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">1.47</td>
<td align="center">0.79</td>
<td align="center">2.65</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">1.48</td>
<td align="center">0.77</td>
<td align="center">2.65</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure 1g and h</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel g</p>
<pre class="r"><code>mean_remating_plot &lt;-
  posterior_prediction_remating %&gt;%
  mutate(SD = recode(posterior_prediction_remating$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;%
  filter(Block == &quot;1&quot;) %&gt;% # here we select data from the first block to make the plot look nicer - this doesn&#39;t affect the differences between each SD variant and the control, it just shrinks the uncertainty that occurs because of differences between blocks
  ggplot(aes(SD, percent_females_remating)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  ylab(&quot;Estimated % of females that mated twice&quot;) +
  xlab(&quot;Male variant mated to first&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel h</p>
<pre class="r"><code>Remating_logodds &lt;-
  post_r %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72,
         `SD-5` = b_SD5) %&gt;% 
  gather(key = parameter, value = logodds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;)) %&gt;%
  as_tibble() %&gt;%
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))) %&gt;%
  
  ggplot(aes(parameter, logodds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(&quot;Log-odds difference from w1118&quot;,
                     breaks = c(-1, 0, 1, 2, 3)) +
  xlab(NULL) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank()) </code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="remating-latency" class="section level2">
<h2>Remating latency</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we take a more in depth look at remating by looking at remating latency within the three hour remating period.</p>
<p>Fit the model - this time using remating latency as the response variable and specifying a weibull distribution (a time-to-event model).</p>
<p>Fixed effects are shown on the odds scale <strong>(the weibull model uses a log-link function rather than a logit-link function)</strong></p>
<pre class="r"><code># Fit the model with right censoring

remating_survival_model_censored &lt;- 
  brm(Latency_2 | cens(Censored) ~ SD + Block + (1|Rearing_vial),
      data = remating_data,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      family = weibull, inits = 0,
      cores = 4, chains = 4, iter = 8000, warmup = 2000,
      control = list(adapt_delta = 0.95, max_treedepth = 10),
      seed = 1, file = &quot;fits/remating_survival_model_censored&quot;)

# display model results

fixef(remating_survival_model_censored) %&gt;%
  pander(emphasize.strong.rows = 4:5)</code></pre>
<table style="width:99%;">
<colgroup>
<col width="22%" />
<col width="19%" />
<col width="18%" />
<col width="19%" />
<col width="19%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">8.846</td>
<td align="center">0.3019</td>
<td align="center">8.265</td>
<td align="center">9.455</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">-0.215</td>
<td align="center">0.3167</td>
<td align="center">-0.8382</td>
<td align="center">0.4079</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">-0.1452</td>
<td align="center">0.3137</td>
<td align="center">-0.7689</td>
<td align="center">0.4681</td>
</tr>
<tr class="even">
<td align="center"><strong>SD5</strong></td>
<td align="center"><strong>-0.6866</strong></td>
<td align="center"><strong>0.2866</strong></td>
<td align="center"><strong>-1.263</strong></td>
<td align="center"><strong>-0.1422</strong></td>
</tr>
<tr class="odd">
<td align="center"><strong>Block2</strong></td>
<td align="center"><strong>0.645</strong></td>
<td align="center"><strong>0.2874</strong></td>
<td align="center"><strong>0.08167</strong></td>
<td align="center"><strong>1.217</strong></td>
</tr>
<tr class="even">
<td align="center"><strong>Block3</strong></td>
<td align="center">-0.09898</td>
<td align="center">0.2633</td>
<td align="center">-0.6337</td>
<td align="center">0.4039</td>
</tr>
</tbody>
</table>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(remating_survival_model_censored, type = &quot;hist&quot;, nsamples = 11, binwidth = 600) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/diagnostic%20survival%20plot-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-diagnostic-survival-plot-1">
Past versions of “diagnostic survival plot-1.png”
</button>
</p>
<div id="fig-no-spaces-diagnostic-survival-plot-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/diagnostic survival plot-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/diagnostic survival plot-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_remating_latency &lt;-
  expand.grid(SD = unique(remating_data$SD), Block = unique(remating_data$Block)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:12, sep = &quot;&quot;)) %&gt;% 
  as_tibble()
  

# Get predicted means

predictions_remating_latency &lt;- 
  as.data.frame(fitted(remating_survival_model_censored, newdata = new_posterior_remating_latency, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = avg_remating_latency, -posterior_draw) %&gt;%
  left_join(new_posterior_remating_latency, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD, Block) %&gt;% 
  summarise(`Estimated time to remating (mins)` = median(avg_remating_latency) / 60,
            `2.5%` = quantile(avg_remating_latency, probs = 0.025) / 60,
            `97.5%` = quantile(avg_remating_latency, probs = 0.975) / 60)


# Now get the predicted posterior

posterior_prediction_remating_latency &lt;- 
  as.data.frame(fitted(remating_survival_model_censored, newdata = new_posterior_remating_latency, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = avg_remating_latency, -posterior_draw) %&gt;%
  left_join(new_posterior_remating_latency, by = &quot;id&quot;) %&gt;%
  mutate(avg_remating_latency = avg_remating_latency / 60) %&gt;% 
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p><span class="math inline">\(~\)</span></p>
<p>The estimated time taken for females to remate, when provided with an opportunity, four days after initially mating with a <em>SD/+</em> or control male.</p>
<pre class="r"><code>  predictions_remating_latency %&gt;%
  #mutate(`Variant carried` = SD) %&gt;% 
  mutate(`Variant carried` = fct_recode(SD, &quot;SD-Mad&quot; = &quot;MAD&quot;, &quot;SD-72&quot; = &quot;72&quot;, &quot;SD-5&quot; = &quot;5&quot;)) %&gt;% 
  ungroup() %&gt;% 
  select(`Variant carried`, Block, `Estimated time to remating (mins)`, `2.5%`, `97.5%`) %&gt;%
  pander(split.cell = 40, split.table = Inf, round = 1)</code></pre>
<table>
<colgroup>
<col width="23%" />
<col width="10%" />
<col width="46%" />
<col width="10%" />
<col width="10%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Block</th>
<th align="center">Estimated time to remating (mins)</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">1</td>
<td align="center">115</td>
<td align="center">64.8</td>
<td align="center">212.8</td>
</tr>
<tr class="even">
<td align="center">W1118</td>
<td align="center">2</td>
<td align="center">218.3</td>
<td align="center">127.4</td>
<td align="center">402.6</td>
</tr>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">3</td>
<td align="center">104.6</td>
<td align="center">65.5</td>
<td align="center">169.7</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">1</td>
<td align="center">92.4</td>
<td align="center">53.6</td>
<td align="center">172.1</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">2</td>
<td align="center">176.1</td>
<td align="center">100.4</td>
<td align="center">332.5</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">3</td>
<td align="center">84.3</td>
<td align="center">53.2</td>
<td align="center">136.9</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">1</td>
<td align="center">99.4</td>
<td align="center">57.4</td>
<td align="center">182.4</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">2</td>
<td align="center">188.9</td>
<td align="center">114.9</td>
<td align="center">332.9</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">3</td>
<td align="center">90.5</td>
<td align="center">55.2</td>
<td align="center">149.7</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">1</td>
<td align="center">58.1</td>
<td align="center">36.6</td>
<td align="center">94.6</td>
</tr>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">2</td>
<td align="center">110.6</td>
<td align="center">71.2</td>
<td align="center">176</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">3</td>
<td align="center">52.9</td>
<td align="center">33.6</td>
<td align="center">81.9</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_l &lt;- posterior_samples(remating_survival_model_censored) %&gt;% 
  as_tibble()

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the inv_logit_scaled() function converts the posterior draws onto the response scale 

post_diff_l &lt;- post_l %&gt;% 
  mutate(p_w1118 =  exp(b_Intercept),
         p_SD_MAD = exp(b_SDMAD + b_Intercept),
         p_SD_72 = exp(b_SD72 + b_Intercept),
         p_SD_5 = exp(b_SD5 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118,
         `SD-5` = p_SD_5 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = proportion) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))</code></pre>
<p>The mean remating latency of females that initially mated with <em>SD/+</em> males, relative to the <em>w<sup>1118</sup></em> control males. A value of 0.5 means that females remated in half the time of females that initially mated with a <em>w<sup>1118</sup></em> male.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_l %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean remating latency relative to w1118`  = mean(proportion),
            `2.5%` = quantile(proportion, probs = 0.025),
            `97.5%` = quantile(proportion, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 2)</code></pre>
<table style="width:97%;">
<colgroup>
<col width="18%" />
<col width="58%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean remating latency relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">0.52</td>
<td align="center">0.28</td>
<td align="center">0.88</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">0.91</td>
<td align="center">0.47</td>
<td align="center">1.6</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">0.85</td>
<td align="center">0.43</td>
<td align="center">1.5</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure S5</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel a</p>
<pre class="r"><code>mean_latency_plot &lt;-
  posterior_prediction_remating_latency %&gt;%
  mutate(SD = recode(posterior_prediction_remating_latency$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;%
  filter(Block == &quot;1&quot;) %&gt;% 
  ggplot(aes(SD, avg_remating_latency)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip(ylim = c(0, 250)) +
  ylab(&quot;Estimated time to remating\n (minutes)&quot;) +
  xlab(&quot;Male variant mated to first&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel b</p>
<pre class="r"><code>latency_odds &lt;- 
  post_l %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72,
         `SD-5` = b_SD5) %&gt;% 
  gather(key = parameter, value = logodds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))) %&gt;%
  
  
  ggplot(aes(parameter, logodds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(&quot;Odds difference from w1118&quot;,
                     breaks = c(-2, -1, 0, 1)) +
  xlab(&quot;Male variant mated to first&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())   </code></pre>
<p>Combine panels into a single plot</p>
<pre class="r"><code>mean_latency_plot /
(latency_odds) + plot_annotation(tag_levels = &quot;a&quot;)</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-45-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-45-1">
Past versions of unnamed-chunk-45-1.png
</button>
</p>
<div id="fig-unnamed-chunk-45-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-45-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S5:</strong> The effect that <em>SD/+</em> males have on female remating latency and how this compares to <em>w<sup>1118</sup></em> control males. Panel <strong>a</strong> shows the estimated remating latency for females exposed to <em>LH<sub>m</sub><sup>UBI</sup></em> males over a three-hour period, that had mated with a <em>SD/+</em> or control male four days earlier. Panel <strong>b</strong> shows effect sizes on the odds scale for the <em>SD</em> variants. Negative values indicate that the mates of <em>SD/+</em> males remated faster than the mates of <em>w<sup>1118</sup></em> control males. Black points indicate the estimated mean with associated 66 and 95% uncertainty intervals, while coloured area shows the posterior distribution.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="copulation-duration-in-the-first-mating-role" class="section level2">
<h2>Copulation duration in the first mating role</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we look at copulation duration - an indicator of male investment in mating.</p>
<p>Fit the model using copulation duration as the response variable and specifying a weibull distribution (a time-to-event model).</p>
<p>Fixed effects are shown on the odds scale (<strong>the weibull model uses a log-link function rather than a logit-link function</strong>)</p>
<pre class="r"><code># Filter so that only trials where the SD male mated with the female second are included

copulation_data_1 &lt;- 
  sperm_comp_data %&gt;%
  filter(Mating.Order == &quot;first&quot;) 

copulation_duration_survival_model_1 &lt;- 
  brm(Duration_1 ~ SD + Block + (1|Rearing_vial),
      data = copulation_data_1,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      family = weibull, inits = 0,
      cores = 4, chains = 4, iter = 8000, warmup = 2000,
      control = list(adapt_delta = 0.99, max_treedepth = 10),
      seed = 1, file = &quot;fits/copulation_duration_survival_model_1&quot;)

# display model results

fixef(copulation_duration_survival_model_1) %&gt;%
  pander()</code></pre>
<table style="width:85%;">
<colgroup>
<col width="22%" />
<col width="15%" />
<col width="16%" />
<col width="15%" />
<col width="15%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">7.084</td>
<td align="center">0.08901</td>
<td align="center">6.914</td>
<td align="center">7.266</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">0.03873</td>
<td align="center">0.09641</td>
<td align="center">-0.1529</td>
<td align="center">0.2254</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center">0.1313</td>
<td align="center">0.09418</td>
<td align="center">-0.05449</td>
<td align="center">0.316</td>
</tr>
<tr class="even">
<td align="center"><strong>SD5</strong></td>
<td align="center">0.1076</td>
<td align="center">0.08616</td>
<td align="center">-0.06409</td>
<td align="center">0.2748</td>
</tr>
<tr class="odd">
<td align="center"><strong>Block2</strong></td>
<td align="center">-0.09988</td>
<td align="center">0.08873</td>
<td align="center">-0.2755</td>
<td align="center">0.07346</td>
</tr>
<tr class="even">
<td align="center"><strong>Block3</strong></td>
<td align="center">0.008328</td>
<td align="center">0.08041</td>
<td align="center">-0.154</td>
<td align="center">0.1611</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(copulation_duration_survival_model_1, type = &quot;hist&quot;, nsamples = 11, binwidth = 200) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-47-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-47-1">
Past versions of unnamed-chunk-47-1.png
</button>
</p>
<div id="fig-unnamed-chunk-47-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-47-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_copulation_duration_1 &lt;-
  expand.grid(SD = unique(copulation_data_1$SD), Block = unique(copulation_data_1$Block)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:12, sep = &quot;&quot;)) %&gt;% 
  as_tibble()

# Get predicted means

predictions_copulation_duration_1 &lt;- 
  as.data.frame(fitted(copulation_duration_survival_model_1, newdata = new_posterior_copulation_duration_1, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_copulation_duration, -posterior_draw) %&gt;%
  mutate(mean_copulation_duration = mean_copulation_duration / 60) %&gt;%
  left_join(new_posterior_copulation_duration_1, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD) %&gt;% 
  summarise(`Estimated copulation duration (mins)` = median(mean_copulation_duration),
            `2.5%` = quantile(mean_copulation_duration, probs = 0.025),
            `97.5%` = quantile(mean_copulation_duration, probs = 0.975))


# Now get the predicted posterior

posterior_prediction_copulation_duration_1 &lt;- 
  as.data.frame(fitted(copulation_duration_survival_model_1, newdata = new_posterior_copulation_duration_1, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_copulation_duration, -posterior_draw) %&gt;%
  mutate(mean_copulation_duration = mean_copulation_duration / 60) %&gt;% # if we want to express in secs then delete this
  left_join(new_posterior_copulation_duration_1, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p>The mean copulation duration when a <em>SD/+</em> male mated with a <em>LH<sub>m</sub></em> female in the first mating role.</p>
<pre class="r"><code>predictions_copulation_duration_1 %&gt;%
  mutate(`Variant carried` = recode(predictions_copulation_duration_1$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;%
  pander(split.cell = 40, split.table = Inf, round = 1)</code></pre>
<table>
<colgroup>
<col width="25%" />
<col width="54%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Estimated copulation duration (mins)</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">19.3</td>
<td align="center">16</td>
<td align="center">23.2</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">20.1</td>
<td align="center">16.4</td>
<td align="center">24</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">22</td>
<td align="center">18.3</td>
<td align="center">26.5</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">21.5</td>
<td align="center">18</td>
<td align="center">25.3</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_cd_1 &lt;- posterior_samples(copulation_duration_survival_model_1) %&gt;% 
  as_tibble()

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the exp() function converts the posterior draws onto the response scale 

post_diff_cd_1 &lt;- post_cd_1 %&gt;% 
  mutate(p_w1118 =  exp(b_Intercept),
         p_SD_MAD = exp(b_SDMAD + b_Intercept),
         p_SD_72 = exp(b_SD72 + b_Intercept),
         p_SD_5 = exp(b_SD5 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118,
         `SD-5` = p_SD_5 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = proportion) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))</code></pre>
<p>The mean copulation duration for <em>SD/+</em> males that mated with a <em>LH<sub>m</sub></em> female in the second mating role, relative to the <em>w<sup>1118</sup></em> control males. A value of 0.5 means that males mated for half the time of <em>w<sup>1118</sup></em> males.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_cd_1 %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean copulation duration relative to w1118`  = mean(proportion),
            `2.5%` = quantile(proportion, probs = 0.025),
            `97.5%` = quantile(proportion, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 3)</code></pre>
<table style="width:94%;">
<colgroup>
<col width="18%" />
<col width="54%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean copulation duration relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">1.118</td>
<td align="center">0.936</td>
<td align="center">1.314</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">1.145</td>
<td align="center">0.946</td>
<td align="center">1.371</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">1.044</td>
<td align="center">0.857</td>
<td align="center">1.253</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure S6</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel a</p>
<pre class="r"><code>mean_copulation_duration_plot_1 &lt;-
  posterior_prediction_copulation_duration_1 %&gt;%
  mutate(SD = recode(posterior_prediction_copulation_duration_1$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;%
  ggplot(aes(SD, mean_copulation_duration)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip(ylim = c(10, 35)) +
  ylab(&quot;Estimated copulation duration in\nfirst mating role (minutes)&quot;) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel b</p>
<pre class="r"><code>copulation_duration_odds_1 &lt;- 
  post_cd_1 %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72,
         `SD-5` = b_SD5) %&gt;% 
  gather(key = parameter, value = odds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))) %&gt;%
  
  
  ggplot(aes(parameter, odds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(&quot;Odds difference from w1118&quot;,
                     breaks = c(-0.25, 0, 0.25, 0.5)) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())  </code></pre>
<p>Combine panels into a single plot</p>
<pre class="r"><code>mean_copulation_duration_plot_1 /
(copulation_duration_odds_1) + plot_annotation(tag_levels = &quot;a&quot;)</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-54-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-54-1">
Past versions of unnamed-chunk-54-1.png
</button>
</p>
<div id="fig-unnamed-chunk-54-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-54-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S6</strong>: No difference in the duration of mating between a <em>SD/+</em> male and a <em>LH<sub>m</sub></em> female, compared to <em>w<sup>1118</sup></em> control males, when the <em>SD/+</em> (or control) male mated first. Panel <strong>a</strong> shows the estimated copulation duration for <em>SD/+</em> and control males. Panel <strong>b</strong> shows effect sizes on the odds scale for the <em>SD</em>-variants. Positive values indicate that <em>SD/+</em> males mated for longer than <em>w<sup>1118</sup></em> control males. Black points indicate the estimated mean with associated 66 and 95% uncertainty intervals, while coloured area shows the posterior distribution.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="copulation-duration-in-the-second-mating-role" class="section level2">
<h2>Copulation duration in the second mating role</h2>
<p><span class="math inline">\(~\)</span></p>
<p>Here we look at copulation duration - an indicator of male investment in mating.</p>
<p>Fit the model using copulation duration as the response variable and specifying a weibull distribution (a time-to-event model).</p>
<p>Fixed effects are shown on the odds scale (<strong>the weibull model uses a log-link function rather than a logit-link function</strong>)</p>
<pre class="r"><code># Filter so that only trials where the SD male mated with the female second are included

copulation_data_2 &lt;- 
  sperm_comp_data %&gt;%
  filter(Mating.Order == &quot;second&quot;) 

copulation_duration_survival_model_2 &lt;- 
  brm(Duration_2 ~ SD + Block + (1|Rearing_vial),
      data = copulation_data_2,
      prior = c(prior(normal(0, 5), class = Intercept), 
                prior(normal(0, 3), class = b)),
      family = weibull, inits = 0,
      cores = 4, chains = 4, iter = 8000, warmup = 2000,
      control = list(adapt_delta = 0.99, max_treedepth = 10),
      seed = 1, file = &quot;fits/copulation_duration_survival_model_2&quot;)

# display model results

fixef(copulation_duration_survival_model_2) %&gt;%
  pander(emphasize.strong.rows = 3)</code></pre>
<table>
<colgroup>
<col width="22%" />
<col width="18%" />
<col width="18%" />
<col width="20%" />
<col width="20%" />
</colgroup>
<thead>
<tr class="header">
<th align="center"> </th>
<th align="center">Estimate</th>
<th align="center">Est.Error</th>
<th align="center">Q2.5</th>
<th align="center">Q97.5</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center"><strong>Intercept</strong></td>
<td align="center">7.047</td>
<td align="center">0.1219</td>
<td align="center">6.811</td>
<td align="center">7.29</td>
</tr>
<tr class="even">
<td align="center"><strong>SDMAD</strong></td>
<td align="center">0.1024</td>
<td align="center">0.1436</td>
<td align="center">-0.1853</td>
<td align="center">0.3842</td>
</tr>
<tr class="odd">
<td align="center"><strong>SD72</strong></td>
<td align="center"><strong>0.2894</strong></td>
<td align="center"><strong>0.1426</strong></td>
<td align="center"><strong>0.005337</strong></td>
<td align="center"><strong>0.5686</strong></td>
</tr>
<tr class="even">
<td align="center"><strong>SD5</strong></td>
<td align="center">0.05127</td>
<td align="center">0.1415</td>
<td align="center">-0.2255</td>
<td align="center">0.3313</td>
</tr>
<tr class="odd">
<td align="center"><strong>Block2</strong></td>
<td align="center">-0.05703</td>
<td align="center">0.1236</td>
<td align="center">-0.2994</td>
<td align="center">0.1879</td>
</tr>
<tr class="even">
<td align="center"><strong>Block3</strong></td>
<td align="center">-0.06026</td>
<td align="center">0.1265</td>
<td align="center">-0.3065</td>
<td align="center">0.193</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Evaluate model fit using a posterior predictive check</strong></p>
<p>A posterior predictive check works by using the model to predict the dataset upon which it was trained. Here, 11 draws are shown from the posterior predictive distribution (light blue), as well as the real dataset (dark blue). The datasets predicted from the model are reasonably similar to the real dataset, indicating that the model is a reasonably good fit.</p>
<pre class="r"><code>pp_check(copulation_duration_survival_model_2, type = &quot;hist&quot;, nsamples = 11, binwidth = 200) +
  theme_minimal() +
  theme(panel.background = element_blank())</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-56-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-56-1">
Past versions of unnamed-chunk-56-1.png
</button>
</p>
<div id="fig-unnamed-chunk-56-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-56-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>Get predictions from the model</p>
<pre class="r"><code># Define new data for prediction with posteriors

new_posterior_copulation_duration_2 &lt;-
  expand.grid(SD = unique(copulation_data_2$SD), Block = unique(copulation_data_2$Block)) %&gt;% 
  mutate(id = paste(&quot;V&quot;, 1:12, sep = &quot;&quot;)) %&gt;% 
  as_tibble()

# Get predicted means

predictions_copulation_duration_2 &lt;- 
  as.data.frame(fitted(copulation_duration_survival_model_2, newdata = new_posterior_copulation_duration_2, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_copulation_duration, -posterior_draw) %&gt;%
  mutate(mean_copulation_duration = mean_copulation_duration / 60) %&gt;%
  left_join(new_posterior_copulation_duration_2, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble() %&gt;% 
  group_by(SD) %&gt;% 
  summarise(`Estimated copulation duration (mins)` = median(mean_copulation_duration),
            `2.5%` = quantile(mean_copulation_duration, probs = 0.025),
            `97.5%` = quantile(mean_copulation_duration, probs = 0.975))


# Now get the predicted posterior

posterior_prediction_copulation_duration_2 &lt;- 
  as.data.frame(fitted(copulation_duration_survival_model_2, newdata = new_posterior_copulation_duration_2, re_formula = NA, summary = FALSE)) %&gt;% # 12 cols, 8000ish rows
  mutate(posterior_draw = 1:n()) %&gt;%
  gather(key = id, value = mean_copulation_duration, -posterior_draw) %&gt;%
  mutate(mean_copulation_duration = mean_copulation_duration / 60) %&gt;% # if we want to express in secs then delete this
  left_join(new_posterior_copulation_duration_2, by = &quot;id&quot;) %&gt;%
  select(-id) %&gt;% 
  as_tibble()</code></pre>
<p>The mean copulation duration when a <em>SD/+</em> male mated with a <em>LH<sub>m</sub></em> female in the second mating role.</p>
<pre class="r"><code>predictions_copulation_duration_2 %&gt;%
  mutate(`Variant carried` = recode(predictions_copulation_duration_2$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;% 
  select(`Variant carried`, everything(), -SD) %&gt;%
  pander(split.cell = 40, split.table = Inf, round = 1)</code></pre>
<table>
<colgroup>
<col width="25%" />
<col width="54%" />
<col width="9%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">Variant carried</th>
<th align="center">Estimated copulation duration (mins)</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">W1118</td>
<td align="center">18.4</td>
<td align="center">14.5</td>
<td align="center">23.7</td>
</tr>
<tr class="even">
<td align="center">SD-Mad</td>
<td align="center">20.4</td>
<td align="center">15.8</td>
<td align="center">26.4</td>
</tr>
<tr class="odd">
<td align="center">SD-72</td>
<td align="center">24.6</td>
<td align="center">19.1</td>
<td align="center">31.6</td>
</tr>
<tr class="even">
<td align="center">SD-5</td>
<td align="center">19.4</td>
<td align="center">15.1</td>
<td align="center">25.1</td>
</tr>
</tbody>
</table>
<pre class="r"><code># get posterior samples

post_cd_2 &lt;- posterior_samples(copulation_duration_survival_model_2) %&gt;% 
  as_tibble()

# now find the differences between the reference level (w1118) and the remaining levels (the SD lines)

# the exp() function converts the posterior draws onto the response scale 

post_diff_cd_2 &lt;- post_cd_2 %&gt;% 
  mutate(p_w1118 =  exp(b_Intercept),
         p_SD_MAD = exp(b_SDMAD + b_Intercept),
         p_SD_72 = exp(b_SD72 + b_Intercept),
         p_SD_5 = exp(b_SD5 + b_Intercept),
         `SD-Mad` = p_SD_MAD / p_w1118,
         `SD-72` = p_SD_72 / p_w1118,
         `SD-5` = p_SD_5 / p_w1118) %&gt;% 
  gather(key = `difference comparison`, value = proportion) %&gt;% 
  filter(`difference comparison` == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))</code></pre>
<p>The mean copulation duration for <em>SD/+</em> males that mated with a <em>LH<sub>m</sub></em> female in the second mating role, relative to the <em>w<sup>1118</sup></em> control males. A value of 0.5 means that males mated for half the time of <em>w<sup>1118</sup></em> males.</p>
<pre class="r"><code># Create a table summarising the proportions of offspring sired relative to w1118

post_diff_cd_2 %&gt;% 
  group_by(`difference comparison`) %&gt;% 
  summarise(`Mean copulation duration relative to w1118`  = mean(proportion),
            `2.5%` = quantile(proportion, probs = 0.025),
            `97.5%` = quantile(proportion, probs = 0.975)) %&gt;% 
  rename(`SD-variant` = `difference comparison`) %&gt;% 
   pander(split.cell = 40, split.table = Inf, round = 3)</code></pre>
<table style="width:94%;">
<colgroup>
<col width="18%" />
<col width="54%" />
<col width="11%" />
<col width="11%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD-variant</th>
<th align="center">Mean copulation duration relative to w1118</th>
<th align="center">2.5%</th>
<th align="center">97.5%</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">1.063</td>
<td align="center">0.795</td>
<td align="center">1.392</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">1.351</td>
<td align="center">1.002</td>
<td align="center">1.764</td>
</tr>
<tr class="odd">
<td align="center">SD-Mad</td>
<td align="center">1.119</td>
<td align="center">0.833</td>
<td align="center">1.47</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Creating Figure S7</strong></p>
<p><span class="math inline">\(~\)</span></p>
<p>Panel a</p>
<pre class="r"><code>mean_copulation_duration_plot_2 &lt;-
  posterior_prediction_copulation_duration_2 %&gt;%
  mutate(SD = recode(posterior_prediction_copulation_duration_2$SD, &quot;MAD&quot; = &quot;SD-Mad&quot;, &quot;72&quot; = &quot;SD-72&quot;, &quot;5&quot; = &quot;SD-5&quot;)) %&gt;%
  ggplot(aes(SD, mean_copulation_duration)) + 
  stat_halfeye(aes(fill = SD), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;W1118&quot; = &quot;#d3f2a3&quot;, &quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip(ylim = c(10, 35)) +
  ylab(&quot;Estimated copulation duration in\nsecond mating role (minutes)&quot;) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())</code></pre>
<p>Panel b</p>
<pre class="r"><code>copulation_duration_odds_2 &lt;- 
  post_cd_2 %&gt;% 
  mutate(`SD-Mad` = b_SDMAD,
         `SD-72` = b_SD72,
         `SD-5` = b_SD5) %&gt;% 
  gather(key = parameter, value = odds) %&gt;% 
  filter(parameter == c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;)) %&gt;%
  as_tibble() %&gt;% 
  mutate(parameter =factor(parameter, levels=c(&quot;SD-Mad&quot;, &quot;SD-72&quot;, &quot;SD-5&quot;))) %&gt;%
  
  
  ggplot(aes(parameter, odds)) + 
  stat_halfeye(aes(fill = parameter), .width = c(0.66, 0.95)) + # width indicates the uncertainty intervals: here we have 66% and 95% intervals
  scale_fill_manual(values = c(&quot;SD-Mad&quot; = &quot;#97e196&quot;, &quot;SD-72&quot; = &quot;#6cc08b&quot;, &quot;SD-5&quot; = &quot;#4c9b82&quot;)) +
  coord_flip() +
  geom_hline(yintercept = 0, linetype = 2) +
  scale_y_continuous(&quot;Odds difference from w1118&quot;,
                     breaks = c(-0.25, 0, 0.25, 0.5)) +
  xlab(&quot;Variant carried by male&quot;) +
  theme_bw() + 
  theme(legend.position = &quot;none&quot;,
        panel.grid.minor = element_blank())  </code></pre>
<p>Combine panels into a single plot</p>
<pre class="r"><code>mean_copulation_duration_plot_2 /
(copulation_duration_odds_2) + plot_annotation(tag_levels = &quot;a&quot;)</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-63-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-63-1">
Past versions of unnamed-chunk-63-1.png
</button>
</p>
<div id="fig-unnamed-chunk-63-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-63-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S7</strong>: mating duration between a <em>SD/+</em> male and a <em>LH<sub>m</sub></em> female, compared to <em>w<sup>1118</sup></em> control males, when the <em>SD/+</em> (or control) male mated second. Panel <strong>a</strong> shows the estimated copulation duration for <em>SD/+</em> and control males. Panel <strong>b</strong> shows effect sizes on the odds scale for the <em>SD</em> variants. Positive values indicate that <em>SD/+</em> males mated for longer than <em>w<sup>1118</sup></em> control males. Black points indicate the estimated mean with associated 66 and 95% uncertainty intervals, while coloured area shows the posterior distribution.</p>
<p><span class="math inline">\(~\)</span></p>
</div>
</div>
<div id="building-figure-1" class="section level1">
<h1>Building Figure 1</h1>
<pre class="r"><code>mean_mating_success_plot + Mating_success_logodds +
mean_P1_plot + P1_logodds +
mean_P2_plot + P2_logodds + 
mean_remating_plot + Remating_logodds + 
  plot_annotation(tag_levels = &quot;a&quot;, tag_suffix = &quot;  &quot;) +
  plot_layout(widths = c(2, 1))</code></pre>
<p><img src="figure/Empirical_analysis.Rmd/unnamed-chunk-64-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-64-1">
Past versions of unnamed-chunk-64-1.png
</button>
</p>
<div id="fig-unnamed-chunk-64-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/bfbb766ecab13ca6424dc808319c92afb02df293/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-64-1.png" target="_blank">bfbb766</a>
</td>
<td>
tomkeaney
</td>
<td>
2021-06-02
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Empirical_analysis.Rmd/unnamed-chunk-64-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 1</strong>. The effect of <em>SD</em> on male mating success, fertilisation success and female remating propensity. Black points indicate the estimated mean, with associated 66 and 95% uncertainty intervals, while coloured area shows the posterior distribution. Panels a, c, e and g show results on the response scale, while panels b, d, f, and h show log-odds differences between the <em>SD</em> variants and the control allele; 95% uncertainty intervals that do not overlap zero indicate a significant effect.</p>
<p><span class="math inline">\(~\)</span></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] DT_0.18           patchwork_1.1.1   rcartocolor_2.0.0 lubridate_1.7.10 
 [5] kableExtra_1.3.4  pander_0.6.3      tidybayes_2.3.1   brms_2.15.0      
 [9] Rcpp_1.0.6        reshape2_1.4.4    forcats_0.5.1     stringr_1.4.0    
[13] dplyr_1.0.5       purrr_0.3.4       readr_1.4.0       tidyr_1.1.3      
[17] tibble_3.1.1      ggplot2_3.3.3     tidyverse_1.3.1   workflowr_1.6.2  

loaded via a namespace (and not attached):
  [1] readxl_1.3.1         backports_1.2.1      systemfonts_1.0.1   
  [4] plyr_1.8.6           igraph_1.2.6         splines_4.0.5       
  [7] svUnit_1.0.6         crosstalk_1.1.1      rstantools_2.1.1    
 [10] inline_0.3.17        digest_0.6.27        htmltools_0.5.1.1   
 [13] rsconnect_0.8.17     fansi_0.4.2          magrittr_2.0.1      
 [16] modelr_0.1.8         RcppParallel_5.1.2   matrixStats_0.58.0  
 [19] svglite_2.0.0        xts_0.12.1           prettyunits_1.1.1   
 [22] colorspace_2.0-0     rvest_1.0.0          ggdist_2.4.0        
 [25] haven_2.4.0          xfun_0.22            callr_3.7.0         
 [28] crayon_1.4.1         jsonlite_1.7.2       lme4_1.1-26         
 [31] zoo_1.8-9            glue_1.4.2           gtable_0.3.0        
 [34] emmeans_1.5.5-1      webshot_0.5.2        V8_3.4.0            
 [37] distributional_0.2.2 pkgbuild_1.2.0       rstan_2.21.2        
 [40] abind_1.4-5          scales_1.1.1         mvtnorm_1.1-1       
 [43] DBI_1.1.1            miniUI_0.1.1.1       viridisLite_0.4.0   
 [46] xtable_1.8-4         stats4_4.0.5         StanHeaders_2.21.0-7
 [49] htmlwidgets_1.5.3    httr_1.4.2           threejs_0.3.3       
 [52] arrayhelpers_1.1-0   ellipsis_0.3.1       farver_2.1.0        
 [55] pkgconfig_2.0.3      loo_2.4.1            sass_0.3.1          
 [58] dbplyr_2.1.1         utf8_1.2.1           labeling_0.4.2      
 [61] tidyselect_1.1.0     rlang_0.4.10         later_1.1.0.1       
 [64] munsell_0.5.0        cellranger_1.1.0     tools_4.0.5         
 [67] cli_2.4.0            generics_0.1.0       broom_0.7.6         
 [70] ggridges_0.5.3       evaluate_0.14        fastmap_1.1.0       
 [73] yaml_2.2.1           processx_3.5.1       knitr_1.32          
 [76] fs_1.5.0             nlme_3.1-152         whisker_0.4         
 [79] mime_0.10            projpred_2.0.2       xml2_1.3.2          
 [82] compiler_4.0.5       bayesplot_1.8.0      shinythemes_1.2.0   
 [85] rstudioapi_0.13      gamm4_0.2-6          curl_4.3            
 [88] reprex_2.0.0         statmod_1.4.35       bslib_0.2.4         
 [91] stringi_1.5.3        highr_0.9            ps_1.6.0            
 [94] Brobdingnag_1.2-6    lattice_0.20-41      Matrix_1.3-2        
 [97] nloptr_1.2.2.2       markdown_1.1         shinyjs_2.0.0       
[100] vctrs_0.3.7          pillar_1.6.0         lifecycle_1.0.0     
[103] jquerylib_0.1.3      bridgesampling_1.1-2 estimability_1.3    
[106] httpuv_1.5.5         R6_2.5.0             promises_1.2.0.1    
[109] gridExtra_2.3        codetools_0.2-18     boot_1.3-27         
[112] colourpicker_1.1.0   MASS_7.3-53.1        gtools_3.8.2        
[115] assertthat_0.2.1     rprojroot_2.0.2      withr_2.4.2         
[118] shinystan_2.5.0      mgcv_1.8-35          parallel_4.0.5      
[121] hms_1.0.0            grid_4.0.5           coda_0.19-4         
[124] minqa_1.2.4          rmarkdown_2.7        git2r_0.28.0        
[127] shiny_1.6.0          base64enc_0.1-3      dygraphs_1.1.1.6    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
