<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Thomas Keaney, Theresa Jones and Luke Holman" />


<title>Sexual selection and the population genetics of a selfish gene</title>

<script src="site_libs/header-attrs-2.7/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/simplex.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-sm-12 col-md-4 col-lg-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-sm-12 col-md-8 col-lg-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Sexual selection & the population genetics of a selfish gene</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="Supp_methods.html">Supplementary methods</a>
</li>
<li>
  <a href="Empirical_analysis.html">Empirical analysis</a>
</li>
<li>
  <a href="Population_genetic_model.html">Population genetic model</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/tomkeaney/SD_sexual_selection">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Sexual selection and the population genetics of a selfish gene</h1>
<h3 class="subtitle">Population genetic model</h3>
<h4 class="author">Thomas Keaney, Theresa Jones and Luke Holman</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-08-07
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>SD_sexual_selection/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200925code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200925)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200925code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200925)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneySDsexualselectiontree94fc3208e19d5697e4023bfc68702aeb698f5773targetblank94fc320a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/tomkeaney/SD_sexual_selection/tree/94fc3208e19d5697e4023bfc68702aeb698f5773" target="_blank">94fc320</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomtomkeaneySDsexualselectiontree94fc3208e19d5697e4023bfc68702aeb698f5773targetblank94fc320a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/tomkeaney/SD_sexual_selection/tree/94fc3208e19d5697e4023bfc68702aeb698f5773" target="_blank">94fc320</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rapp.history
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.Rapp.history
    Ignored:    simulation_results/.DS_Store

Unstaged changes:
    Modified:   Supplementary_material.Rmd
    Modified:   Supplementary_material.pdf
    Modified:   analysis/Supp_methods.Rmd
    Modified:   simulation_results/output.rds
    Modified:   simulation_results/output1.rds
    Modified:   simulation_results/output10.rds
    Modified:   simulation_results/output11.rds
    Modified:   simulation_results/output12.rds
    Modified:   simulation_results/output13.rds
    Modified:   simulation_results/output14.rds
    Modified:   simulation_results/output15.rds
    Modified:   simulation_results/output16.rds
    Modified:   simulation_results/output17.rds
    Modified:   simulation_results/output18.rds
    Modified:   simulation_results/output19.rds
    Modified:   simulation_results/output2.rds
    Modified:   simulation_results/output20.rds
    Modified:   simulation_results/output21.rds
    Modified:   simulation_results/output22.rds
    Modified:   simulation_results/output23.rds
    Modified:   simulation_results/output24.rds
    Modified:   simulation_results/output25.rds
    Modified:   simulation_results/output26.rds
    Modified:   simulation_results/output27.rds
    Modified:   simulation_results/output28.rds
    Modified:   simulation_results/output29.rds
    Modified:   simulation_results/output3.rds
    Modified:   simulation_results/output30.rds
    Modified:   simulation_results/output31.rds
    Modified:   simulation_results/output32.rds
    Modified:   simulation_results/output33.rds
    Modified:   simulation_results/output4.rds
    Modified:   simulation_results/output5.rds
    Modified:   simulation_results/output6.rds
    Modified:   simulation_results/output7.rds
    Modified:   simulation_results/output8.rds
    Modified:   simulation_results/output9.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/Population_genetic_model.Rmd</code>) and HTML (<code>docs/Population_genetic_model.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/94fc3208e19d5697e4023bfc68702aeb698f5773/analysis/Population_genetic_model.Rmd" target="_blank">94fc320</a>
</td>
<td>
tkeaney
</td>
<td>
2021-08-07
</td>
<td>
fixes in response to reviewer comments
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/e16caf90d34a0ecf444935ef545bc91447457c17/docs/Population_genetic_model.html" target="_blank">e16caf9</a>
</td>
<td>
tkeaney
</td>
<td>
2021-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/51d8aeeb704bba08847036dd9c11780f52cf4139/analysis/Population_genetic_model.Rmd" target="_blank">51d8aee</a>
</td>
<td>
tkeaney
</td>
<td>
2021-06-02
</td>
<td>
more final touches
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/9a9d65072ed3515f10cd5932e282b3befedf81d1/docs/Population_genetic_model.html" target="_blank">9a9d650</a>
</td>
<td>
tkeaney
</td>
<td>
2021-06-02
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/fd7b4780c8164d81ff0975d4c2653a5b81982384/docs/Population_genetic_model.html" target="_blank">fd7b478</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-26
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/2faa81697e2113bfc1abb8dc0bae5307c9038f46/analysis/Population_genetic_model.Rmd" target="_blank">2faa816</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-26
</td>
<td>
final touch pre submission
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/f30258aac3f1b78957f599e4001cf340ea4a4304/docs/Population_genetic_model.html" target="_blank">f30258a</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/877c09b4c442811e7ffc734deea113689bd61bd1/analysis/Population_genetic_model.Rmd" target="_blank">877c09b</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
<td>
final updates before submission
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/e28d4c6ee418b8caa4ad46c3e2b7ef2f67dc9932/docs/Population_genetic_model.html" target="_blank">e28d4c6</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/tomkeaney/SD_sexual_selection/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/Population_genetic_model.html" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/3565002e7456108b162fe417c020d80f199d63d3/analysis/Population_genetic_model.Rmd" target="_blank">3565002</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
<td>
Publish the initial files for myproject
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p><span class="math inline">\(~\)</span></p>
<p>We shall refer to segregation distorting alleles as <strong>SD</strong> and non-distorting wildtype alleles as <strong>W</strong> throughout this analysis. This notation can be found in our code chunks and in the accompanying explanations. For example, we refer to an individual homozygous for the <em>SD</em> allele as <strong>SDSD</strong>, a heterozygous individual as <strong>WSD</strong> and an individual homozygous for the wildtype allele as <strong>WW</strong>.</p>
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
<pre class="r"><code>library(tidyverse) # for purrr, ggplot, dplyr and more 
library(patchwork) # for combining plots
library(rcartocolor) # for cool colours
library(ggpubr) # for combining plots
library(kableExtra) # scrolling tables
library(pander) # other nice tables
library(stickylabeller) # for labelling ggplot facets</code></pre>
</div>
<div id="enter-empirical-data-from-present-study" class="section level1">
<h1>Enter empirical data from present study</h1>
<p><span class="math inline">\(~\)</span></p>
<p>In our empirical investigations, we have estimated the following parameters for the <em>SD-5</em>, <em>SD-72</em> and <em>SD-Mad</em> alleles. We use these to parameterise our model, and to estimate the predicted equilibrium frequency for each <em>SD</em> allele, within the model’s parameter space (see Figure 2 and Figure S6)</p>
<ol style="list-style-type: decimal">
<li><p>k<sub>c</sub> values.</p></li>
<li><p><span class="math inline">\(S_{precop}\)</span>, the mating success of <em>SD/+</em> males (no measurements for <em>SD-5</em>).</p></li>
<li><p><span class="math inline">\(P1_{cost}\)</span> and <span class="math inline">\(P2_{cost}\)</span> for WSD males*.</p></li>
<li><p>The probability of a female remating (albeit in very specific circumstances) after mating with a WSD male (<span class="math inline">\(p_{WSD}\)</span>) or a WW male (<span class="math inline">\(p_{WW}\)</span>).</p></li>
</ol>
<p>*We did not measure the sperm competitive ability of SDSD homozygotes, because <em>SD-5</em> and <em>SD-72</em> are homozygous lethal. <em>SD-Mad</em> is homozygous viable, but a previous study (<a href="https://doi.org/10.1111/jeb.13549">Wong and Holman, 2019</a>) indicates that the fitness of <em>SD-Mad</em> homozygotes is very close to 0.</p>
<pre class="r"><code># k_c values

estimated_k_SD5 &lt;- 0.944 # 94.4% of progeny inherit SD 

estimated_k_SD72 &lt;- 0.909 # 90.9% of progeny inherit SD 

estimated_k_SDMAD &lt;- 0.868 # 86.8% of progeny inherit SD

# The relative mating success for each SD variant

estimated_precop_SD5 &lt;- NA # not measured 

estimated_precop_SD72 &lt;- 1.24
  
estimated_precop_SDMAD &lt;- 1.19

# P1cost and P2cost values, plugged in from Table S7b and Table S8b

P1_SD5 &lt;- 1 - 0.15 

P2_SD5 &lt;- 1 - 0.945 

P1_SD72 &lt;- 1 - 0.43 

P2_SD72 &lt;- 1 - 0.997 

P1_SDMAD &lt;- 1 - 0.39 

P2_SDMAD &lt;- 0 # no cost

# We also include 95% credible intervals for P1 and P2 cost, to use in Figure 2 and Figure S8

lower_P1_SD5 &lt;- 1 - 0.02 

lower_P2_SD5 &lt;- 1 - 0.855

lower_P1_SD72 &lt;- 1 - 0.05 

lower_P2_SD72 &lt;- 1 - 0.969 

lower_P1_SDMAD &lt;-  NA #1 - 0.04 essentially identical to SD-72 so to avoid clutter in our figures we don&#39;t include this value

lower_P2_SDMAD &lt;- 0 # no cost

upper_P1_SD5 &lt;- 1 - 0.6

upper_P2_SD5 &lt;- 1 - 0.997

upper_P1_SD72 &lt;- 0 # no cost

upper_P2_SD72 &lt;- 0 # no cost

upper_P1_SDMAD &lt;- 0 # no cost

upper_P2_SDMAD &lt;- 0 # no cost

# Remating probabilities from Table S9a, using estimates from Block 1

estimated_prob_remating_W = 0.304 

estimated_prob_remating_SD5 = 0.755 

estimated_prob_remating_SD72 = 0.425 

estimated_prob_remating_SDMAD = 0.429

# Create a tibble containing empirical estimates

empirical_estimates &lt;-
  tibble(SD = c(&quot;SD-5&quot;, &quot;SD-72&quot;, &quot;SD-MAD&quot;),
         kc = c(estimated_k_SD5, estimated_k_SD72, estimated_k_SDMAD),
         Precop = c(estimated_precop_SD5, estimated_precop_SD72, estimated_precop_SDMAD),
         P1cost = c(P1_SD5, P1_SD72, P1_SDMAD),
         lower_P1cost = c(lower_P1_SD5, lower_P1_SD72, lower_P1_SDMAD),
         upper_P1cost = c(upper_P1_SD5, upper_P1_SD72, upper_P1_SDMAD),
         P2cost = c(P2_SD5, P2_SD72, P2_SDMAD),
         lower_P2cost = c(lower_P2_SD5, lower_P2_SD72, lower_P2_SDMAD),
         upper_P2cost = c(upper_P2_SD5, upper_P2_SD72, upper_P2_SDMAD),
         remating_prob = c(estimated_prob_remating_SD5, estimated_prob_remating_SD72, estimated_prob_remating_SDMAD))

pander(empirical_estimates, split.cell = 40, split.table = Inf)</code></pre>
<table>
<colgroup>
<col width="7%" />
<col width="6%" />
<col width="7%" />
<col width="7%" />
<col width="12%" />
<col width="12%" />
<col width="7%" />
<col width="12%" />
<col width="12%" />
<col width="13%" />
</colgroup>
<thead>
<tr class="header">
<th align="center">SD</th>
<th align="center">kc</th>
<th align="center">Precop</th>
<th align="center">P1cost</th>
<th align="center">lower_P1cost</th>
<th align="center">upper_P1cost</th>
<th align="center">P2cost</th>
<th align="center">lower_P2cost</th>
<th align="center">upper_P2cost</th>
<th align="center">remating_prob</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="center">SD-5</td>
<td align="center">0.944</td>
<td align="center">NA</td>
<td align="center">0.85</td>
<td align="center">0.98</td>
<td align="center">0.4</td>
<td align="center">0.055</td>
<td align="center">0.145</td>
<td align="center">0.003</td>
<td align="center">0.755</td>
</tr>
<tr class="even">
<td align="center">SD-72</td>
<td align="center">0.909</td>
<td align="center">1.24</td>
<td align="center">0.57</td>
<td align="center">0.95</td>
<td align="center">0</td>
<td align="center">0.003</td>
<td align="center">0.031</td>
<td align="center">0</td>
<td align="center">0.425</td>
</tr>
<tr class="odd">
<td align="center">SD-MAD</td>
<td align="center">0.868</td>
<td align="center">1.19</td>
<td align="center">0.61</td>
<td align="center">NA</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0</td>
<td align="center">0.429</td>
</tr>
</tbody>
</table>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="modelling-segregation-distortion-and-sperm-competition-costs" class="section level1">
<h1>Modelling segregation distortion and sperm competition costs</h1>
<p><span class="math inline">\(~\)</span></p>
<p>Here we describe how we found zygote frequencies when offspring were produced by individuals with different genotypes (WW, WSD or SDSD), considering both segregation distortion and sperm competition.</p>
<p>In our model we assume that SDSD individuals are homozygous lethal, as is the case with many <em>SD</em> variants (see modelling viability and precopulatory sexual selection). However, in the event that this assumption is relaxed, we assume that WW and SDSD homozygotes perform identically in sperm competition against themselves or each other. This is because it is assumed that the reduction in sperm number resulting from drive is the mechanism behind the loss in competitive ability, and drive is absent in viable <em>SD</em> homozygotes.</p>
<p><strong>Parameter meanings</strong>:</p>
<ul>
<li><p><span class="math inline">\(P1_{normal}\)</span> is the average P1 for the population, where <span class="math inline">\(0 \le P1_{normal} \le 1\)</span>.</p></li>
<li><p><span class="math inline">\(P1_{cost}\)</span> is a multiplicative cost to the proportion of offspring sired when the SD male mates first, where <span class="math inline">\(0 \le P1_{cost} \le 1\)</span> (larger value = greater cost).</p></li>
<li><p><span class="math inline">\(P2_{cost}\)</span> is a multiplicative cost to the proportion of offspring sired when the SD male mates second, where <span class="math inline">\(0 \le P2_{cost} \le 1\)</span> (larger value = greater cost).</p></li>
<li><p>For example, if we measured P1 for the <em>SD/+</em> males as being 1/4 as high as for wildtype controls, we would set <span class="math inline">\(P1_{cost}\)</span> = 0.75 to recapitulate nature in the model</p></li>
<li><p><span class="math inline">\(k\)</span> is the proportion of <em>SD</em> bearing sperm a female is inseminated with when mating with a <em>SD/+</em> male.</p></li>
</ul>
<p><span class="math inline">\(~\)</span></p>
<p>Next we provide examples of how we calculated zygote frequencies for two of the potential mating combinations in our population. There are 36 possible mating combinations considered by our model and zygote frequencies for each of these combinations are displayed in the code chunk found under the heading <strong>Function to make the mating table</strong> (in the code these are named <code>rep5</code>, <code>rep25</code> and <code>p1-18</code>).</p>
<p><span class="math inline">\(~\)</span></p>
<p><strong>1. Finding offspring genotypes when a WW female mates with a WW male then a WSD male</strong></p>
<p>We think of the production of zygote genotypes as a two-step process. First, which male sires the offspring? Second, which of his possible sperm genotypes won? The second step can be ignored for WW males, but for the WSD males there is segregation distortion to consider. WW progeny can be created two ways - sired by the WW male, or sired by the WSD male, with one of his W sperm fertilising the egg. The WSD progeny can only be sired by the WSD male, and an SD sperm needs to fertilise the egg.</p>
<p><strong>Scenario 1:</strong> W sperm from the WW male fertilises the W egg</p>
<ul>
<li><p>A sperm from the WW male wins with probability: <span class="math inline">\(P1_{normal} + (1 - P1_{normal}) * P2_{cost}\)</span></p></li>
<li><p>This is the ‘normal’ or mean P1 value for the population, plus an additional amount due to an advantage in sperm competition, up to a maximum of 1 when <span class="math inline">\(P2cost = 1\)</span></p></li>
</ul>
<p><strong>Scenario 2:</strong> W sperm from the WSD male fertilises the W egg</p>
<ul>
<li><p>A sperm from the WSD male wins with probability: <span class="math inline">\(1 - (P1_{normal} + (1 - P1_{normal}) * P2_{cost})\)</span> which simplifies to: <span class="math inline">\((P1_{normal} - 1) * (P2_{cost} - 1)\)</span></p></li>
<li><p>A W sperm from this male wins with probability: <span class="math inline">\(1 - k\)</span></p></li>
<li><p>Multiply these probabilities to get the chance both occur, and you get: <span class="math inline">\((P1_{normal} - 1) * (P2_{cost} - 1) * (1 - k)\)</span></p></li>
</ul>
<p>Adding <strong>Scenario 1</strong> and <strong>Scenario 2</strong> gives the overall probability each offspring is WW:</p>
<ul>
<li><span class="math inline">\(P1_{normal} + (1 - P1_{normal}) * P2_{cost} + (P1_{normal} - 1) * (P2_{cost} - 1) * (1 - k)\)</span></li>
</ul>
<p><strong>Scenario 3:</strong> SD sperm from the WSD male fertilises the W egg, producing a zygote with the genotype WSD</p>
<ul>
<li><span class="math inline">\((P1_{normal} - 1) * (P2_{cost} - 1) * k\)</span></li>
</ul>
<p><span class="math inline">\(~\)</span></p>
<p><strong>2. Finding offspring genotypes when a WW female mates with a WSD male then a WW male (i.e. the reverse mating order)</strong></p>
<p><strong>Scenario 1:</strong> W sperm from the WW male fertilises the W egg</p>
<ul>
<li><p>A sperm from the WW male wins with probability: <span class="math inline">\(1 - (P1_{normal} - P1_{normal} * P1_{cost})\)</span></p></li>
<li><p>Similar to Scenario 1 in the first mating trio example, this is P2 (or equivalently 1 - P1), plus an additional amount due to an advantage in sperm competition.</p></li>
</ul>
<p><strong>Scenario 2:</strong> W sperm from the WSD male fertilises the W egg</p>
<ul>
<li><p>A sperm from the WSD male wins with probability: <span class="math inline">\(P1_{normal} - P1_{normal} * P1_{cost}\)</span></p></li>
<li><p>A W sperm from this male wins with probability: <span class="math inline">\(1 - k\)</span></p></li>
<li><p>Multiply these probabilities to get the chance both occur, and you get: <span class="math inline">\((P1_{normal} - P1_{normal} * P1_{cost}) * (1 - k)\)</span></p></li>
</ul>
<p>Adding <strong>Scenario 1</strong> and <strong>Scenario 2</strong> gives us the overall probability each offspring is WW:</p>
<ul>
<li><span class="math inline">\(1 - (P1_{normal} - P1_{normal} * P1_{cost}) + (P1_{normal} - P1_{normal} * P1_{cost}) * (1 - k)\)</span></li>
</ul>
<p><strong>Scenario 3:</strong> SD sperm from the WSD male wins, giving a WSD offspring</p>
<ul>
<li><span class="math inline">\((P1_{normal} - P1_{normal} * P1cost) * k\)</span></li>
</ul>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="function-to-make-the-mating-table" class="section level1">
<h1>Function to make the mating table</h1>
<p><span class="math inline">\(~\)</span></p>
<p>The “mating table” is list of all the possible mating types (e.g. a WSD mother with a WSD father, etc), the resulting possible offspring genotypes, and their associated frequencies. The precise values in the mating table are a function of <em>k</em> (i.e. the strength of segregation distortion) and the expected paternity of WSD males when the female is double-mated to a WW and WSD male, with specific paternity values depending on the order of mating (as described above). We assume that meiosis is Mendelian except in WSD males, and that all crosses produce a 50:50 sex ratio.</p>
<p>Explore the code chunk below to see how we calculated zygote frequencies for every possible mating combination (<code>rep5</code>, <code>rep25</code> and <code>p1-18</code>).</p>
<pre class="r"><code>make_mating_table &lt;- function(k, normalP1, P1cost, P2cost){
  
  make_offspring &lt;- function(X, Y, Z, type, zygote_freq){
    tibble(mother = X,
           father1 = Y,
           father2 = Z,
           type,
           zygote_freq)
  }
  
  # Set up the 36 possible mating trio types
  
  mating_types &lt;- expand_grid(
    mother = c(&quot;WW_female&quot;,
               &quot;WSD_female&quot;,
               &quot;SDSD_female&quot;),
    father1 = c(&quot;WW_male&quot;,
                &quot;WSD_male&quot;,
                &quot;SDSD_male&quot;),
    father2 = c(&quot;none&quot;,
                &quot;WW_male&quot;,
                &quot;WSD_male&quot;,
                &quot;SDSD_male&quot;))
  
  # Specify the possible offspring genotypes for all the potential crosses; we use these for the type argument in the make_offspring function
  
  # WW x WW
  
  g0 &lt;- c(&quot;WW_female&quot;, &quot;WW_male&quot;)
  
  # WSD x WW &amp; WW x WSD
  
  g1 &lt;- c(&quot;WW_female&quot;, &quot;WW_male&quot;, &quot;WSD_female&quot;, &quot;WSD_male&quot;)
  
  # SDSD x WW &amp; WW x SDSD
  
  g2 &lt;- c(&quot;WSD_female&quot;, &quot;WSD_male&quot;)
  
  # WSD x WSD
  
  g3 &lt;- c(&quot;WW_female&quot;, &quot;WW_male&quot;, &quot;WSD_female&quot;, &quot;WSD_male&quot;, &quot;SDSD_female&quot;, &quot;SDSD_male&quot;)
  
  # WSD x SDSD &amp; SDSD x WSD
  
  g4 &lt;- c(&quot;WSD_female&quot;, &quot;WSD_male&quot;, &quot;SDSD_female&quot;, &quot;SDSD_male&quot;)
  
  # SDSD x SDSD
  
  g5 &lt;- c(&quot;SDSD_female&quot;, &quot;SDSD_male&quot;)
  
  
  # Now calculate the zygote frequencies for each cross
  
  # even frequency of four offspring types
  
  rep25 &lt;- rep(0.25, 4)
  
  # even frequency between two offspring types
  
  rep5 &lt;- rep(0.5, 2)
  
  # now offspring frequencies when there&#39;s drive
  
  # when there are 4 offspring genotypes
  
  p1 &lt;- c(0.5 * (1 - k), 
          0.5 * (1 - k), 
          0.5 * k, 
          0.5 * k)
  
  # when there are 6 offspring genotypes
  
  p2 &lt;- c(0.25 * (1 - k),
          0.25 * (1 - k),
          (0.25 * (1 - k)) + (0.25 * k),
          (0.25 * (1 - k)) + (0.25 * k),
          0.25 * k,
          0.25 * k)
  
  # Now add sperm comp zygote frequencies
  
  # when WW mates with WW &amp; WSD
  
  p3 &lt;- c(0.5 * ((normalP1 + (1 - normalP1) * P2cost) + (normalP1 - 1) * (P2cost - 1) * (1-k)), # WW female offspring
          0.5 * ((normalP1 + (1 - normalP1) * P2cost) + (normalP1 - 1) * (P2cost - 1) * (1-k)), # WW male offspring
          0.5 * ((normalP1 - 1) * (P2cost - 1) * k), # WSD female offspring
          0.5 * ((normalP1 - 1) * (P2cost - 1) * k)) # WSD female offspring
  
  # when WW mates with WSD &amp; WW
  
  p4 &lt;- c(0.5 * ((normalP1 - normalP1 * P1cost) * (1 - k) + 1 - (normalP1 - normalP1 * P1cost)), # WW female offspring
          0.5 * ((normalP1 - normalP1 * P1cost) * (1 - k) + 1 - (normalP1 - normalP1 * P1cost)), # WW male offspring
          0.5 * ((normalP1 - normalP1 * P1cost) * k), # WSD females
          0.5 * ((normalP1 - normalP1 * P1cost) * k)) # WSD females
  
  
  # when WW mates with WW and SDSD &amp; when SDSD mates with WW and SDSD
  
  p5 &lt;- c(0.5 * normalP1, # WW females &amp; WSD females
          0.5 * normalP1, # WW males &amp; WSD males
          0.5 * (1 - normalP1), # WSD females and SDSD females
          0.5 * (1 - normalP1)) # WSD males and SDSD males
  
  # when WW mates with SDSD and WW &amp; when SDSD mates with SDSD and WW
  
  p6 &lt;- c(0.5 * (1 - normalP1), # WW females &amp; WSD females
          0.5 * (1 - normalP1), # WW males &amp; WSD males
          0.5 * normalP1, # WSD females &amp; SDSD females
          0.5 * normalP1) # WSD males &amp; SDSD females
  
  # when WSD mates with WW and WSD
  
  p7 &lt;- c(0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5), # WW females
          0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5), # WW males
          0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5 + (normalP1 - 1) * (P2cost - 1) * k * 0.5), # WSD females
          0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5 + (normalP1 - 1) * (P2cost - 1) * k * 0.5), # WSD males
          0.5 * (normalP1 - 1) * (P2cost - 1) * k * 0.5, # SDSD females
          0.5 * (normalP1 - 1) * (P2cost - 1) * k * 0.5) # SDSD males
  
  
  # when WSD mates with WSD and WW
  
  p8 &lt;- c(0.5 * ((normalP1 - normalP1 * P1cost) * (1 - k) * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5), # WW female 
          0.5 * ((normalP1 - normalP1 * P1cost) * (1 - k) * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5), # WW male
          0.5 * ((normalP1 - normalP1 * P1cost) * k * 0.5 + (normalP1 - normalP1 * P1cost) * (1 - k) * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5), # WSD female
          0.5 * ((normalP1 - normalP1 * P1cost) * k * 0.5 + (normalP1 - normalP1 * P1cost) * (1 - k) * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5), # WSD male
          0.5 * (normalP1 - normalP1 * P1cost) * k * 0.5, # SDSD female
          0.5 * (normalP1 - normalP1 * P1cost) * k * 0.5) # SDSD male
  
  
  # when WW mates with WSD and SDSD 
  # Because there is no drive we assume SDSD = WW for sperm comp
  
  p9 &lt;- c( 0.5 * ((normalP1 - normalP1 * P1cost) * (1 - k)), # WW females
           0.5 * ((normalP1 - normalP1 * P1cost) * (1 - k)), # WW males
           0.5 * ((normalP1 - normalP1 * P1cost) * k + 1 - (normalP1 - normalP1 * P1cost)), # WSD female offspring
           0.5 * ((normalP1 - normalP1 * P1cost) * k + 1 - (normalP1 - normalP1 * P1cost))) # WSD male offspring
  
  # when WW mates with SDSD and WSD
  
  p10 &lt;- c(0.5 * (normalP1 - 1) * (P2cost - 1) * (1 - k), # WW female offspring
           0.5 * (normalP1 - 1) * (P2cost - 1) * (1 - k), # WW male offspring
           0.5 * ((normalP1 + (1 - normalP1) * P2cost) + (normalP1 - 1) * (P2cost - 1) * k), # WSD female offspring
           0.5 * ((normalP1 + (1 - normalP1) * P2cost) + (normalP1 - 1) * (P2cost - 1) * k)) # WSD female offspring
  
  
  # when WSD mates with WW and SDSD
  
  p11 &lt;- c(0.5 * (normalP1 * 0.5), # WW females
           0.5 * (normalP1 * 0.5), # WW males
           0.5 * ((normalP1 * 0.5) + (1 - normalP1) * 0.5), # WSD females
           0.5 * ((normalP1 * 0.5) + (1 - normalP1) * 0.5), # WSD males
           0.5 * ((1 - normalP1) * 0.5), # SDSD females
           0.5 * ((1 - normalP1) * 0.5)) # SDSD males
  
  
  # when WSD mates with SDSD and WW
  
  p12 &lt;- c(0.5 * ((1 - normalP1) * 0.5), # WW females
           0.5 * ((1 - normalP1) * 0.5), # WW males
           0.5 * ((normalP1 * 0.5) + (1 - normalP1) * 0.5), # WSD females
           0.5 * ((normalP1 * 0.5) + (1 - normalP1) * 0.5), # WSD males
           0.5 * (normalP1 * 0.5), # SDSD females
           0.5 * (normalP1 * 0.5)) # SDSD males
  
  
  # WSD mates with WSD and SDSD
  
  p13 &lt;- c(0.5 * (normalP1 - normalP1 * P1cost) * (1 - k) * 0.5, # WW female
           0.5 * (normalP1 - normalP1 * P1cost) * (1 - k) * 0.5, # WW male
           0.5 * ((normalP1 - normalP1 * P1cost) * k * 0.5 + (normalP1 - normalP1 * P1cost) * (1 - k) * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5), # WSD female
           0.5 * ((normalP1 - normalP1 * P1cost) * k * 0.5 + (normalP1 - normalP1 * P1cost) * (1 - k) * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5), # WSD male
           0.5 * ((normalP1 - normalP1 * P1cost) * k * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5), # SDSD female 
           0.5 * ((normalP1 - normalP1 * P1cost) * k * 0.5 + (1 - (normalP1 - normalP1 * P1cost)) * 0.5)) # SDSD male
  
  
  
  # WSD mates with SDSD and WSD
  
  p14 &lt;- c(0.5 * (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5, # WW females
           0.5 * (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5, # WW males
           0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5 + (normalP1 - 1) * (P2cost - 1) * k * 0.5), # WSD females
           0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * (1 - k) * 0.5 + (normalP1 - 1) * (P2cost - 1) * k * 0.5), # WSD males
           0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * k * 0.5), # SDSD females
           0.5 * (((normalP1 + (1 - normalP1) * P2cost) * 0.5) + (normalP1 - 1) * (P2cost - 1) * k * 0.5)) # SDSD males
  
  
  # SDSD mates with WW and WSD
  
  p15 &lt;- c(0.5 * ((normalP1 + (1 - normalP1) * P2cost) + (normalP1 - 1) * (P2cost - 1) * (1 - k)), # WSD females 
           0.5 * ((normalP1 + (1 - normalP1) * P2cost) + (normalP1 - 1) * (P2cost - 1) * (1 - k)), # WSD males 
           0.5 * (normalP1 - 1) * (P2cost - 1) * k, # SDSD females
           0.5 * (normalP1 - 1) * (P2cost - 1) * k) # SDSD males
  
  # SDSD mates with WSD and WW
  
  p16 &lt;- c(0.5 * ((1 - (normalP1 - normalP1 * P1cost)) + (normalP1 - normalP1 * P1cost) * (1 - k)), # WSD females 
           0.5 * ((1 - (normalP1 - normalP1 * P1cost)) + (normalP1 - normalP1 * P1cost) * (1 - k)), # WSD males 
           0.5 * (normalP1 - normalP1 * P1cost) * k, # SDSD females
           0.5 * (normalP1 - normalP1 * P1cost) * k) # SDSD males
  
  
  # SDSD mates with WSD and SDSD
  
  p17 &lt;- c(0.5 * (normalP1 - normalP1 * P1cost) * (1 - k), ## WSD female
           0.5 *(normalP1 - normalP1 * P1cost) * (1 - k), ## WSD male
           0.5 * ((normalP1 - normalP1 * P1cost) * k + (1 - (normalP1 - normalP1 * P1cost))), ## SDSD female
           0.5 * ((normalP1 - normalP1 * P1cost) * k + (1 - (normalP1 - normalP1 * P1cost)))) ## SDSD male
  
  
  # SDSD mates with SDSD and WSD
  
  p18 &lt;- c(0.5 * ((normalP1 - 1) * (P2cost - 1) * (1 - k)), # WSD females
           0.5 * ((normalP1 - 1) * (P2cost - 1) * (1 - k)), # WSD males
           0.5 * ((normalP1 - 1) * (P2cost - 1) * k + (normalP1 + (1 - normalP1) * P2cost)), # SDSD females
           0.5 * ((normalP1 - 1) * (P2cost - 1) * k + (normalP1 + (1 - normalP1) * P2cost))) # SDSD males
  
  
  bind_rows(
    list(
      make_offspring(&quot;WW&quot;, &quot;WW&quot;, &quot;none&quot;, g0, rep5),
      make_offspring(&quot;WW&quot;, &quot;WSD&quot;, &quot;none&quot;, g1, p1),
      make_offspring(&quot;WW&quot;, &quot;SDSD&quot;, &quot;none&quot;, g2, rep5),
      make_offspring(&quot;WSD&quot;, &quot;WW&quot;, &quot;none&quot;, g1, rep25),
      make_offspring(&quot;WSD&quot;, &quot;WSD&quot;, &quot;none&quot;, g3, p2),
      make_offspring(&quot;WSD&quot;, &quot;SDSD&quot;, &quot;none&quot;, g4, rep25),
      make_offspring(&quot;SDSD&quot;, &quot;WW&quot;, &quot;none&quot;, g2, rep5),
      make_offspring(&quot;SDSD&quot;, &quot;WSD&quot;, &quot;none&quot;, g4, p1),
      make_offspring(&quot;SDSD&quot;, &quot;SDSD&quot;, &quot;none&quot;, g5, rep5),
      make_offspring(&quot;WW&quot;, &quot;WW&quot;, &quot;WW&quot;, g0, rep5),
      make_offspring(&quot;WW&quot;, &quot;WW&quot;, &quot;WSD&quot;, g1, p3), 
      make_offspring(&quot;WW&quot;, &quot;WW&quot;, &quot;SDSD&quot;, g1, p5),
      make_offspring(&quot;WW&quot;, &quot;WSD&quot;, &quot;WW&quot;, g1, p4),
      make_offspring(&quot;WW&quot;, &quot;WSD&quot;, &quot;WSD&quot;, g1, rep25),
      make_offspring(&quot;WW&quot;, &quot;WSD&quot;, &quot;SDSD&quot;, g1, p9),
      make_offspring(&quot;WW&quot;, &quot;SDSD&quot;, &quot;WW&quot;, g1, p6),
      make_offspring(&quot;WW&quot;, &quot;SDSD&quot;, &quot;WSD&quot;, g1, p10),
      make_offspring(&quot;WW&quot;, &quot;SDSD&quot;, &quot;SDSD&quot;, g2, rep5),
      make_offspring(&quot;WSD&quot;, &quot;WW&quot;, &quot;WW&quot;, g1, rep25),
      make_offspring(&quot;WSD&quot;, &quot;WW&quot;, &quot;WSD&quot;, g3, p7),
      make_offspring(&quot;WSD&quot;, &quot;WW&quot;, &quot;SDSD&quot;, g3, p11),
      make_offspring(&quot;WSD&quot;, &quot;WSD&quot;, &quot;WW&quot;, g3, p8),
      make_offspring(&quot;WSD&quot;, &quot;WSD&quot;, &quot;WSD&quot;, g3, p2),
      make_offspring(&quot;WSD&quot;, &quot;WSD&quot;, &quot;SDSD&quot;, g3, p13),
      make_offspring(&quot;WSD&quot;, &quot;SDSD&quot;, &quot;WW&quot;, g3, p12),
      make_offspring(&quot;WSD&quot;, &quot;SDSD&quot;, &quot;WSD&quot;, g3, p14),
      make_offspring(&quot;WSD&quot;, &quot;SDSD&quot;, &quot;SDSD&quot;, g4, rep25),
      make_offspring(&quot;SDSD&quot;, &quot;WW&quot;, &quot;WW&quot;, g2, rep5),
      make_offspring(&quot;SDSD&quot;, &quot;WW&quot;, &quot;WSD&quot;, g4, p15),
      make_offspring(&quot;SDSD&quot;, &quot;WW&quot;, &quot;SDSD&quot;, g4, p5),
      make_offspring(&quot;SDSD&quot;, &quot;WSD&quot;, &quot;WW&quot;, g4, p16),
      make_offspring(&quot;SDSD&quot;, &quot;WSD&quot;, &quot;WSD&quot;, g4, p1),
      make_offspring(&quot;SDSD&quot;, &quot;WSD&quot;, &quot;SDSD&quot;, g4, p17),
      make_offspring(&quot;SDSD&quot;, &quot;SDSD&quot;, &quot;WW&quot;, g4, p6),
      make_offspring(&quot;SDSD&quot;, &quot;SDSD&quot;, &quot;WSD&quot;, g4, p18),
      make_offspring(&quot;SDSD&quot;, &quot;SDSD&quot;, &quot;SDSD&quot;, g5, rep5)
    )) %&gt;% 
    arrange(father2, mother, father1)
  
}</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="an-example-of-a-mating-table" class="section level1">
<h1>An example of a mating table</h1>
<p><span class="math inline">\(~\)</span></p>
<p>Assume <span class="math inline">\(k\)</span> = 0.9, and the following paternity values: <span class="math inline">\(normal_{P1}\)</span> = 0.1, <span class="math inline">\(P1_{cost}\)</span> = 0.5, and <span class="math inline">\(P2_{cost}\)</span> = 0.5</p>
<pre class="r"><code># Make the table

make_mating_table(k = 0.9, normalP1 = 0.1, P1cost = 0.5, P2cost = 0.5) %&gt;% 
  rename(zygote_type = type) %&gt;% 
  kable() %&gt;% 
  kable_styling() %&gt;% 
  scroll_box(height = &quot;500px&quot;)</code></pre>
<div style="border: 1px solid #ddd; padding: 0px; overflow-y: scroll; height:500px; ">
<table class="table" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
mother
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
father1
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
father2
</th>
<th style="text-align:left;position: sticky; top:0; background-color: #FFFFFF;">
zygote_type
</th>
<th style="text-align:right;position: sticky; top:0; background-color: #FFFFFF;">
zygote_freq
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
none
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.00250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.00250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.49750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.49750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.00125
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.00125
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.24875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.24875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.00250
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.00250
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.49750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.49750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.29750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.29750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.20250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.20250
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.01125
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.01125
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.23875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.23875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.14875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.14875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.10125
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.10125
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.29750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.29750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.20250
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.20250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.22500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.02500
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.23875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.23875
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_female
</td>
<td style="text-align:right;">
0.01125
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD_male
</td>
<td style="text-align:right;">
0.01125
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.25000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.45000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
SDSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.05000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.47750
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_female
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WSD_male
</td>
<td style="text-align:right;">
0.02250
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_female
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
<tr>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW
</td>
<td style="text-align:left;">
WW_male
</td>
<td style="text-align:right;">
0.50000
</td>
</tr>
</tbody>
</table>
</div>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="function-to-determine-the-mating-type-frequencies-of-the-population" class="section level1">
<h1>Function to determine the mating type frequencies of the population</h1>
<p><span class="math inline">\(~\)</span></p>
<p>This is a two-step process. First we determine the frequencies of each genotype in the mating population, then we calculate the frequencies of each potential mating combination between these genotypes.</p>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Step 1 - finding the genotype frequencies in the mating population</strong></p>
<p>While not measured in our study, we assume individuals with the genotype SDSD do not survive to adulthood (<code>egg_adult_viability</code> = 0), and that the viability of WSD individuals is unaffected.</p>
<pre class="r"><code># viability of each genotype

egg_adult_viability_WSD_female &lt;- 1

egg_adult_viability_SDSD_female &lt;- 0

egg_adult_viability_WSD_male &lt;- 1

egg_adult_viability_SDSD_male &lt;- 0</code></pre>
<p>Next, we model the effect of the WSD genotype on a male’s mating success. We explore the parameter space where WSD males suffer costs when competing for copulations (<code>S_precop$S_precop_WSD_male</code> &lt; 1), where the WSD genotype has the same mating success as the WW genotype (<code>S_precop$S_precop_WSD_male</code> = 1), and the parameter space where the WSD genotype is more successful than the WW genotype (<code>S_precop$S_precop_WSD_male</code> &gt; 1). While it is possible that male choice for females may disadvantage <em>SD</em> carrying females, we have focused our efforts on measuring how precopulatory sexual selection affects the <em>SD</em> allele in males, as low quality males are more likely to go unmated than low quality females in the <em>D. melanogaster</em> polyandrous mating system.</p>
<p>Using these parameters, we implement selection, so that each genotype is represented in the mating combinations according to the product of its frequency, its viability and its mating success.</p>
<p><span class="math inline">\(~\)</span></p>
<p><strong>Step 2 - finding the frequency of all potential mating combinations in the mating population</strong></p>
<p>We now calculate the expected frequencies of each possible ‘mating combination’ (i.e. the combination of the maternal and paternal genotypes) across the population. For singly-mated females, the frequency of each mating type is given by <span class="math inline">\(F_iM_j(1 – p_j)\)</span>, where <span class="math inline">\(F_i\)</span> and <span class="math inline">\(M_j\)</span> are the female and male parental genotype frequencies, and <span class="math inline">\(p_j\)</span> is the probability of female remating following a first mating with a male of genotype <span class="math inline">\(j\)</span>. Similarly, we found the expected frequencies of each possible mating type for females that mated with two males. We accomplished this via the formula <span class="math inline">\(F_iM_jN_kp_j\)</span>, where <span class="math inline">\(N_k\)</span> represents the genotype frequency of the second male to mate.</p>
<pre class="r"><code>find_mating_type_frequencies &lt;- function(pop, prob_remating_SD, prob_remating_W){
  
  # Step 1
  
  pop$prop &lt;- pop$prop * pop$egg_adult_viability * pop$S_precop # here&#39;s the selection step 
  pop$prop &lt;- pop$prop / sum(pop$prop) # now restandardise so that the props sum to 1
  
  # Step 2
  
  # Calculate the frequency of each single-mating combination
  
  first_mating_types &lt;- 
    expand_grid(mother = pop$type[str_detect(pop$type, &quot;female&quot;)], # find all possible mother genotypes
                father1 = pop$type[!str_detect(pop$type, &quot;female&quot;)]) %&gt;% # find all possible father1 genotypes
    mutate(prop_matings = 0, remating_probability = 0) # make placeholder columns we will fill in below
  
  # find genotype frequencies calculated for each sex (mate choice means these will not necessarily be 50:50)
  
  female_frequencies &lt;- pop %&gt;% filter(str_detect(type, &quot;female&quot;)) %&gt;% mutate(prop = prop / sum(prop))
  male_frequencies &lt;- pop %&gt;% filter(!str_detect(type, &quot;female&quot;)) %&gt;% mutate(prop = prop / sum(prop))
  
  # This next step multiplies the proportion[i] of females with genotype[i] with the proportion[i] of males with genotype[i] 
  
  for(i in 1:nrow(first_mating_types)){
    first_mating_types$prop_matings[i] &lt;- female_frequencies$prop[female_frequencies$type == first_mating_types$mother[i]] * 
      male_frequencies$prop[male_frequencies$type == first_mating_types$father1[i]]
  }
  
  # Now we incorporate female remating probabilities (SDSD males assumed to have same effect on females as WW males)
  
  first_mating_types$remating_probability &lt;- ifelse(str_detect(first_mating_types$father1, &quot;WSD&quot;), prob_remating_SD, prob_remating_W)
  
  # Now lets calculate the frequency of each double-mating combination, with the end product being the mating_type_frequency of each mating possibility
  
  second_mating_types &lt;- expand_grid(first_mating_types, father2 = c(&quot;none&quot;, unique(first_mating_types$father1))) %&gt;%
    mutate(mating_type_freq = 0)
  
  # scale the proportion of matings calculated above by the likelihood that these females will mate once
  
  for(i in 1:nrow(second_mating_types)){
    if(second_mating_types$father2[i] == &quot;none&quot;){
      second_mating_types$mating_type_freq[i] &lt;- 
        second_mating_types$prop_matings[i] * # Frequency of females with this value of father1
        (1 - second_mating_types$remating_probability[i])
    }
    
    # now calculate the frequency of mating tryads for the females that mate twice
    
    if(second_mating_types$father2[i] != &quot;none&quot;){
      second_mating_types$mating_type_freq[i] &lt;- 
        second_mating_types$prop_matings[i] * # Frequency of females with this value of father1
        second_mating_types$remating_probability[i] *
        male_frequencies$prop[male_frequencies$type == second_mating_types$father2[i]] # pop. freq of the 2nd male type
    }
  }
  
  second_mating_types %&gt;% # output
    select(mother, father1, father2, mating_type_freq) %&gt;%
    # clean up the output by remving redundant info (makes it easier later)
    mutate(mother = str_remove_all(mother, &quot;_female&quot;),
           father1 = str_remove_all(father1, &quot;_male&quot;),
           father2 = str_remove_all(father2, &quot;_male&quot;))
}</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="create-a-simple-function-to-find-the-population-wide-frequency-of-sd" class="section level1">
<h1>Create a simple function to find the population-wide frequency of <em>SD</em></h1>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code>calc_prop_SD &lt;- function(pop){
  0.5 * sum(pop$prop[str_detect(pop$type, &quot;WSD&quot;)]) + sum(pop$prop[str_detect(pop$type, &quot;SDSD&quot;)])
}</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="define-the-main-simulation-function" class="section level1">
<h1>Define the main simulation function</h1>
<p><span class="math inline">\(~\)</span></p>
<p>This function iterates over generations, implementing selection and reproduction each time, until either A) the <em>SD</em> chromosome fixes, B) the <em>SD</em> chromosome goes extinct or C) the generation timer expires</p>
<pre class="r"><code>run_simulation &lt;- function(row, parameters){
  
  print(paste(&quot;Doing row&quot;, row)) # this tells you which row in the parameter space is being modelled
  
  # Get the focal parameter space out of parameters[row], this makes it easier later
  
  generations &lt;- parameters$generations[row]
  k &lt;- parameters$k[row]
  P1cost &lt;- parameters$P1cost[row]
  P2cost &lt;- parameters$P2cost[row]
  normalP1 &lt;- parameters$normalP1[row]
  prob_remating_SD &lt;- parameters$prob_remating_SD[row]
  prob_remating_W &lt;- parameters$prob_remating_W[row]
  egg_adult_viability_WSD_female &lt;- parameters$egg_adult_viability_WSD_female[row]
  egg_adult_viability_SDSD_female &lt;- parameters$egg_adult_viability_SDSD_female[row]
  egg_adult_viability_WSD_male &lt;- parameters$egg_adult_viability_WSD_male[row]
  egg_adult_viability_SDSD_male &lt;- parameters$egg_adult_viability_SDSD_male[row]
  S_precop_WSD_male &lt;- parameters$S_precop_WSD_male[row]
  initial_freq_SD &lt;- parameters$initial_freq_SD[row]
  
  
  # Make the initial population
  pop &lt;- 
    tibble(
      type = c(&quot;WW_female&quot;,
               &quot;WSD_female&quot;,
               &quot;SDSD_female&quot;,
               &quot;WW_male&quot;,
               &quot;WSD_male&quot;,
               &quot;SDSD_male&quot;),
      prop = c(0.5 * (1 - initial_freq_SD) ^ 2,
               0.5 * (1 - initial_freq_SD) * initial_freq_SD * 2,
               0.5 * (initial_freq_SD^2),
               0.5 * (1 - initial_freq_SD) ^ 2,
               0.5 * (1 - initial_freq_SD) * initial_freq_SD * 2,
               0.5 * (initial_freq_SD^2)),
      egg_adult_viability = c(1, egg_adult_viability_WSD_female, egg_adult_viability_SDSD_female, 1,
                              egg_adult_viability_WSD_male, egg_adult_viability_SDSD_male),
      S_precop = c(1, 1, 1, 1, S_precop_WSD_male, 1)) %&gt;% 
    arrange(type)
  
  # Make the mating table for this specific parameter space
  
  mating_table &lt;- make_mating_table(k, normalP1, P1cost, P2cost)
  
  # Save the fitnesses of each genotype
  
  fitnesses &lt;- pop %&gt;% 
    select(type, egg_adult_viability, S_precop) 
  
  current_generation &lt;- 1
  keep_going &lt;- TRUE
  
  # Iterate over generations
  
  while(current_generation &lt;= generations &amp; keep_going){
    
    # Find the mating type frequencies, find the offspring frequencies, and renormalise the frequencies to sum to one every generation
    
    mating_type_frequencies &lt;- find_mating_type_frequencies(
      pop = pop, 
      prob_remating_SD = prob_remating_SD, 
      prob_remating_W = prob_remating_W)
    
    pop &lt;- left_join(mating_table, mating_type_frequencies,
                     by = c(&quot;mother&quot;, &quot;father1&quot;, &quot;father2&quot;)) %&gt;%
      mutate(prop = mating_type_freq * zygote_freq) %&gt;%
      group_by(type) %&gt;%
      summarise(prop = sum(prop)) %&gt;%
      left_join(fitnesses, by = &quot;type&quot;)
    
    # Calculate the frequency of SD, quit early if SD fixes or goes extinct-ish
    prop_SD &lt;- calc_prop_SD(pop)
    
    if(prop_SD &gt; 0.99 | prop_SD &lt; 0.0001) keep_going &lt;- FALSE
    
    # Otherwise, increment generation counter by 1 and return to top of the &#39;while&#39; loop
    
    current_generation &lt;- current_generation + 1
  }
  
  pop # returns the final pop
  
}</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="define-the-parameter-space-to-investigate" class="section level1">
<h1>Define the parameter space to investigate</h1>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code>resolution &lt;- 25

gen &lt;- 1000

initial_freq_SD &lt;- 0.01

# updated with latest parameters

parameters &lt;- rbind(
  expand_grid(
    generations = gen,
    k = c(estimated_k_SD5, estimated_k_SD72, estimated_k_SDMAD),
    normalP1 = c(0.1, 0.5),
    P1cost = seq(0, 1, length = resolution),
    P2cost = seq(0, 1, length = resolution),
    prob_remating_W = estimated_prob_remating_W,
    prob_remating_SD = c(estimated_prob_remating_W, estimated_prob_remating_SD72, estimated_prob_remating_SD5),
    egg_adult_viability_WSD_female = egg_adult_viability_WSD_female,
    egg_adult_viability_SDSD_female = egg_adult_viability_SDSD_female,
    egg_adult_viability_WSD_male = egg_adult_viability_WSD_male,
    egg_adult_viability_SDSD_male = egg_adult_viability_SDSD_male,
    S_precop_WSD_male = c(0.8, 1, 1.2),
    initial_freq_SD = initial_freq_SD)
) %&gt;% mutate(parameter_space_ID = row_number())</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="run-the-simulation" class="section level1">
<h1>Run the simulation</h1>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code># We run the simulation if the results don&#39;t already exist. The output is very large, which causes my mac to run out of memory before the simulation can finish. Therefore we break the simulation into 33 chunks to accommodate my 16gB of ram. After each chunk finishes, its output is saved to the hard disk and is then deleted from local memory, allowing the simulation to finish. We then combine the 33 chunks into a single object and save this as output.rds

if(!file.exists(&quot;simulation_results/output1.rds&quot;)){
  output1 &lt;- lapply(1:1000, run_simulation, parameters = parameters)
  saveRDS(output1, &quot;simulation_results/output1.rds&quot;)
}

rm(output1)
#gc()

if(!file.exists(&quot;simulation_results/output2.rds&quot;)){
  output2 &lt;- lapply(1001:2000, run_simulation, parameters = parameters)
  saveRDS(output2, &quot;simulation_results/output2.rds&quot;)
}

rm(output2)
#gc()

if(!file.exists(&quot;simulation_results/output3.rds&quot;)){
  output3 &lt;- lapply(2001:3000, run_simulation, parameters = parameters)
  saveRDS(output3, &quot;simulation_results/output3.rds&quot;)
}

rm(output3)
#gc()

if(!file.exists(&quot;simulation_results/output4.rds&quot;)){
  output4 &lt;- lapply(3001:4000, run_simulation, parameters = parameters)
  saveRDS(output4, &quot;simulation_results/output4.rds&quot;)
}

rm(output4)
#gc()
  
if(!file.exists(&quot;simulation_results/output5.rds&quot;)){
  output5 &lt;- lapply(4001:5000, run_simulation, parameters = parameters)
  saveRDS(output5, &quot;simulation_results/output5.rds&quot;)
}

rm(output5)
#gc()

if(!file.exists(&quot;simulation_results/output6.rds&quot;)){
  output6 &lt;- lapply(5001:6000, run_simulation, parameters = parameters)
  saveRDS(output6, &quot;simulation_results/output6.rds&quot;)
}

rm(output6)
#gc()

if(!file.exists(&quot;simulation_results/output7.rds&quot;)){
  output7 &lt;- lapply(6001:7000, run_simulation, parameters = parameters)
  saveRDS(output7, &quot;simulation_results/output7.rds&quot;)
}

rm(output7)
#gc()

if(!file.exists(&quot;simulation_results/output8.rds&quot;)){
  output8 &lt;- lapply(7001:8000, run_simulation, parameters = parameters)
  saveRDS(output8, &quot;simulation_results/output8.rds&quot;)
}

rm(output8)
#gc()

if(!file.exists(&quot;simulation_results/output9.rds&quot;)){
  output9 &lt;- lapply(8001:9000, run_simulation, parameters = parameters)
  saveRDS(output9, &quot;simulation_results/output9.rds&quot;)
}

rm(output9)
#gc()
  
if(!file.exists(&quot;simulation_results/output10.rds&quot;)){
  output10 &lt;- lapply(9001:10000, run_simulation, parameters = parameters)
  saveRDS(output10, &quot;simulation_results/output10.rds&quot;)
}

rm(output10)
#gc()

if(!file.exists(&quot;simulation_results/output11.rds&quot;)){
  output11 &lt;- lapply(10001:11000, run_simulation, parameters = parameters)
  saveRDS(output11, &quot;simulation_results/output11.rds&quot;)
}

rm(output11)
#gc()

if(!file.exists(&quot;simulation_results/output12.rds&quot;)){
  output12 &lt;- lapply(11001:12000, run_simulation, parameters = parameters)
  saveRDS(output12, &quot;simulation_results/output12.rds&quot;)
}

rm(output12)
#gc()

if(!file.exists(&quot;simulation_results/output13.rds&quot;)){
  output13 &lt;- lapply(12001:13000, run_simulation, parameters = parameters)
  saveRDS(output13, &quot;simulation_results/output13.rds&quot;)
}

rm(output13)
#gc()

if(!file.exists(&quot;simulation_results/output14.rds&quot;)){
  output14 &lt;- lapply(13001:14000, run_simulation, parameters = parameters)
  saveRDS(output14, &quot;simulation_results/output14.rds&quot;)
}

rm(output14)
#gc()
  
if(!file.exists(&quot;simulation_results/output15.rds&quot;)){
  output15 &lt;- lapply(14001:15000, run_simulation, parameters = parameters)
  saveRDS(output15, &quot;simulation_results/output15.rds&quot;)
}

rm(output15)
#gc()

if(!file.exists(&quot;simulation_results/output16.rds&quot;)){
  output16 &lt;- lapply(15001:16000, run_simulation, parameters = parameters)
  saveRDS(output16, &quot;simulation_results/output16.rds&quot;)
}

rm(output16)
#gc()

if(!file.exists(&quot;simulation_results/output17.rds&quot;)){
  output17 &lt;- lapply(16001:17000, run_simulation, parameters = parameters)
  saveRDS(output17, &quot;simulation_results/output17.rds&quot;)
}

rm(output17)
#gc()

if(!file.exists(&quot;simulation_results/output18.rds&quot;)){
  output18 &lt;- lapply(17001:18000, run_simulation, parameters = parameters)
  saveRDS(output18, &quot;simulation_results/output18.rds&quot;)
}

rm(output18)
#gc()

if(!file.exists(&quot;simulation_results/output19.rds&quot;)){
  output19 &lt;- lapply(18001:19000, run_simulation, parameters = parameters)
  saveRDS(output19, &quot;simulation_results/output19.rds&quot;)
}

rm(output19)
#gc()
  
if(!file.exists(&quot;simulation_results/output20.rds&quot;)){
  output20 &lt;- lapply(19001:20000, run_simulation, parameters = parameters)
  saveRDS(output20, &quot;simulation_results/output20.rds&quot;)
}

rm(output20)
#gc()


if(!file.exists(&quot;simulation_results/output21.rds&quot;)){
  output21 &lt;- lapply(20001:21000, run_simulation, parameters = parameters)
  saveRDS(output21, &quot;simulation_results/output21.rds&quot;)
}

rm(output21)
#gc()

if(!file.exists(&quot;simulation_results/output22.rds&quot;)){
  output22 &lt;- lapply(21001:22000, run_simulation, parameters = parameters)
  saveRDS(output22, &quot;simulation_results/output22.rds&quot;)
}

rm(output22)
#gc()

if(!file.exists(&quot;simulation_results/output23.rds&quot;)){
  output23 &lt;- lapply(22001:23000, run_simulation, parameters = parameters)
  saveRDS(output23, &quot;simulation_results/output23.rds&quot;)
}

rm(output23)
#gc()

if(!file.exists(&quot;simulation_results/output24.rds&quot;)){
  output24 &lt;- lapply(23001:24000, run_simulation, parameters = parameters)
  saveRDS(output24, &quot;simulation_results/output24.rds&quot;)
}

rm(output24)
#gc()
  
if(!file.exists(&quot;simulation_results/output25.rds&quot;)){
  output25 &lt;- lapply(24001:25000, run_simulation, parameters = parameters)
  saveRDS(output25, &quot;simulation_results/output25.rds&quot;)
}

rm(output25)
#gc()


if(!file.exists(&quot;simulation_results/output26.rds&quot;)){
  output26 &lt;- lapply(25001:26000, run_simulation, parameters = parameters)
  saveRDS(output26, &quot;simulation_results/output26.rds&quot;)
}

rm(output26)
#gc()

if(!file.exists(&quot;simulation_results/output27.rds&quot;)){
  output27 &lt;- lapply(26001:27000, run_simulation, parameters = parameters)
  saveRDS(output27, &quot;simulation_results/output27.rds&quot;)
}

rm(output27)
#gc()

if(!file.exists(&quot;simulation_results/output28.rds&quot;)){
  output28 &lt;- lapply(27001:28000, run_simulation, parameters = parameters)
  saveRDS(output28, &quot;simulation_results/output28.rds&quot;)
}

rm(output28)
#gc()

if(!file.exists(&quot;simulation_results/output29.rds&quot;)){
  output29 &lt;- lapply(28001:29000, run_simulation, parameters = parameters)
  saveRDS(output29, &quot;simulation_results/output29.rds&quot;)
}

rm(output29)
#gc()
  
if(!file.exists(&quot;simulation_results/output30.rds&quot;)){
  output30 &lt;- lapply(29001:30000, run_simulation, parameters = parameters)
  saveRDS(output30, &quot;simulation_results/output30.rds&quot;)
}

rm(output30)
#gc()

if(!file.exists(&quot;simulation_results/output31.rds&quot;)){
  output31 &lt;- lapply(30001:31000, run_simulation, parameters = parameters)
  saveRDS(output31, &quot;simulation_results/output31.rds&quot;)
}

rm(output31)
#gc()

if(!file.exists(&quot;simulation_results/output32.rds&quot;)){
  output32 &lt;- lapply(31001:32000, run_simulation, parameters = parameters)
  saveRDS(output32, &quot;simulation_results/output32.rds&quot;)
}

rm(output32)
#gc()

if(!file.exists(&quot;simulation_results/output33.rds&quot;)){
  output33 &lt;- lapply(32001:33750, run_simulation, parameters = parameters)
  saveRDS(output33, &quot;simulation_results/output33.rds&quot;)
}

# reload chunks from disk (output 33 was never deleted)
output1 &lt;- readRDS(&quot;simulation_results/output1.rds&quot;)
output2 &lt;- readRDS(&quot;simulation_results/output2.rds&quot;)
output3 &lt;- readRDS(&quot;simulation_results/output3.rds&quot;)
output4 &lt;- readRDS(&quot;simulation_results/output4.rds&quot;)
output5 &lt;- readRDS(&quot;simulation_results/output5.rds&quot;)
output6 &lt;- readRDS(&quot;simulation_results/output6.rds&quot;)
output7 &lt;- readRDS(&quot;simulation_results/output7.rds&quot;)
output8 &lt;- readRDS(&quot;simulation_results/output8.rds&quot;)
output9 &lt;- readRDS(&quot;simulation_results/output9.rds&quot;)
output10 &lt;- readRDS(&quot;simulation_results/output10.rds&quot;)
output11 &lt;- readRDS(&quot;simulation_results/output11.rds&quot;)
output12 &lt;- readRDS(&quot;simulation_results/output12.rds&quot;)
output13 &lt;- readRDS(&quot;simulation_results/output13.rds&quot;)
output14 &lt;- readRDS(&quot;simulation_results/output14.rds&quot;)
output15 &lt;- readRDS(&quot;simulation_results/output15.rds&quot;)
output16 &lt;- readRDS(&quot;simulation_results/output16.rds&quot;)
output17 &lt;- readRDS(&quot;simulation_results/output17.rds&quot;)
output18 &lt;- readRDS(&quot;simulation_results/output18.rds&quot;)
output19 &lt;- readRDS(&quot;simulation_results/output19.rds&quot;)
output20 &lt;- readRDS(&quot;simulation_results/output20.rds&quot;)
output21 &lt;- readRDS(&quot;simulation_results/output21.rds&quot;)
output22 &lt;- readRDS(&quot;simulation_results/output22.rds&quot;)
output23 &lt;- readRDS(&quot;simulation_results/output23.rds&quot;)
output24 &lt;- readRDS(&quot;simulation_results/output24.rds&quot;)
output25 &lt;- readRDS(&quot;simulation_results/output25.rds&quot;)
output26 &lt;- readRDS(&quot;simulation_results/output26.rds&quot;)
output27 &lt;- readRDS(&quot;simulation_results/output27.rds&quot;)
output28 &lt;- readRDS(&quot;simulation_results/output28.rds&quot;)
output29 &lt;- readRDS(&quot;simulation_results/output29.rds&quot;)
output30 &lt;- readRDS(&quot;simulation_results/output30.rds&quot;)
output31 &lt;- readRDS(&quot;simulation_results/output31.rds&quot;)
output32 &lt;- readRDS(&quot;simulation_results/output32.rds&quot;)
output33 &lt;- readRDS(&quot;simulation_results/output33.rds&quot;)



if(!file.exists(&quot;simulation_results/output.rds&quot;)){
  output_combined &lt;- c(output1, output2, output3, output4, output5, output6,
                       output7, output8, output9, output10, output11, output12,
                       output13, output14, output15, output16, output17, output18,
                       output19, output20, output21, output22, output23, output24,
                       output25, output26, output27, output28, output29, output30,
                       output31, output32, output33)
  saveRDS(output_combined, &quot;simulation_results/output.rds&quot;)
} else{
  output &lt;- readRDS(&quot;simulation_results/output.rds&quot;)
}

final_SD_props_among_preselection_zygotes &lt;- sapply(output, calc_prop_SD)

final_SD_props_among_adults &lt;- sapply(output, function(pop){
  
  pop$prop &lt;- pop$prop * pop$egg_adult_viability # apply selection, then calculate allele freq of SD
  pop$prop &lt;- pop$prop / sum(pop$prop) 
  calc_prop_SD(pop)
  
})</code></pre>
<p><span class="math inline">\(~\)</span></p>
</div>
<div id="create-figures-2-and-s8" class="section level1">
<h1>Create Figures 2 and S8</h1>
<p><span class="math inline">\(~\)</span></p>
<pre class="r"><code># Combine the SD frequencies with the parameter data and transform data to improve clarity

plotting_data &lt;- parameters %&gt;%
  mutate(prop_SD = final_SD_props_among_adults,
         P1_actual = (normalP1 -  P1cost * normalP1) * 100,
         P2_actual = ((1 - normalP1) - P2cost * (1 - normalP1)) * 100,
         remating_ratio = prob_remating_SD / prob_remating_W) %&gt;% 
  mutate(remating_ratio = round(remating_ratio, digits = 1))

# Transform our empirical estimates to match the plotting data  

empirical_estimates_0.1 &lt;-
  empirical_estimates %&gt;% 
  mutate(W_remating_prob = 0.304,
         normalP1 = 0.1,
         P1_actual = (normalP1 -  P1cost * normalP1) * 100,
         lower_P1_actual = (normalP1 -  lower_P1cost * normalP1) * 100,
         upper_P1_actual = (normalP1 -  upper_P1cost * normalP1) * 100,
         P2_actual = ((1 - normalP1) - P2cost * (1 - normalP1)) * 100,
         lower_P2_actual = ((1 - normalP1) - lower_P2cost * (1 - normalP1)) * 100,
         upper_P2_actual = ((1 - normalP1) - upper_P2cost * (1 - normalP1)) * 100,
         remating_ratio = remating_prob / estimated_prob_remating_W) %&gt;% 
  mutate(remating_ratio = round(remating_ratio, digits = 1))

# Now make Figure 2

(p1 &lt;- 
    plotting_data %&gt;% 
    rename(`Wildtype P1` = normalP1,
           `Relative risk of sperm competition` = remating_ratio,
           `Relative mating success` = S_precop_WSD_male) %&gt;% 
    filter(k == 0.944, `Wildtype P1` == 0.1) %&gt;% 
    ggplot(aes(x = P1_actual, y = P2_actual)) +
    geom_blank() +
    geom_raster(aes(fill = 100 * prop_SD)) + 
    stat_contour(aes(z = 100 * prop_SD), colour = &quot;black&quot;, binwidth = 10) +
    stat_contour(aes(z = 100 * prop_SD), colour = &quot;black&quot;, breaks = 8, linetype = 2) +
    geom_errorbar(data = empirical_estimates_0.1,
                  aes(x = P1_actual, ymin = lower_P2_actual, ymax = upper_P2_actual),
                  width = 0.25, colour = &quot;grey11&quot;, alpha = 0.8) +
    geom_errorbarh(data = empirical_estimates_0.1,
                   aes(y = P2_actual, xmin = lower_P1_actual, xmax = upper_P1_actual),
                   colour = &quot;grey11&quot;, alpha = 0.8, height = 4) +
    geom_point(data = empirical_estimates_0.1,
               aes(x = P1_actual, y = P2_actual), 
               pch = 21, fill = &quot;#eb7f86&quot;, colour = &quot;grey11&quot;, size = 3) +
    facet_wrap(`Relative risk of sperm competition` ~ `Relative mating success`, 
               scales = &quot;free&quot;, nrow = 3, strip.position = c(&quot;top&quot;),
               labeller = label_glue(&#39;({.l}) Relative risk of sperm competition: {`Relative risk of sperm competition`}\nRelative mating success: {`Relative mating success`}&#39;)) +
    #annotate(&quot;text&quot;, label = c(&quot;italic(SD-5)&quot;, &quot;italic(SD-Mad)&quot;, &quot;italic(SD-72)&quot;),
    #x = c(1.5, 3, 5.2), y = c(80, 90, 88), size = 5, colour = &quot;black&quot;, parse = TRUE) +
    labs(x = &quot;% offspring sired by SD/+ male in the first mating role (P1)&quot;,
         y = &quot;% offspring sired by SD/+ male in the second mating role (P2)&quot;) +
    scale_fill_carto_c(type = &quot;sequential&quot;, palette = &quot;BluYl&quot;, direction = 1, name = &quot;% SD at equilibrium&quot;) +
    scale_x_continuous(expand = c(0, 0)) + 
    scale_y_continuous(expand = c(0, 0)) + # labels = c(0, 25, 50, 75, 90)) +
    theme(panel.border = element_rect(fill = NA, colour = &quot;black&quot;, size = .8),
          strip.background = element_rect(colour = &quot;black&quot;, fill = &quot;Aliceblue&quot;, size = .8))
)</code></pre>
<p><img src="figure/Population_genetic_model.Rmd/unnamed-chunk-9-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/f30258aac3f1b78957f599e4001cf340ea4a4304/docs/figure/Population_genetic_model.Rmd/unnamed-chunk-9-1.png" target="_blank">f30258a</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Population_genetic_model.Rmd/unnamed-chunk-9-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure 2</strong>: Predicted equilibrium frequency of the <em>SD</em> allele, calculated from the population genetic model. The plot depicts the interaction between the P1 (x-axis) and P2 (y-axis) costs suffered by <em>SD/+</em> males in their effects on the equilibrium frequency of <em>SD</em> (shown by the colour scale and 10% contour lines). The dashed line shows an equilibrium frequency of 8%, the upper estimate for <em>SD</em> alleles in natural populations. The plot is split into nine panels, with varying levels of <em>SD/+</em> male mating success (<span class="math inline">\(S_{precop}\)</span>) across the columns and an increasing risk of sperm competition caused by a female remating after first mating to an <em>SD/+</em> male, <span class="math inline">\(p_{SD/+}\)</span>, down the rows (values correspond to the risk of sperm competition we estimated in Experiment 2). The three orange points (with associated 95% credible intervals) in each panel represent males carrying <em>SD-5</em> (leftmost point), and <em>SD-72</em> and <em>SD-Mad</em> (rightmost points); these points show where males carrying each <em>SD</em> variant fall in the figure’s parameter space. In the parameter space presented, <span class="math inline">\(k\)</span> = 0.944, <span class="math inline">\(P1_{normal}\)</span> = 0.1, and <em>SD</em> homozygotes are non-viable.</p>
<p><span class="math inline">\(~\)</span></p>
<p>We measured P1 and P2 with a set interval of four days between matings. While we found that many females did not remate within the three-hour allotted mating period, many studies show that a low % of females will remate much faster than this, and that the interval between matings affects P1 and P2. Matings that occur closer temporally result in higher P1 values (e.g. <a href="https://dx.doi.org/10.1002/evl3.50">Laturney, Eijk and Billeter</a>, 2018, <em>Evolution Letters</em>). In addition, many populations/species do not exhibit the extreme second male precedence we observed in our study. Therefore, we also modelled the scenario where there was no effect of mating order on fertilisation success, that is <span class="math inline">\(P1_{normal}\)</span> = 0.5. The results are shown in <strong>Figure S8</strong> below.</p>
<pre class="r"><code># Transform our empirical estimates to match the plotting data  

empirical_estimates_0.5 &lt;-
  empirical_estimates %&gt;% 
  mutate(W_remating_prob = 0.30,
         normalP1 = 0.5,
         P1_actual = (normalP1 -  P1cost * normalP1) * 100,
         lower_P1_actual = (normalP1 -  lower_P1cost * normalP1) * 100,
         upper_P1_actual = (normalP1 -  upper_P1cost * normalP1) * 100,
         P2_actual = ((1 - normalP1) - P2cost * (1 - normalP1)) * 100,
         lower_P2_actual = ((1 - normalP1) - lower_P2cost * (1 - normalP1)) * 100,
         upper_P2_actual = ((1 - normalP1) - upper_P2cost * (1 - normalP1)) * 100,
         remating_ratio = remating_prob / estimated_prob_remating_W) %&gt;% 
  mutate(remating_ratio = round(remating_ratio, digits = 1))

# Make Figure S8

(p2 &lt;-
  plotting_data %&gt;% 
  rename(`Wildtype P1` = normalP1,
         `Relative risk of sperm competition` = remating_ratio,
         `Relative mating success` = S_precop_WSD_male) %&gt;% 
  filter(k == 0.944, `Wildtype P1` == 0.5) %&gt;% 
 ggplot(aes(x = P1_actual, y = P2_actual)) +
    geom_blank() +
    geom_raster(aes(fill = 100 * prop_SD)) + 
    stat_contour(aes(z = 100 * prop_SD), colour = &quot;black&quot;, binwidth = 10) +
    stat_contour(aes(z = 100 * prop_SD), colour = &quot;black&quot;, breaks = 8, linetype = 2) +
    geom_errorbar(data = empirical_estimates_0.5,
                  aes(x = P1_actual, ymin = lower_P2_actual, ymax = upper_P2_actual),
                  width = 2, colour = &quot;grey11&quot;, alpha = 0.8) +
    geom_errorbarh(data = empirical_estimates_0.5,
                   aes(y = P2_actual, xmin = lower_P1_actual, xmax = upper_P1_actual),
                   colour = &quot;grey11&quot;, alpha = 0.8, height = 2) +
    geom_point(data = empirical_estimates_0.5,
               aes(x = P1_actual, y = P2_actual), 
               pch = 21, fill = &quot;#eb7f86&quot;, colour = &quot;grey11&quot;, size = 3) +
    facet_wrap(`Relative risk of sperm competition` ~ `Relative mating success`, 
               scales = &quot;free&quot;, nrow = 3, strip.position = c(&quot;top&quot;),
               labeller = label_glue(&#39;({.l}) Relative risk of sperm competition: {`Relative risk of sperm competition`}\nRelative mating success: {`Relative mating success`}&#39;)) +
    #annotate(&quot;text&quot;, label = c(&quot;italic(SD-5)&quot;, &quot;italic(SD-Mad)&quot;, &quot;italic(SD-72)&quot;),
    #x = c(1.5, 3, 5.2), y = c(80, 90, 88), size = 5, colour = &quot;black&quot;, parse = TRUE) +
    labs(x = &quot;% offspring sired by SD/+ male in the first mating role (P1)&quot;,
         y = &quot;% offspring sired by SD/+ male in the second mating role (P2)&quot;) +
    scale_fill_carto_c(type = &quot;sequential&quot;, palette = &quot;BluYl&quot;, direction = 1, name = &quot;% SD at equilibrium&quot;) +
    scale_x_continuous(expand = c(0, 0)) + 
    scale_y_continuous(expand = c(0, 0)) + # labels = c(0, 25, 50, 75, 90)) +
    theme(panel.border = element_rect(fill = NA, colour = &quot;black&quot;, size = .8),
          strip.background = element_rect(colour = &quot;black&quot;, fill = &quot;Aliceblue&quot;, size = .8))
)</code></pre>
<p><img src="figure/Population_genetic_model.Rmd/unnamed-chunk-10-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/f30258aac3f1b78957f599e4001cf340ea4a4304/docs/figure/Population_genetic_model.Rmd/unnamed-chunk-10-1.png" target="_blank">f30258a</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-25
</td>
</tr>
<tr>
<td>
<a href="https://github.com/tomkeaney/SD_sexual_selection/blob/ddaa2d189d24c2e6ccdebef310708f44cc8383a4/docs/figure/Population_genetic_model.Rmd/unnamed-chunk-10-1.png" target="_blank">ddaa2d1</a>
</td>
<td>
tkeaney
</td>
<td>
2021-05-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><strong>Figure S8</strong>: Predicted equilibrium frequency of the <em>SD</em> allele, calculated from the population genetic model. The plot depicts the interaction between the P1 (x-axis) and P2 (y-axis) costs suffered by <em>SD/+</em> males in their effects on the equilibrium frequency of <em>SD</em> (shown by the colour scale and 10% contour lines). The dashed line shows an equilibrium frequency of 8%, the upper estimate for <em>SD</em> alleles in natural populations. The plot is split into nine panels, with varying levels of <em>SD/+</em> male mating success across the rows and increasing likelihoods of a female remating after first mating to an <em>SD/+</em> male, <span class="math inline">\(p_{SD/+}\)</span>, down the columns. The three orange points (with associated 95% credible intervals) in each panel represent males carrying the <em>SD-5</em> (leftmost point), and <em>SD-72</em> and <em>SD-Mad</em> (rightmost points) alleles; these points show where males carrying these alleles fall in the figure’s parameter space, as measured in Experiment 2. In the parameter space presented here, <span class="math inline">\(k\)</span> = 0.944 (our highest estimated value), <span class="math inline">\(P1_{normal}\)</span> = 0.5, <em>SD</em> homozygotes are non-viable and <em>SD</em> heterozygotes suffer no fitness costs outside of mating success and sperm competition.</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.5 (2021-03-31)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRblas.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] stickylabeller_0.0.0.9100 pander_0.6.3             
 [3] kableExtra_1.3.4          ggpubr_0.4.0             
 [5] rcartocolor_2.0.0         patchwork_1.1.1          
 [7] forcats_0.5.1             stringr_1.4.0            
 [9] dplyr_1.0.5               purrr_0.3.4              
[11] readr_1.4.0               tidyr_1.1.3              
[13] tibble_3.1.1              ggplot2_3.3.3            
[15] tidyverse_1.3.1           workflowr_1.6.2          

loaded via a namespace (and not attached):
 [1] fs_1.5.0          lubridate_1.7.10  webshot_0.5.2     httr_1.4.2       
 [5] rprojroot_2.0.2   tools_4.0.5       backports_1.2.1   bslib_0.2.4      
 [9] utf8_1.2.1        R6_2.5.0          DBI_1.1.1         colorspace_2.0-0 
[13] withr_2.4.2       tidyselect_1.1.0  curl_4.3          compiler_4.0.5   
[17] git2r_0.28.0      cli_2.4.0         rvest_1.0.0       xml2_1.3.2       
[21] isoband_0.2.4     labeling_0.4.2    sass_0.3.1        scales_1.1.1     
[25] systemfonts_1.0.1 digest_0.6.27     foreign_0.8-81    rmarkdown_2.7    
[29] svglite_2.0.0     rio_0.5.26        pkgconfig_2.0.3   htmltools_0.5.1.1
[33] dbplyr_2.1.1      highr_0.9         rlang_0.4.10      readxl_1.3.1     
[37] rstudioapi_0.13   jquerylib_0.1.3   generics_0.1.0    farver_2.1.0     
[41] jsonlite_1.7.2    zip_2.1.1         car_3.0-10        magrittr_2.0.1   
[45] Rcpp_1.0.6        munsell_0.5.0     fansi_0.4.2       abind_1.4-5      
[49] lifecycle_1.0.0   stringi_1.5.3     whisker_0.4       yaml_2.2.1       
[53] carData_3.0-4     grid_4.0.5        promises_1.2.0.1  crayon_1.4.1     
[57] haven_2.4.0       hms_1.0.0         knitr_1.32        pillar_1.6.0     
[61] ggsignif_0.6.1    reprex_2.0.0      glue_1.4.2        evaluate_0.14    
[65] data.table_1.14.0 modelr_0.1.8      vctrs_0.3.7       httpuv_1.5.5     
[69] cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1  xfun_0.22        
[73] openxlsx_4.2.3    broom_0.7.6       rstatix_0.7.0     later_1.1.0.1    
[77] viridisLite_0.4.0 ellipsis_0.3.1   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
